<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Frano Malinarich G.">
<meta name="dcterms.date" content="2025-05-25">
<title>Spatial Tutorial: Mouse Brain Section (Coronal) – 10x Genomics</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="260225_Spatial_Mouse_Brain_Coronal_10x_files/libs/clipboard/clipboard.min.js"></script>
<script src="260225_Spatial_Mouse_Brain_Coronal_10x_files/libs/quarto-html/quarto.js"></script>
<script src="260225_Spatial_Mouse_Brain_Coronal_10x_files/libs/quarto-html/popper.min.js"></script>
<script src="260225_Spatial_Mouse_Brain_Coronal_10x_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="260225_Spatial_Mouse_Brain_Coronal_10x_files/libs/quarto-html/anchor.min.js"></script>
<link href="260225_Spatial_Mouse_Brain_Coronal_10x_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="260225_Spatial_Mouse_Brain_Coronal_10x_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="260225_Spatial_Mouse_Brain_Coronal_10x_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="260225_Spatial_Mouse_Brain_Coronal_10x_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="260225_Spatial_Mouse_Brain_Coronal_10x_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><link href="260225_Spatial_Mouse_Brain_Coronal_10x_files/libs/tabwid-1.1.3/tabwid.css" rel="stylesheet">
<script src="260225_Spatial_Mouse_Brain_Coronal_10x_files/libs/tabwid-1.1.3/tabwid.js"></script><link rel="stylesheet" href="styles.css">
</head>
<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>

  <ul>
<li><a href="#spatial-transcriptomics" id="toc-spatial-transcriptomics" class="nav-link active" data-scroll-target="#spatial-transcriptomics"><span class="header-section-number">1</span> Spatial transcriptomics</a></li>
  <li><a href="#loading-installing-package" id="toc-loading-installing-package" class="nav-link" data-scroll-target="#loading-installing-package"><span class="header-section-number">2</span> <em>Loading / Installing package</em></a></li>
  <li><a href="#load-spatial-rna-sequencing-data" id="toc-load-spatial-rna-sequencing-data" class="nav-link" data-scroll-target="#load-spatial-rna-sequencing-data"><span class="header-section-number">3</span> <em>Load spatial RNA sequencing data</em></a></li>
  <li><a href="#qc-analysis" id="toc-qc-analysis" class="nav-link" data-scroll-target="#qc-analysis"><span class="header-section-number">4</span> <em>QC analysis</em></a></li>
  <li><a href="#data-pre-processing-normalization-scaling-and-findspatiallyvariablefeatures" id="toc-data-pre-processing-normalization-scaling-and-findspatiallyvariablefeatures" class="nav-link" data-scroll-target="#data-pre-processing-normalization-scaling-and-findspatiallyvariablefeatures"><span class="header-section-number">5</span> <em>Data pre-processing: normalization, scaling and FindSpatiallyVariableFeatures</em></a></li>
  <li><a href="#dimensional-reduction-clustering-and-visualizations" id="toc-dimensional-reduction-clustering-and-visualizations" class="nav-link" data-scroll-target="#dimensional-reduction-clustering-and-visualizations"><span class="header-section-number">6</span> <em>Dimensional reduction, clustering and visualizations</em></a></li>
  <li><a href="#summary-table-of-spots-identifiers-and-spot-amount" id="toc-summary-table-of-spots-identifiers-and-spot-amount" class="nav-link" data-scroll-target="#summary-table-of-spots-identifiers-and-spot-amount"><span class="header-section-number">7</span> <em>Summary table of spots identifiers and spot amount</em></a></li>
  <li><a href="#differential-expression-gene-deg-analysis-across-spatial-clusters" id="toc-differential-expression-gene-deg-analysis-across-spatial-clusters" class="nav-link" data-scroll-target="#differential-expression-gene-deg-analysis-across-spatial-clusters"><span class="header-section-number">8</span> <em>Differential Expression Gene (DEG) analysis across spatial clusters</em></a></li>
  <li><a href="#brain-region-annotation-summary-table" id="toc-brain-region-annotation-summary-table" class="nav-link" data-scroll-target="#brain-region-annotation-summary-table"><span class="header-section-number">9</span> <em>Brain region annotation / Summary table</em></a></li>
  <li><a href="#converting-seurat-object-brain-into-h5ad-format" id="toc-converting-seurat-object-brain-into-h5ad-format" class="nav-link" data-scroll-target="#converting-seurat-object-brain-into-h5ad-format"><span class="header-section-number">10</span> <em>Converting Seurat object “brain” into H5AD format</em></a></li>
  </ul></nav>
</div>
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Spatial Tutorial: Mouse Brain Section (Coronal) – 10x Genomics</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Frano Malinarich G. </p>
          </div>
  </div>

    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">May 25, 2025</p>
    </div>
  </div>


  </div>



</header><section id="spatial-transcriptomics" class="level2" data-number="1"><h2 data-number="1" class="anchored" data-anchor-id="spatial-transcriptomics">
<span class="header-section-number">1</span> Spatial transcriptomics</h2>
<p><strong>Tutorials:</strong></p>
<p><a href="https://satijalab.org/seurat/articles/spatial_vignette" class="uri">https://satijalab.org/seurat/articles/spatial_vignette</a></p>
<p><a href="https://www.youtube.com/watch?v=L_7VdCeJ4Z8&amp;list=PLOLdjuxsfI4MXCyKZDY0jKU-K6Uo4o1_d&amp;index=2." class="uri">https://www.youtube.com/watch?v=L_7VdCeJ4Z8&amp;list=PLOLdjuxsfI4MXCyKZDY0jKU-K6Uo4o1_d&amp;index=2.</a></p>
<p><strong>Data source: 10XGenomics, Mouse brain section (coronal)</strong> <a href="https://www.10xgenomics.com/datasets/mouse-brain-section-coronal-1-standard-1-0-0" class="uri">https://www.10xgenomics.com/datasets/mouse-brain-section-coronal-1-standard-1-0-0</a></p>
</section><section id="loading-installing-package" class="level2" data-number="2"><h2 data-number="2" class="anchored" data-anchor-id="loading-installing-package">
<span class="header-section-number">2</span> <em>Loading / Installing package</em>
</h2>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Installing necessary packages</span></span>
<span><span class="co"># install.packages("extrafont")</span></span>
<span><span class="co"># BiocManager::install("SpatialExperiment")</span></span>
<span><span class="co"># install.packages('Rfast2')</span></span>
<span><span class="co"># install.packages("Boruta")</span></span>
<span><span class="co"># Install Bioconductor manager (if not already installed)</span></span>
<span><span class="co"># Install mouse annotation database</span></span>
<span><span class="co"># BiocManager::install("org.Mm.eg.db")</span></span>
<span></span>
<span><span class="co"># Load packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://conflicted.r-lib.org/">conflicted</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com">patchwork</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ardata-fr.github.io/flextable-book/">flextable</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/wch/extrafont">extrafont</a></span><span class="op">)</span></span>
<span><span class="co"># font_import()</span></span>
<span><span class="co"># loadfonts(device = "all")</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ardata-fr.github.io/officeverse/">officer</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/">cowplot</a></span><span class="op">)</span> <span class="co"># For plot arrangement</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span> <span class="co"># For grid-based layout</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">grid</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RfastOfficial/Rfast2">Rfast2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Mm.eg.db</span><span class="op">)</span>  <span class="co"># For Mus musculus (mouse)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/contribution-knowledge-mining/">clusterProfiler</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drighelli/SpatialExperiment">SpatialExperiment</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Resolve conflicts</span></span>
<span><span class="fu"><a href="https://conflicted.r-lib.org/reference/conflict_prefer.html">conflict_prefer</a></span><span class="op">(</span><span class="st">"filter"</span>, <span class="st">"dplyr"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://conflicted.r-lib.org/reference/conflict_prefer.html">conflict_prefer</a></span><span class="op">(</span><span class="st">"combine"</span>, <span class="st">"dplyr"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://conflicted.r-lib.org/reference/conflict_prefer.html">conflict_prefer</a></span><span class="op">(</span><span class="st">"setdiff"</span>, <span class="st">"base"</span><span class="op">)</span>  <span class="co"># Specify base::setdiff to avoid errors</span></span>
<span><span class="fu"><a href="https://conflicted.r-lib.org/reference/conflict_prefer.html">conflict_prefer</a></span><span class="op">(</span><span class="st">"rename"</span>, <span class="st">"dplyr"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://conflicted.r-lib.org/reference/conflict_prefer.html">conflict_prefer</a></span><span class="op">(</span><span class="st">"select"</span>, <span class="st">"dplyr"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://conflicted.r-lib.org/reference/conflict_prefer.html">conflict_prefer</a></span><span class="op">(</span><span class="st">"width"</span>, <span class="st">"flextable"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="load-spatial-rna-sequencing-data" class="level2" data-number="3"><h2 data-number="3" class="anchored" data-anchor-id="load-spatial-rna-sequencing-data">
<span class="header-section-number">3</span> <em>Load spatial RNA sequencing data</em>
</h2>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Load data</span></span>
<span></span>
<span><span class="co"># data_dir = "~/path/to/your/data/"</span></span>
<span></span>
<span><span class="va">brain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/Load10X_Spatial.html">Load10X_Spatial</a></span><span class="op">(</span>data.dir <span class="op">=</span> <span class="va">data_dir</span>,</span>
<span>                         filename <span class="op">=</span> <span class="st">"V1_Adult_Mouse_Brain_filtered_feature_bc_matrix.h5"</span>,</span>
<span>                         assay <span class="op">=</span> <span class="st">"Spatial"</span>,</span>
<span>                         slice <span class="op">=</span> <span class="st">"slice1"</span>,</span>
<span>                         filter.matrix <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                         to.upper <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                         image <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># &gt; brain</span></span>
<span><span class="co"># An object of class Seurat </span></span>
<span><span class="co"># 31053 features across 2698 samples within 1 assay </span></span>
<span><span class="co"># Active assay: Spatial (31053 features, 0 variable features)</span></span>
<span><span class="co">#  1 layer present: counts</span></span>
<span><span class="co">#  1 spatial field of view present: slice1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="qc-analysis" class="level2" data-number="4"><h2 data-number="4" class="anchored" data-anchor-id="qc-analysis">
<span class="header-section-number">4</span> <em>QC analysis</em>
</h2>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># View first few rows of metadata</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>                      orig.ident nCount_Spatial nFeature_Spatial
AAACAAGTATCTCCCA-1 SeuratProject          19340             4928
AAACAATCTACTAGCA-1 SeuratProject          13750             3448
AAACACCAATAACTGC-1 SeuratProject          32710             6022
AAACAGAGCGACTCCT-1 SeuratProject          15909             4311
AAACCGGGTAGGTACC-1 SeuratProject          31856             5787
AAACCGTTCGTCCAGG-1 SeuratProject          30862             5898</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># view(brain@meta.data)</span></span>
<span><span class="co"># View(brain) #'View' and 'view' cannot be rendered in HTML so comment out, replace with 'head()'</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1] "orig.ident"       "nCount_Spatial"   "nFeature_Spatial"</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Mitochondrial Gene Percentage incorporated into brain@meta.data</span></span>
<span><span class="va">brain</span><span class="op">[[</span><span class="st">"percent.mt"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/PercentageFeatureSet.html">PercentageFeatureSet</a></span><span class="op">(</span><span class="va">brain</span>, pattern <span class="op">=</span> <span class="st">"^mt-"</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>                      orig.ident nCount_Spatial nFeature_Spatial percent.mt
AAACAAGTATCTCCCA-1 SeuratProject          19340             4928   19.94312
AAACAATCTACTAGCA-1 SeuratProject          13750             3448   23.76000
AAACACCAATAACTGC-1 SeuratProject          32710             6022   15.01070
AAACAGAGCGACTCCT-1 SeuratProject          15909             4311   20.55440
AAACCGGGTAGGTACC-1 SeuratProject          31856             5787   21.01017
AAACCGTTCGTCCAGG-1 SeuratProject          30862             5898   16.90104</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Violin plot for spatial counts</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/VlnPlot.html">VlnPlot</a></span><span class="op">(</span><span class="va">brain</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"nCount_Spatial"</span>, <span class="st">"nFeature_Spatial"</span>, <span class="st">"percent.mt"</span><span class="op">)</span>, pt.size <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>   <span class="op">+</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html">NoLegend</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Filter low-quality spots</span></span>
<span><span class="co"># brain &lt;- subset(</span></span>
<span><span class="co">#   brain,</span></span>
<span><span class="co">#   nCount_Spatial &gt; 500 &amp; </span></span>
<span><span class="co">#   nFeature_Spatial &gt; 300 &amp; </span></span>
<span><span class="co">#   percent.mt &lt; 20</span></span>
<span><span class="co"># )</span></span>
<span></span>
<span><span class="co"># Ribosomal Protein Gene Percentage </span></span>
<span><span class="va">brain</span><span class="op">[[</span><span class="st">"percent.rb"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/PercentageFeatureSet.html">PercentageFeatureSet</a></span><span class="op">(</span><span class="va">brain</span>, pattern <span class="op">=</span> <span class="st">"^Rp[sl]"</span><span class="op">)</span>  <span class="co"># Elevated ribosomal RNA can indicate residual cytoplasmic RNA or stress.</span></span>
<span></span>
<span><span class="co"># High percent.rb + low nCount_Spatial → Technical artifact (filter)</span></span>
<span><span class="co"># High percent.rb + high percent.mt → Cell stress (consider filtering).</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeatureScatter.html">FeatureScatter</a></span><span class="op">(</span><span class="va">brain</span>, feature1 <span class="op">=</span> <span class="st">"percent.rb"</span>, feature2 <span class="op">=</span> <span class="st">"nCount_Spatial"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/FeatureScatter.html">FeatureScatter</a></span><span class="op">(</span><span class="va">brain</span>, feature1 <span class="op">=</span> <span class="st">"percent.rb"</span>, feature2 <span class="op">=</span> <span class="st">"percent.mt"</span><span class="op">)</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-4-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Identify spots with high ribosomal gene expression</span></span>
<span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/WhichCells.html">WhichCells</a></span><span class="op">(</span><span class="va">brain</span>, expression <span class="op">=</span> <span class="va">percent.rb</span> <span class="op">&gt;</span> <span class="fl">19</span><span class="op">)</span> <span class="co"># only one spot "CTCTGCAGGCATTCTT-1"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1] "CTCTGCAGGCATTCTT-1"</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Summary statistics for key QC metrics</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">brain</span><span class="op">$</span><span class="va">nCount_Spatial</span><span class="op">)</span>  <span class="co"># Confirm min &gt;500</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.
    558   19624   27198   29965   37121  149086 </code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">brain</span><span class="op">$</span><span class="va">nFeature_Spatial</span><span class="op">)</span>  <span class="co"># Confirm min &gt;300</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.
    403    4737    5798    5719    6752   10691 </code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">brain</span><span class="op">$</span><span class="va">percent.mt</span><span class="op">)</span>  <span class="co"># Confirm max &lt;20%</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.
  5.727  13.666  15.699  16.425  18.470  44.658 </code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">brain</span><span class="op">$</span><span class="va">percent.rb</span><span class="op">)</span> <span class="co"># Confim &gt; 20</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.
  3.519   6.353   7.400   7.777   8.978  23.129 </code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># The spot with percent.rb &gt; 20 does not show low nCount_Spatial or high percent.mt.</span></span>
<span><span class="co"># Therefore, this elevated ribosomal signal may not indicate a technical artifact or stress.</span></span>
<span><span class="co"># Filtering may be skipped in this case.</span></span>
<span></span>
<span><span class="co"># Violin plot for spatial counts +  Ribosomal Protein Gene Percentage (percent.rb)</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/VlnPlot.html">VlnPlot</a></span><span class="op">(</span><span class="va">brain</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"nCount_Spatial"</span>, <span class="st">"nFeature_Spatial"</span>, <span class="st">"percent.mt"</span>, <span class="st">"percent.rb"</span><span class="op">)</span>, pt.size <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>   <span class="op">+</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html">NoLegend</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-4-3.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Spatial visualization of percent.rb across the brain section</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">brain</span>, features <span class="op">=</span> </span>
<span>                     <span class="st">"percent.rb"</span>, alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Spatial visualization of nCount_Spatial across the brain section</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">brain</span>, features <span class="op">=</span> </span>
<span>                     <span class="st">"nCount_Spatial"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Spatial visualization of nFeature_Spatial across the brain section</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">brain</span>, features <span class="op">=</span> </span>
<span>                     <span class="st">"nFeature_Spatial"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Spatial visualization of nFeature_Spatial across the brain section</span></span>
<span><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">brain</span>, features <span class="op">=</span> </span>
<span>                     <span class="st">"percent.mt"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span> <span class="op">+</span> <span class="va">p3</span> <span class="op">+</span> <span class="va">p4</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-4-4.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Range of counts and features</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">nCount_Spatial</span><span class="op">)</span> <span class="co"># 558 149086</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1]    558 149086</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">nFeature_Spatial</span><span class="op">)</span> <span class="co"># 403 10691</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1]   403 10691</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">percent.mt</span><span class="op">)</span> <span class="co"># 5.726924 44.657763</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1]  5.726924 44.657763</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">percent.rb</span><span class="op">)</span> <span class="co"># 3.518981 23.129093</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1]  3.518981 23.129093</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># normally, the brain has a high % of ribosome</span></span>
<span><span class="co"># 23.1%: Moderately high but plausible for brain regions with active protein synthesis (e.g., hippocampus).</span></span>
<span><span class="co"># Do NOT filter solely on ribosomal % unless: Spots are outliers (e.g., &gt;25%) AND correlate with low RNA quality.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="data-pre-processing-normalization-scaling-and-findspatiallyvariablefeatures" class="level2" data-number="5"><h2 data-number="5" class="anchored" data-anchor-id="data-pre-processing-normalization-scaling-and-findspatiallyvariablefeatures">
<span class="header-section-number">5</span> <em>Data pre-processing: normalization, scaling and FindSpatiallyVariableFeatures</em>
</h2>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">meta.data</span>,<span class="fl">3</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>                      orig.ident nCount_Spatial nFeature_Spatial percent.mt
AAACAAGTATCTCCCA-1 SeuratProject          19340             4928   19.94312
AAACAATCTACTAGCA-1 SeuratProject          13750             3448   23.76000
AAACACCAATAACTGC-1 SeuratProject          32710             6022   15.01070
                   percent.rb
AAACAAGTATCTCCCA-1   3.536711
AAACAATCTACTAGCA-1   6.487273
AAACACCAATAACTGC-1  10.189544</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">brain</span><span class="op">@</span><span class="va">commands</span><span class="op">$</span><span class="va">SCTransform.Spatial</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#########</span></span>
<span><span class="co">#########</span></span>
<span><span class="co">#########</span></span>
<span></span>
<span><span class="co"># 1. Preprocessing</span></span>
<span><span class="va">brain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SCTransform.html">SCTransform</a></span><span class="op">(</span><span class="va">brain</span>, assay <span class="op">=</span> <span class="st">"Spatial"</span>, new.assay.name <span class="op">=</span> <span class="st">"SCT"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/ObjectAccess.html">Assays</a></span><span class="op">(</span><span class="va">brain</span><span class="op">)</span> <span class="co"># expected output: "Spatial" "SCT"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1] "Spatial" "SCT"    </code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># The second step FindSpatiallyVariableFeatures() is OPTIONAL!</span></span>
<span><span class="co"># This step: </span></span>
<span><span class="co"># 1. Used to find genes with spatial expression gradients (e.g., Moran's I).</span></span>
<span><span class="co"># 2. Not required for standard clustering/visualization (UMAP, spatial plots).</span></span>
<span></span>
<span><span class="co"># 2. Set SCT as active assay and find spatially variable features</span></span>
<span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/DefaultAssay.html">DefaultAssay</a></span><span class="op">(</span><span class="va">brain</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"SCT"</span></span>
<span></span>
<span><span class="va">brain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindSpatiallyVariableFeatures.html">FindSpatiallyVariableFeatures</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>,</span>
<span>  assay <span class="op">=</span> <span class="st">"SCT"</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/VariableFeatures.html">VariableFeatures</a></span><span class="op">(</span><span class="va">brain</span><span class="op">)</span>, <span class="co"># this argument is important!</span></span>
<span>  selection.method <span class="op">=</span> <span class="st">"moransi"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Warning message:</span></span>
<span><span class="co"># In dist(x = pos) : NAs introduced by coercion</span></span>
<span><span class="co"># suggests that there are missing or non-numeric values in the spatial coordinates used for the Moran’s I computation.</span></span>
<span></span>
<span><span class="co"># The "features = VariableFeatures(brain)" allows to get more biologically meaningful spatially variable genes.</span></span>
<span><span class="co"># Without "features = VariableFeatures(brain)" </span></span>
<span></span>
<span><span class="co"># Verify the presence of NAs in 'coords'</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">images</span><span class="op">$</span><span class="va">slice1</span><span class="op">$</span><span class="va">centroids</span><span class="op">@</span><span class="va">coords</span><span class="op">)</span><span class="op">)</span> <span class="co"># If you get 0, the coordinates are fine. There are no NAs.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb40"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">images</span><span class="op">$</span><span class="va">slice1</span><span class="op">$</span><span class="va">centroids</span><span class="op">@</span><span class="va">coords</span><span class="op">)</span> <span class="co"># coordinates 'x' and 'y' are numeric "num [1:2698, 1:2] 7237 1611 8315 2927 6280 ..."</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code> num [1:2698, 1:2] 7237 1611 8315 2927 6280 ...
 - attr(*, "dimnames")=List of 2
  ..$ : NULL
  ..$ : chr [1:2] "x" "y"</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb42"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">images</span><span class="op">$</span><span class="va">slice1</span><span class="op">$</span><span class="va">centroids</span><span class="op">@</span><span class="va">coords</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1] "x" "y"</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb44"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># brain@images$slice1$centroids@coords</span></span>
<span></span>
<span><span class="co"># Check if Spatially Variable Features Exist</span></span>
<span><span class="co"># Check if Moran's I results are stored in metadata</span></span>
<span><span class="va">svf_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/VariableFeatures.html">SVFInfo</a></span><span class="op">(</span><span class="va">brain</span>, assay <span class="op">=</span> <span class="st">"SCT"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># If empty, check the assay's meta.features directly</span></span>
<span><span class="va">moran_results</span> <span class="op">&lt;-</span> <span class="va">brain</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">SCT</span><span class="op">@</span><span class="va">meta.features</span></span>
<span></span>
<span><span class="co"># Look for columns like "MoransI_observed" or "moransi"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">moran_results</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1] "MoransI_observed"                "MoransI_p.value"
[3] "moransi.spatially.variable"      "moransi.spatially.variable.rank"</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb46"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Expected results</span></span>
<span><span class="co"># [1] "MoransI_observed"                "MoransI_p.value"                </span></span>
<span><span class="co"># [3] "moransi.spatially.variable"      "moransi.spatially.variable.rank"</span></span>
<span></span>
<span><span class="co"># getting this output means that FindSpatiallyVariableFeatures() function worked, and results are stored in the SCT assay’s metadata.</span></span>
<span></span>
<span><span class="co"># View(brain)</span></span>
<span></span>
<span><span class="co"># Check How Many Genes Have Non-NA Moran’s I Values</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">moran_results</span><span class="op">$</span><span class="va">MoransI_observed</span><span class="op">)</span><span class="op">)</span> <span class="co"># 3000</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1] 3000</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb48"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># 1️⃣ There are 3000 genes with valid Moran’s I values, which is a normal number for spatially variable genes.</span></span>
<span><span class="co"># Why Are Many Other Genes NA? because only Highly Variable Genes Were Analyzed.</span></span>
<span><span class="co"># By default, FindSpatiallyVariableFeatures() only computes Moran’s I for genes in VariableFeatures(brain) (often 3000 genes).</span></span>
<span><span class="co"># Other genes were not tested, which is why they show NA.</span></span>
<span><span class="co"># 2️⃣  Genes with Low or No Expression</span></span>
<span><span class="co"># If a gene has low counts across spots, it gets excluded.</span></span>
<span><span class="co"># Moran’s I needs enough expression variation to be meaningful.</span></span>
<span><span class="co"># 3️⃣ Sparse or Noisy Data</span></span>
<span><span class="co"># Some genes might be filtered due to low-quality spots or technical artifacts.</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">moran_results</span><span class="op">$</span><span class="va">MoransI_observed</span>, breaks <span class="op">=</span> <span class="fl">50</span>, main <span class="op">=</span> <span class="st">"Distribution of Moran's I"</span>, xlab <span class="op">=</span> <span class="st">"Moran's I Score"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb49"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">moran_results</span><span class="op">$</span><span class="va">MoransI_observed</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's
 -0.001   0.077   0.132   0.158   0.215   0.663   15078 </code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb51"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Interpretation:</span></span>
<span><span class="co"># 14968 genes have NA values were ignored because they were not selected as highly variable genes (via " VariableFeatures(brain)")</span></span>
<span><span class="co"># 3000 genes were selected as the most spatial variable genes.</span></span>
<span><span class="co"># This filtering step reduces noise and focuses on biologically relevant genes. It's more meaningful.</span></span>
<span><span class="co"># Moran’s I values range from -0.003 to 0.685.</span></span>
<span><span class="co"># Median Moran’s I = 0.149 → most genes show moderate spatial autocorrelation.</span></span>
<span><span class="co"># Mean Moran’s I = 0.175, with a max value of 0.685 → some genes have strong spatial patterns.</span></span>
<span><span class="co"># 1️⃣ Your spatial analysis worked correctly → The non-NA genes have meaningful Moran’s I values.</span></span>
<span><span class="co"># 2️⃣ The distribution is reasonable → Many genes have moderate spatial variation, and some have strong spatial patterns.</span></span>
<span><span class="co"># 3️⃣ You can confidently use the top Moran’s I genes for visualization &amp; clustering.</span></span>
<span></span>
<span><span class="co"># </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/VariableFeatures.html">VariableFeatures</a></span><span class="op">(</span><span class="va">brain</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1] 3000</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb53"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># How to use the Spatial Variable Genes (SVG)</span></span>
<span><span class="co"># Identify Top Spatially Variable Genes</span></span>
<span><span class="va">top_spatial_genes</span> <span class="op">&lt;-</span> <span class="va">moran_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">moransi.spatially.variable</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span><span class="va">MoransI_observed</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize Spatial Expression</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialFeaturePlot</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">top_spatial_genes</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>,  <span class="co"># Top 4 genes</span></span>
<span>  pt.size.factor <span class="op">=</span> <span class="fl">2.5</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-5-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</section><section id="dimensional-reduction-clustering-and-visualizations" class="level2" data-number="6"><h2 data-number="6" class="anchored" data-anchor-id="dimensional-reduction-clustering-and-visualizations">
<span class="header-section-number">6</span> <em>Dimensional reduction, clustering and visualizations</em>
</h2>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb54"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Similar to scRNAseq analysis</span></span>
<span><span class="va">brain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html">RunPCA</a></span><span class="op">(</span><span class="va">brain</span>, assay <span class="op">=</span> <span class="st">"SCT"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># ElbowPlot to determine optimal PCs</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/ElbowPlot.html">ElbowPlot</a></span><span class="op">(</span><span class="va">brain</span>, ndims <span class="op">=</span> <span class="fl">50</span><span class="op">)</span> <span class="co"># compared to scRNAseq, the amount of PCs are typically 30-50. ndims --&gt; how many PCs to display in elbow plot. For spatial data (Visium): ndims = 50 is a good stating point</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb55"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">brain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html">FindNeighbors</a></span><span class="op">(</span><span class="va">brain</span>, reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span></span>
<span><span class="va">brain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html">FindClusters</a></span><span class="op">(</span><span class="va">brain</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">brain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html">RunUMAP</a></span><span class="op">(</span><span class="va">brain</span>, reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># &gt; brain</span></span>
<span><span class="co"># An object of class Seurat </span></span>
<span><span class="co"># 49131 features across 2698 samples within 2 assays </span></span>
<span><span class="co"># Active assay: SCT (18078 features, 3000 variable features)</span></span>
<span><span class="co">#  3 layers present: counts, data, scale.data</span></span>
<span><span class="co">#  1 other assay present: Spatial</span></span>
<span><span class="co">#  2 dimensional reductions calculated: pca, umap</span></span>
<span><span class="co">#  1 spatial field of view present: slice1</span></span>
<span></span>
<span><span class="va">p5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html">DimPlot</a></span><span class="op">(</span><span class="va">brain</span>, </span>
<span>              reduction <span class="op">=</span> <span class="st">"umap"</span>, </span>
<span>              label <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>              pt.size <span class="op">=</span> <span class="fl">2.5</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"UMAP spots representation"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span></span>
<span>    plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span>, family <span class="op">=</span> <span class="st">"Arial"</span><span class="op">)</span>,</span>
<span>    text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"Arial"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p6</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialDimPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>,</span>
<span>  label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  label.size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  pt.size.factor <span class="op">=</span> <span class="fl">2.5</span>,</span>
<span>  image.alpha <span class="op">=</span> <span class="fl">0</span>  <span class="co"># Make spatial image transparent</span></span>
<span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Spatial spots representation"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span></span>
<span>    plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span>, family <span class="op">=</span> <span class="st">"sans-serif"</span>, face <span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"sans-serif"</span><span class="op">)</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_line</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"grey"</span><span class="op">)</span>,</span>
<span>    legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,   <span class="co"># Increase cluster number font size</span></span>
<span>    legend.key.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="st">"cm"</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_rect</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"white"</span>, color <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span>,  <span class="co"># Plot area</span></span>
<span>    plot.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_rect</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"white"</span>, color <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span>    <span class="co"># Entire background</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">p5</span> <span class="op">+</span> <span class="va">p6</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-6-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</section><section id="summary-table-of-spots-identifiers-and-spot-amount" class="level2" data-number="7"><h2 data-number="7" class="anchored" data-anchor-id="summary-table-of-spots-identifiers-and-spot-amount">
<span class="header-section-number">7</span> <em>Summary table of spots identifiers and spot amount</em>
</h2>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb56"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># view(brain@meta.data)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1] "orig.ident"       "nCount_Spatial"   "nFeature_Spatial" "percent.mt"
[5] "percent.rb"       "nCount_SCT"       "nFeature_SCT"     "SCT_snn_res.0.8"
[9] "seurat_clusters" </code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb58"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">seurat_clusters</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>
  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17
364 279 267 256 244 219 197 189 134 134  92  69  60  57  46  38  31  22 </code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb60"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Install if needed</span></span>
<span><span class="co"># install.packages("flextable")</span></span>
<span></span>
<span><span class="co"># Create frequency table of cluster counts</span></span>
<span><span class="va">cluster_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">seurat_clusters</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert to data frame and calculate percentages</span></span>
<span><span class="va">spot_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">cluster_counts</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>Spot <span class="op">=</span> <span class="va">Var1</span>, `Amount of Spots` <span class="op">=</span> <span class="va">Freq</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span></span>
<span>    `% from total` <span class="op">=</span> <span class="va">`Amount of Spots`</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">`Amount of Spots`</span><span class="op">)</span>,</span>
<span>    `% from total` <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/percent_format.html">percent</a></span><span class="op">(</span><span class="va">`% from total`</span>, accuracy <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Add total row</span></span>
<span><span class="va">total_spots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">spot_table</span><span class="op">$</span><span class="va">`Amount of Spots`</span><span class="op">)</span></span>
<span><span class="va">total_row</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span></span>
<span>  Spot <span class="op">=</span> <span class="st">"Total"</span>,</span>
<span>  `Amount of Spots` <span class="op">=</span> <span class="va">total_spots</span>,</span>
<span>  `% from total` <span class="op">=</span> <span class="st">"100.0%"</span>,</span>
<span>  check.names <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Combine with original data</span></span>
<span><span class="va">spot_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">spot_table</span>, <span class="va">total_row</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create flextable</span></span>
<span><span class="va">ft</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/flextable.html">flextable</a></span><span class="op">(</span><span class="va">spot_table</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/theme_vanilla.html">theme_vanilla</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>  <span class="co"># Apply theme FIRST</span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/set_header_labels.html">set_header_labels</a></span><span class="op">(</span></span>
<span>    Spot <span class="op">=</span> <span class="st">"Cluster identifier"</span>,</span>
<span>    `Amount of Spots` <span class="op">=</span> <span class="st">"Amount of Spots"</span>,</span>
<span>    `% from total` <span class="op">=</span> <span class="st">"% from total"</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/add_header_lines.html">add_header_lines</a></span><span class="op">(</span>values <span class="op">=</span> <span class="st">"Cluster representation and Spot amount from mouse brain section (coronal)*"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/add_footer_lines.html">add_footer_lines</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"* Source: https://www.10xgenomics.com/datasets/mouse-brain-section-coronal-1-standard-1-0-0"</span>, <span class="st">"*Platform: Visium Spatial 10X Genomics."</span>, <span class="st">"*Preservation method: Fresh Frozen"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="co"># Alignment</span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/align.html">align</a></span><span class="op">(</span>align <span class="op">=</span> <span class="st">"center"</span>, part <span class="op">=</span> <span class="st">"header"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>     <span class="co"># Center header (including title line)</span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/align.html">align</a></span><span class="op">(</span>align <span class="op">=</span> <span class="st">"center"</span>, part <span class="op">=</span> <span class="st">"body"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>       <span class="co"># Center all body cells</span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/align.html">align</a></span><span class="op">(</span>align <span class="op">=</span> <span class="st">"left"</span>, part <span class="op">=</span> <span class="st">"footer"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>       <span class="co"># Left-align footnote</span></span>
<span>  <span class="co"># Total row styling</span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/bold.html">bold</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Spot</span> <span class="op">==</span> <span class="st">"Total"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/bg.html">bg</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Spot</span> <span class="op">==</span> <span class="st">"Total"</span>, bg <span class="op">=</span> <span class="st">"#2D2D2D"</span>, part <span class="op">=</span> <span class="st">"body"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>  </span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/color.html">color</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Spot</span> <span class="op">==</span> <span class="st">"Total"</span>, color <span class="op">=</span> <span class="st">"white"</span>, part <span class="op">=</span> <span class="st">"body"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>  </span>
<span>  <span class="co"># Borders</span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/border_inner_h.html">border_inner_h</a></span><span class="op">(</span>border <span class="op">=</span> <span class="fu"><a href="https://davidgohel.github.io/officer/reference/fp_border.html">fp_border</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"#D3D3D3"</span>, width <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/border_outer.html">border_outer</a></span><span class="op">(</span>border <span class="op">=</span> <span class="fu"><a href="https://davidgohel.github.io/officer/reference/fp_border.html">fp_border</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"#D3D3D3"</span>, width <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co"># Font</span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/fontsize.html">fontsize</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">11</span>, part <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/font.html">font</a></span><span class="op">(</span>fontname <span class="op">=</span> <span class="st">"Arial"</span>, part <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/autofit.html">autofit</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">ft</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>a flextable object.
col_keys: `Spot`, `Amount of Spots`, `% from total`
header has 2 row(s)
body has 19 row(s)
original dataset sample:
  Spot Amount of Spots % from total
1    0             364        13.5%
2    1             279        10.3%
3    2             267         9.9%
4    3             256         9.5%
5    4             244         9.0%</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb62"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Convert flextable to a grob</span></span>
<span><span class="va">ft_grob</span> <span class="op">&lt;-</span> <span class="fu">flextable</span><span class="fu">::</span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/gen_grob.html">gen_grob</a></span><span class="op">(</span><span class="va">ft</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a vertical layout for p5 and p6</span></span>
<span><span class="va">left_column</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">p5</span>, <span class="va">p6</span>, ncol <span class="op">=</span> <span class="fl">1</span>, rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Combine left_column and ft_grob horizontally</span></span>
<span><span class="va">final_layout</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span></span>
<span>  <span class="va">left_column</span>, </span>
<span>  <span class="va">ft_grob</span>,</span>
<span>  nrow <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1.2</span><span class="op">)</span>,  <span class="co"># Adjust widths (left plots vs. table)</span></span>
<span>  padding <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="st">"cm"</span><span class="op">)</span>  <span class="co"># Add spacing between elements</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb63"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Save</span></span>
<span><span class="co"># ggsave("~/path/to/your/data/Spatial_transcript_Visium_mouse_brain.png", final_layout, width = 16, height = 8, dpi = 600)</span></span>
<span></span>
<span><span class="co"># view(brain)</span></span>
<span><span class="va">brain</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>An object of class Seurat
49131 features across 2698 samples within 2 assays
Active assay: SCT (18078 features, 3000 variable features)
 3 layers present: counts, data, scale.data
 1 other assay present: Spatial
 2 dimensional reductions calculated: pca, umap
 1 spatial field of view present: slice1</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb65"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># View(brain)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">images</span><span class="op">$</span><span class="va">slice1</span><span class="op">@</span><span class="va">boundaries</span><span class="op">$</span><span class="va">centroids</span><span class="op">@</span><span class="va">coords</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>        x    y
[1,] 7237 8230
[2,] 1611 4170
[3,] 8315 2519
[4,] 2927 7679
[5,] 6280 3138
[6,] 7477 4101</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb67"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">images</span><span class="op">$</span><span class="va">slice1</span><span class="op">@</span><span class="va">boundaries</span><span class="op">$</span><span class="va">centroids</span><span class="op">@</span><span class="va">cells</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1] "AAACAAGTATCTCCCA-1" "AAACAATCTACTAGCA-1" "AAACACCAATAACTGC-1"
[4] "AAACAGAGCGACTCCT-1" "AAACCGGGTAGGTACC-1" "AAACCGTTCGTCCAGG-1"</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb69"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">images</span><span class="op">$</span><span class="va">slice1</span><span class="op">@</span><span class="va">scale.factors</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$spot
[1] 89.44476

$fiducial
[1] 144.4877

$hires
[1] 0.1701114

$lowres
[1] 0.05103343</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb71"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Show only the first 6 genes (rows) and first 6 spots (columns)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">SCT</span><span class="op">@</span><span class="va">data</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>, digits <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>       AAACAAGTATCTCCCA-1 AAACAATCTACTAGCA-1 AAACACCAATAACTGC-1
Xkr4                0.000                0.0                0.0
Sox17               0.000                0.0                0.0
Mrpl15              0.693                1.1                0.0
Lypla1              0.693                1.1                0.0
Tcea1               1.386                1.1                1.1
Rgs20               0.000                0.0                0.0
       AAACAGAGCGACTCCT-1 AAACCGGGTAGGTACC-1 AAACCGTTCGTCCAGG-1
Xkr4                    0              0.000              0.000
Sox17                   0              0.000              0.000
Mrpl15                  0              1.099              1.099
Lypla1                  0              0.000              0.693
Tcea1                   0              0.693              0.693
Rgs20                   0              0.000              0.000</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb73"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># head(brain@assays$SCT@data)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="differential-expression-gene-deg-analysis-across-spatial-clusters" class="level2" data-number="8"><h2 data-number="8" class="anchored" data-anchor-id="differential-expression-gene-deg-analysis-across-spatial-clusters">
<span class="header-section-number">8</span> <em>Differential Expression Gene (DEG) analysis across spatial clusters</em>
</h2>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb74"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/Idents.html">Idents</a></span><span class="op">(</span><span class="va">brain</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"seurat_clusters"</span>  <span class="co"># Use your cluster column from metadata</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">active.ident</span><span class="op">)</span>  <span class="co"># Verify cluster levels (0, 1, 2, ...)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code> [1] "0"  "1"  "2"  "3"  "4"  "5"  "6"  "7"  "8"  "9"  "10" "11" "12" "13" "14"
[16] "15" "16" "17"</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb76"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># # Identify DEGs for all clusters (Wilcoxon test by default)</span></span>
<span><span class="va">all_markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html">FindAllMarkers</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>,</span>
<span>  logfc.threshold <span class="op">=</span> <span class="fl">0.25</span>,       <span class="co"># Minimum log2 fold change</span></span>
<span>  min.pct <span class="op">=</span> <span class="fl">0.1</span>,                <span class="co"># Gene detected in at least 10% of cells in either cluster</span></span>
<span>  only.pos <span class="op">=</span> <span class="cn">TRUE</span>,              <span class="co"># Report only positive markers</span></span>
<span>  test.use <span class="op">=</span> <span class="st">"wilcox"</span>,          <span class="co"># Default test (alternatives: "MAST", "DESeq2", etc.)</span></span>
<span>  return.thresh <span class="op">=</span> <span class="fl">0.01</span>          <span class="co"># Return genes with adjusted p-value &lt; 0.01</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View top 5 markers per cluster</span></span>
<span><span class="va">top_markers</span> <span class="op">&lt;-</span> <span class="va">all_markers</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice_max</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">5</span>, order_by <span class="op">=</span> <span class="va">avg_log2FC</span><span class="op">)</span></span>
<span><span class="co"># View(top_markers) # View cannot be rendered in HTML so comment out</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">brain</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>                      orig.ident nCount_Spatial nFeature_Spatial percent.mt
AAACAAGTATCTCCCA-1 SeuratProject          19340             4928   19.94312
AAACAATCTACTAGCA-1 SeuratProject          13750             3448   23.76000
AAACACCAATAACTGC-1 SeuratProject          32710             6022   15.01070
AAACAGAGCGACTCCT-1 SeuratProject          15909             4311   20.55440
AAACCGGGTAGGTACC-1 SeuratProject          31856             5787   21.01017
AAACCGTTCGTCCAGG-1 SeuratProject          30862             5898   16.90104
AAACCTCATGAAGTTG-1 SeuratProject          60319             8331   11.55357
AAACGAAGAACATACC-1 SeuratProject          39264             6921   17.85860
AAACGAGACGGTTGAT-1 SeuratProject          34853             6696   12.15677
AAACGGTTGCGAACTG-1 SeuratProject          28395             6115   15.91830
                   percent.rb nCount_SCT nFeature_SCT SCT_snn_res.0.8
AAACAAGTATCTCCCA-1   3.536711      26168         4915              11
AAACAATCTACTAGCA-1   6.487273      27017         3881               1
AAACACCAATAACTGC-1  10.189544      28141         5967               7
AAACAGAGCGACTCCT-1   5.544032      26397         4416               6
AAACCGGGTAGGTACC-1   7.380085      28105         5761               7
AAACCGTTCGTCCAGG-1   7.352083      28122         5859               9
AAACCTCATGAAGTTG-1   8.889405      28268         6572               7
AAACGAAGAACATACC-1   5.475754      28178         6766               6
AAACGAGACGGTTGAT-1   7.875936      28342         6623               5
AAACGGTTGCGAACTG-1   8.505018      27652         6048              12
                   seurat_clusters
AAACAAGTATCTCCCA-1              11
AAACAATCTACTAGCA-1               1
AAACACCAATAACTGC-1               7
AAACAGAGCGACTCCT-1               6
AAACCGGGTAGGTACC-1               7
AAACCGTTCGTCCAGG-1               9
AAACCTCATGAAGTTG-1               7
AAACGAAGAACATACC-1               6
AAACGAGACGGTTGAT-1               5
AAACGGTTGCGAACTG-1              12</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb78"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Comparing cluster 0 vs cluster 1</span></span>
<span><span class="co"># cluster0_vs_1 &lt;- FindMarkers(</span></span>
<span><span class="co">#   brain,</span></span>
<span><span class="co">#   ident.1 = 0,</span></span>
<span><span class="co">#   ident.2 = 1,</span></span>
<span><span class="co">#   logfc.threshold = 0.25,</span></span>
<span><span class="co">#   min.pct = 0.1</span></span>
<span><span class="co"># )</span></span>
<span></span>
<span><span class="co"># Interpreting results</span></span>
<span><span class="co"># Filter significant markers (adjust p-value cutoff)</span></span>
<span><span class="va">significant_markers</span> <span class="op">&lt;-</span> <span class="va">all_markers</span><span class="op">[</span><span class="va">all_markers</span><span class="op">$</span><span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.01</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Plot heatmap of top markers</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DoHeatmap.html">DoHeatmap</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>,</span>
<span>  features <span class="op">=</span> <span class="va">top_markers</span><span class="op">$</span><span class="va">gene</span>,</span>
<span>  group.colors <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/pal_hue.html">hue_pal</a></span><span class="op">(</span><span class="op">)</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">brain</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,  <span class="co"># Cluster colors</span></span>
<span>  size <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb79"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize Marker Expression</span></span>
<span><span class="co"># Feature plot for a top gene (e.g., "Gfap")</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html">FeaturePlot</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>,</span>
<span>  features <span class="op">=</span> <span class="st">"Tnnt1"</span>,           <span class="co"># Replace with your gene of interest</span></span>
<span>  reduction <span class="op">=</span> <span class="st">"umap"</span>,          <span class="co"># Or "spatial" for spatial visualization</span></span>
<span>  pt.size <span class="op">=</span> <span class="fl">1.5</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-8-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb80"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Violin plot</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/VlnPlot.html">VlnPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>,</span>
<span>  features <span class="op">=</span> <span class="st">"Tnnt1"</span>,</span>
<span>  pt.size <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  ncol <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-8-3.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb81"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Save DEG results</span></span>
<span><span class="co"># write.csv(all_markers, "~/path/to/your/data/Spatial_Data_MouseBrainCoronal/Spatial_mouse_brain_DEG_all_clusters.csv")</span></span>
<span></span>
<span><span class="co"># Save Seurat object with markers</span></span>
<span><span class="co"># saveRDS(brain, file = "brain_with_markers.rds")</span></span>
<span></span>
<span><span class="co"># Spatial Feature Plots</span></span>
<span><span class="co"># Example: Plot expression of gene "Tnnt1" on spatial coordinates, highly expressed in spot #3</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialFeaturePlot</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>,</span>
<span>  features <span class="op">=</span> <span class="st">"Tnnt1"</span>,          <span class="co"># Replace with your gene of interest</span></span>
<span>  pt.size.factor <span class="op">=</span> <span class="fl">2.5</span>,        <span class="co"># Adjust spot size (matching your SpatialDimPlot)</span></span>
<span>  alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">2</span><span class="op">)</span>,           <span class="co"># Adjust transparency (min and max expression)</span></span>
<span>  image.alpha <span class="op">=</span> <span class="fl">0</span>,             <span class="co"># Hide the tissue image (optional)</span></span>
<span>  ncol <span class="op">=</span> <span class="fl">1</span>                     <span class="co"># Number of columns (if plotting multiple genes)</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Spatial expression of Tnnt1, highly expressed in cluster 3"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span></span>
<span>    plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span>, family <span class="op">=</span> <span class="st">"sans-serif"</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_fill_viridis_c</a></span><span class="op">(</span>option <span class="op">=</span> <span class="st">"magma"</span><span class="op">)</span>  <span class="co"># Use viridis/magma for better contrast</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-8-4.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb82"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot two genes in a single row</span></span>
<span><span class="va">cluster3_cluster1_highestDEG</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialFeaturePlot</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Tnnt1"</span>, <span class="st">"Lypd1"</span><span class="op">)</span>,  <span class="co"># Replace with your genes of interest</span></span>
<span>  pt.size.factor <span class="op">=</span> <span class="fl">2.5</span>,           <span class="co"># Match your SpatialDimPlot settings</span></span>
<span>  alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">3</span><span class="op">)</span>,              <span class="co"># Transparency range (0.1 for low, 1 for high)</span></span>
<span>  image.alpha <span class="op">=</span> <span class="fl">0</span>,                <span class="co"># Hide tissue image (optional)</span></span>
<span>  ncol <span class="op">=</span> <span class="fl">2</span>                        <span class="co"># Arrange plots in 2 columns</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_annotation.html">plot_annotation</a></span><span class="op">(</span></span>
<span>    title <span class="op">=</span> <span class="st">"Spatial Expression of Tnnt1 (cluster 3, left) and Lypd1 (cluster 1, right)"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span></span>
<span>    plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span></span>
<span>      size <span class="op">=</span> <span class="fl">30</span>,                   <span class="co"># Larger font size</span></span>
<span>      family <span class="op">=</span> <span class="st">"sans-serif"</span>,       <span class="co"># Font family</span></span>
<span>      face <span class="op">=</span> <span class="st">"bold"</span>,               <span class="co"># Bold text</span></span>
<span>      hjust <span class="op">=</span> <span class="fl">0.5</span>,                <span class="co"># Center-align title</span></span>
<span>      margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span>t <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>     <span class="co"># Add top margin to prevent cutoff</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_fill_viridis_c</a></span><span class="op">(</span>option <span class="op">=</span> <span class="st">"magma"</span><span class="op">)</span>  <span class="co"># Consistent color scale</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">cluster3_cluster1_highestDEG</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-8-5.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb83"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Save</span></span>
<span><span class="co"># ggsave("~/path/to/your/data/Spatial_highest_DEG_Visium_mouse_brain.png", cluster3_cluster1_highestDEG, width = 16, height = 8, dpi = 600)</span></span>
<span></span>
<span><span class="co"># Display a safe summary of the Seurat object</span></span>
<span><span class="co"># str(brain, max.level = 2)  # Show top-level structure only</span></span>
<span></span>
<span><span class="co">###########</span></span>
<span><span class="co">###########</span></span>
<span></span>
<span><span class="co"># Visualize Spatial Expression according to SVG</span></span>
<span><span class="co"># If these genes show clear gradients or regional enrichment</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialFeaturePlot</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Sparc"</span>, <span class="st">"Camk2n1"</span>, <span class="st">"Pmch"</span>, <span class="st">"Nrgn"</span><span class="op">)</span>,</span>
<span>  pt.size.factor <span class="op">=</span> <span class="fl">2.5</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>  image.alpha <span class="op">=</span> <span class="fl">0.25</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-8-6.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb84"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># GO Enrichment analysis</span></span>
<span><span class="va">enrich_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html">enrichGO</a></span><span class="op">(</span></span>
<span>  gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">top_spatial_genes</span><span class="op">)</span>,</span>
<span>  OrgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>,     <span class="co"># org.Hs.eg.db (human)</span></span>
<span>  keyType <span class="op">=</span> <span class="st">"SYMBOL"</span>,       <span class="co"># or "ENSEMBL" if genes are Ensembl IDs</span></span>
<span>  ont <span class="op">=</span> <span class="st">"BP"</span>               <span class="co"># Biological Process ontology</span></span>
<span><span class="op">)</span></span>
<span><span class="co"># Visualize results</span></span>
<span><span class="va">go_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="va">enrich_result</span>, showCategory <span class="op">=</span> <span class="fl">10</span>,</span>
<span>        orderBy <span class="op">=</span> <span class="st">"GeneRatio"</span>, title <span class="op">=</span> <span class="st">"Top Enriched GO Terms (Biological Process)"</span><span class="op">)</span> </span>
<span><span class="co"># top 10 most significantly enriched GO terms based on adjusted p-value or other ranking metrics.</span></span>
<span><span class="co"># Gene Ratio: 0.15 means p = 1e-10</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">go_plot</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-8-7.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb85"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># ggsave(</span></span>
<span><span class="co">#   filename = "~/path/to/your/data/Spatial_brain_mouse_go_enrichment_highres.png",</span></span>
<span><span class="co">#   plot = go_plot,</span></span>
<span><span class="co">#   width = 12,                      # Wider for better label readability</span></span>
<span><span class="co">#   height = 8,</span></span>
<span><span class="co">#   dpi = 600,                       # Higher resolution for publications</span></span>
<span><span class="co">#   bg = "white"                     # Set background to white</span></span>
<span><span class="co"># )</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="brain-region-annotation-summary-table" class="level2" data-number="9"><h2 data-number="9" class="anchored" data-anchor-id="brain-region-annotation-summary-table">
<span class="header-section-number">9</span> <em>Brain region annotation / Summary table</em>
</h2>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb86"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Install if not installed</span></span>
<span><span class="co"># BiocManager::install("ABAData") --&gt; DEPRECATED</span></span>
<span><span class="co"># BiocManager::install("ABAbilaR") --&gt; DEPRECATED</span></span>
<span><span class="co"># BiocManager::install("SingleR")</span></span>
<span><span class="co"># BiocManager::install("spatialLIBD")</span></span>
<span><span class="co"># remotes::install_github("benfulcher/allen_clr") --&gt; DEPRECATED</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LTLA/SingleR">SingleR</a></span><span class="op">)</span></span>
<span><span class="co"># library(ABAData)  # Allen Brain Atlas --&gt; DEPRECATED</span></span>
<span><span class="co"># library(ABAbilaR) # Interface with Allen Brain Atlas --&gt; DEPRECATED</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LieberInstitute/spatialLIBD">spatialLIBD</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Assign broad brain regions manually based on spatial clustering</span></span>
<span><span class="va">brain</span><span class="op">$</span><span class="va">brain_region</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html">case_when</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span><span class="op">$</span><span class="va">seurat_clusters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">8</span>, <span class="fl">6</span><span class="op">)</span> <span class="op">~</span> <span class="st">"Neocortex"</span>, <span class="co">##</span></span>
<span>  <span class="va">brain</span><span class="op">$</span><span class="va">seurat_clusters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">16</span>, <span class="fl">4</span>, <span class="fl">17</span><span class="op">)</span> <span class="op">~</span> <span class="st">"Hippocampal region"</span>,</span>
<span>  <span class="va">brain</span><span class="op">$</span><span class="va">seurat_clusters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span> <span class="op">~</span> <span class="st">"Thalamus"</span>, <span class="co">##</span></span>
<span>  <span class="va">brain</span><span class="op">$</span><span class="va">seurat_clusters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span> <span class="op">~</span> <span class="st">"Hypothalamus"</span>, <span class="co">##</span></span>
<span>  <span class="va">brain</span><span class="op">$</span><span class="va">seurat_clusters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">13</span>, <span class="fl">14</span><span class="op">)</span> <span class="op">~</span> <span class="st">"Caudatoputamen"</span>,</span>
<span>  <span class="va">brain</span><span class="op">$</span><span class="va">seurat_clusters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">11</span><span class="op">)</span> <span class="op">~</span> <span class="st">"Allocortex"</span>,</span>
<span>  <span class="va">brain</span><span class="op">$</span><span class="va">seurat_clusters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">15</span><span class="op">)</span> <span class="op">~</span> <span class="st">"Striatum-like amygdalar nuclei"</span>, <span class="co">##</span></span>
<span>  <span class="va">brain</span><span class="op">$</span><span class="va">seurat_clusters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">9</span>, <span class="fl">10</span>, <span class="fl">5</span>, <span class="fl">7</span><span class="op">)</span> <span class="op">~</span> <span class="st">"Fiber tracts"</span>, <span class="co">##</span></span>
<span>  <span class="cn">TRUE</span> <span class="op">~</span> <span class="st">"N/A"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">brain</span><span class="op">$</span><span class="va">brain_region</span><span class="op">)</span>  <span class="co"># Count spots per region</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>
                    Allocortex                 Caudatoputamen
                           348                            103
                  Fiber tracts             Hippocampal region
                           634                            297
                  Hypothalamus                            N/A
                           267                             60
                     Neocortex Striatum-like amygdalar nuclei
                           695                             38
                      Thalamus
                           256 </code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb88"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># head(brain@meta.data)  # View metadata</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code> [1] "orig.ident"       "nCount_Spatial"   "nFeature_Spatial" "percent.mt"
 [5] "percent.rb"       "nCount_SCT"       "nFeature_SCT"     "SCT_snn_res.0.8"
 [9] "seurat_clusters"  "brain_region"    </code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb90"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p7</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html">DimPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>, </span>
<span>  group.by <span class="op">=</span> <span class="st">"brain_region"</span>, </span>
<span>  label <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>  repel <span class="op">=</span> <span class="cn">TRUE</span>,            <span class="co"># Automatically separates labels</span></span>
<span>  pt.size <span class="op">=</span> <span class="fl">2.5</span>,           <span class="co"># Size of points on UMAP</span></span>
<span>  label.size <span class="op">=</span> <span class="fl">5</span>,          <span class="co"># Font size of cluster labels</span></span>
<span>  label.box <span class="op">=</span> <span class="cn">TRUE</span>,        <span class="co"># Adds background box to labels</span></span>
<span>  alpha <span class="op">=</span> <span class="fl">0.8</span></span>
<span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"UMAP Brain Region Annotation"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html">NoAxes</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span></span>
<span>    <span class="co"># Use `color` instead of `fill` for DimPlot legends</span></span>
<span>    color <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html">guide_legend</a></span><span class="op">(</span></span>
<span>      title <span class="op">=</span> <span class="st">"Brain Region"</span>,</span>
<span>      override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span></span>
<span>        size <span class="op">=</span> <span class="fl">6</span>,          <span class="co"># Size of color dots in legend</span></span>
<span>        alpha <span class="op">=</span> <span class="fl">1</span>          <span class="co"># Ensure full opacity for legend keys</span></span>
<span>      <span class="op">)</span>,</span>
<span>      ncol <span class="op">=</span> <span class="fl">1</span>             <span class="co"># Single column for vertical legend</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span></span>
<span>    plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span>, family <span class="op">=</span> <span class="st">"sans-serif"</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"sans-serif"</span><span class="op">)</span>,</span>
<span>    legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,  <span class="co"># Legend title font size</span></span>
<span>    legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,   <span class="co"># Legend text font size</span></span>
<span>    legend.key.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">1</span>, <span class="st">"cm"</span><span class="op">)</span>,         <span class="co"># Size of legend key boxes</span></span>
<span>    legend.spacing.y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="st">"cm"</span><span class="op">)</span>       <span class="co"># Vertical spacing between legend items</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">p8</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialDimPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>,</span>
<span>  group.by <span class="op">=</span> <span class="st">"brain_region"</span>,</span>
<span>  label <span class="op">=</span> <span class="cn">TRUE</span>,    <span class="co"># Labels on (TRUE) or off (FALSE) on the brain regions</span></span>
<span>  repel <span class="op">=</span> <span class="cn">TRUE</span>,    <span class="co"># Prevents overlap of labels</span></span>
<span>  label.size <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  pt.size.factor <span class="op">=</span> <span class="fl">3.5</span>,</span>
<span>  image.alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">0.8</span></span>
<span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Spatial Brain Region Annotation"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># Customize legend title and key size</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span></span>
<span>    fill <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html">guide_legend</a></span><span class="op">(</span></span>
<span>      title <span class="op">=</span> <span class="st">"Brain Region"</span>,  <span class="co"># New legend title</span></span>
<span>      override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span>  <span class="co"># Increase color circle size</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span></span>
<span>    plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span>, family <span class="op">=</span> <span class="st">"sans-serif"</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"sans-serif"</span><span class="op">)</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_line</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"grey"</span><span class="op">)</span>,</span>
<span>    <span class="co"># legend.position = "none",  # Remove legend from the right side</span></span>
<span>    legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,  <span class="co"># Legend title font size</span></span>
<span>    legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span>,    <span class="co"># Legend text font size</span></span>
<span>    legend.key.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">1</span>, <span class="st">"cm"</span><span class="op">)</span>,        <span class="co"># Overall legend key box size</span></span>
<span>    panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_rect</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"white"</span>, color <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span>, <span class="co"># Plot area</span></span>
<span>    plot.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_rect</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"white"</span>, color <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span>   <span class="co"># Entire background</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">p9</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialDimPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>,</span>
<span>  pt.size.factor <span class="op">=</span> <span class="fl">3.5</span>,</span>
<span>  image.alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">0.0</span></span>
<span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"H&amp;E Brain Region Coronal"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># Semi-transparent white overlay</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotate.html">annotate</a></span><span class="op">(</span></span>
<span>    <span class="st">"rect"</span>, </span>
<span>    xmin <span class="op">=</span> <span class="op">-</span><span class="cn">Inf</span>, xmax <span class="op">=</span> <span class="cn">Inf</span>, </span>
<span>    ymin <span class="op">=</span> <span class="op">-</span><span class="cn">Inf</span>, ymax <span class="op">=</span> <span class="cn">Inf</span>, </span>
<span>    fill <span class="op">=</span> <span class="st">"white"</span>, alpha <span class="op">=</span> <span class="fl">0.2</span>  <span class="co"># Adjust alpha (0 = no overlay, 1 = fully white)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span></span>
<span>    plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span>, family <span class="op">=</span> <span class="st">"sans-serif"</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"none"</span>,          <span class="co"># Remove any residual legend</span></span>
<span>    axis.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,      <span class="co"># Remove x/y axis titles</span></span>
<span>    axis.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,       <span class="co"># Remove x/y axis text</span></span>
<span>    axis.ticks <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,      <span class="co"># Remove axis ticks</span></span>
<span>    panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,      <span class="co"># Remove grid lines</span></span>
<span>    panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_rect</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"white"</span>, color <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,</span>
<span>    plot.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_rect</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"white"</span>, color <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">patch1</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">p9</span> <span class="op">|</span> <span class="va">p8</span><span class="op">)</span> <span class="op">/</span> <span class="va">p7</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">patch1</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb91"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Save 'H&amp;E (p9)'</span></span>
<span><span class="co"># ggsave("~/path/to/your/data/Spatial_Visium_mouse_brain_H&amp;E.png", p9, width = 16, height = 8, dpi = 600)</span></span>
<span></span>
<span><span class="co"># Save 'UMAP_regions (p7)'</span></span>
<span><span class="co"># ggsave("~/path/to/your/data/Spatial_Visium_mouse_brain_UMAP_regions.png", p7, width = 16, height = 8, dpi = 600)</span></span>
<span></span>
<span><span class="co"># Save 'patch1'</span></span>
<span><span class="co"># ggsave("~/path/to/your/data/Spatial_Visium_mouse_brain_regions_patch.png", patch1, width = 16, height = 8, dpi = 600)</span></span>
<span></span>
<span><span class="co"># Save 'Brain_regions (p8)'</span></span>
<span><span class="co"># ggsave("~/path/to/your/data/Spatial_Visium_mouse_brain_regions.png", p8, width = 16, height = 8, dpi = 600)</span></span>
<span></span>
<span><span class="co"># Save 'Brain_regions (p8)'</span></span>
<span><span class="co"># ggsave("~path/to/your/data/Spatial_Visium_mouse_brain_regionsNoLegends.png", p8, width = 16, height = 8, dpi = 600)</span></span>
<span></span>
<span><span class="co"># Define region-specific markers (expand with more genes)</span></span>
<span><span class="va">region_markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span></span>
<span>  Neocortex <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Slc17a7"</span>, <span class="st">"Rorb"</span>, <span class="st">"Foxp2"</span>, <span class="st">"Cux1"</span>,      <span class="co"># Excitatory neurons</span></span>
<span>                <span class="st">"Gad1"</span>,<span class="st">"Gad2"</span>, <span class="st">"Pvalb"</span>, <span class="st">"Sst"</span>, <span class="st">"Vip"</span>,    <span class="co"># Inhibitory neurons</span></span>
<span>                <span class="st">"Gfap"</span>, <span class="st">"Aqp4"</span>, <span class="st">"Aldh1l1"</span>, <span class="st">"Ptprc"</span><span class="op">)</span>,     <span class="co"># Glial cells</span></span>
<span>  Hippocampus <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Wfs1"</span>, <span class="st">"Nr3c2"</span>, <span class="st">"Rgs14"</span>,              <span class="co"># CA1</span></span>
<span>                  <span class="st">"Slc17a7"</span>, <span class="st">"Grik1"</span>, <span class="st">"Prkcd"</span>,           <span class="co"># CA3</span></span>
<span>                  <span class="st">"Prox1"</span>, <span class="st">"C1ql2"</span>, <span class="st">"Neurod1"</span>, <span class="st">"Hpca"</span><span class="op">)</span>,  <span class="co"># Dentate Gyrus (granule cells)</span></span>
<span>  Thalamus <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Gad2"</span>, <span class="st">"Npy"</span>, <span class="st">"Slc32a1"</span>,</span>
<span>               <span class="st">"Calb1"</span>, <span class="st">"Vglut2"</span>,<span class="st">"Foxp2"</span>, <span class="st">"Tbr1"</span><span class="op">)</span>,  <span class="co"># Excitatory thalamic neurons, Sensory relay neurons, Thalamocortical projection neurons</span></span>
<span>  Hypothalamus <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Oxt"</span>, <span class="st">"Avp"</span>,<span class="st">"Cartpt"</span> ,</span>
<span>                   <span class="st">"Pomc"</span>, <span class="st">"Agrp"</span>, <span class="st">"Npy"</span>, <span class="st">"Hcrt"</span>, <span class="st">"Crh"</span><span class="op">)</span>,          <span class="co"># Arcuate nucleus</span></span>
<span>  </span>
<span>  Caudoputamen_Striatum <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Cplx3"</span>,<span class="st">"Drd1"</span>, <span class="st">"Drd2"</span>, <span class="st">"Ppp1r1b"</span><span class="op">)</span>,           <span class="co"># Dopamine receptors</span></span>
<span>  Allocortex <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Rasgrf2"</span>,<span class="st">"Calb2"</span>,<span class="st">"Pcp4"</span><span class="op">)</span>,</span>
<span>  Striatum_like_Amygdalar_Nuclei <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Cplx3"</span>,<span class="st">"Tac2"</span>,<span class="st">"Drd1"</span>, <span class="st">"Drd2"</span>,</span>
<span>                                     <span class="st">"Pvalb"</span>, <span class="st">"Npy"</span><span class="op">)</span>,</span>
<span>  FiberTracts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Mbp"</span>, <span class="st">"Plp1"</span>, <span class="st">"Mog"</span>, <span class="st">"Olig2"</span>, <span class="st">"Sox10"</span>, <span class="st">"Cldn11"</span><span class="op">)</span>              <span class="co"># Oligodendrocytes</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Neocortex region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># Excitatory neurons: Slc17a7, Rorb, Foxp2, Cux1</span></span>
<span><span class="co"># Inhibitory neurons: Gad1, Gad2, Pvalb, Sst, Vip</span></span>
<span><span class="co"># Glial cells: Gfap, Aqp4, Aldh1l1</span></span>
<span></span>
<span><span class="co"># Hippocampal region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># CA1 pyramidal neurons: Wfs1, Nr3c2, Rgs14</span></span>
<span><span class="co"># CA3 pyramidal neurons: Slc17a7, Grik1, Prkcd</span></span>
<span><span class="co"># Dentate gyrus (granule cells): Prox1, C1ql2, Neurod1</span></span>
<span></span>
<span><span class="co"># Thalamus region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># Calb1 (Calbindin)</span></span>
<span><span class="co"># Vglut2 (Slc17a6) – Excitatory thalamic neurons</span></span>
<span><span class="co"># Foxp2 – Sensory relay neurons</span></span>
<span><span class="co"># Tbr1 – Thalamocortical projection neurons</span></span>
<span></span>
<span><span class="co"># Hypothalamus region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># Oxt (Oxytocin)</span></span>
<span><span class="co"># Avp (Vasopressin)</span></span>
<span><span class="co"># Cartpt (Cocaine- and amphetamine-regulated transcript)</span></span>
<span><span class="co"># Hypothalamic nuclei: Pomc, Npy, Hcrt (Orexin), Crh (Corticotropin-releasing hormone)</span></span>
<span></span>
<span><span class="co"># Caudatoputamen (Striatum) region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># Drd1 (D1 dopamine receptor, direct pathway)</span></span>
<span><span class="co"># Drd2 (D2 dopamine receptor, indirect pathway)</span></span>
<span><span class="co"># Ppp1r1b (DARPP-32, marker of medium spiny neurons)</span></span>
<span><span class="co"># Foxp2 (Regulates motor function in striatal neurons)</span></span>
<span></span>
<span><span class="co"># Allocortex region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># Rasgrf2 (Piriform cortex)</span></span>
<span><span class="co"># Calb2 (Layer-specific cortical neurons)</span></span>
<span><span class="co"># Pcp4 (Parvalbumin interneurons)</span></span>
<span></span>
<span><span class="co"># Striatum-like Amygdalar Nuclei region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># Cplx3, Tac2 (Central amygdala)</span></span>
<span><span class="co"># Foxp2 (Basolateral amygdala)</span></span>
<span><span class="co"># Drd1, Drd2 (Dopamine receptor expression in striatum-like regions)</span></span>
<span><span class="co"># Pvalb, Npy (Interneurons in the amygdala)</span></span>
<span></span>
<span><span class="co"># Fiber Tracts region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># Mbp (Myelin basic protein, oligodendrocytes)</span></span>
<span><span class="co"># Plp1, Cnp (Myelination markers)</span></span>
<span><span class="co"># Olig2, Sox10 (Oligodendrocyte precursor cells)</span></span>
<span><span class="co"># Cldn11 (Tight junctions in myelin sheaths)</span></span>
<span></span>
<span><span class="co"># Dot plot to validate markers</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DotPlot.html">DotPlot</a></span><span class="op">(</span><span class="va">brain</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">region_markers</span><span class="op">)</span><span class="op">)</span>, group.by <span class="op">=</span> <span class="st">"brain_region"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html">RotatedAxis</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_color_viridis_c</a></span><span class="op">(</span>option <span class="op">=</span> <span class="st">"magma"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-9-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb92"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># The current mapping has potential overlaps (e.g., cluster 11 in both Neocortex and Allocortex). Investigate ambiguous clusters:</span></span>
<span><span class="co"># Subset ambiguous clusters (e.g., cluster 11)</span></span>
<span><span class="va">cluster11</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">brain</span>, <span class="va">seurat_clusters</span> <span class="op">==</span> <span class="fl">11</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Find unique markers for cluster 11</span></span>
<span><span class="va">cluster11_markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindMarkers.html">FindMarkers</a></span><span class="op">(</span><span class="va">brain</span>, ident.1 <span class="op">=</span> <span class="fl">11</span>, logfc.threshold <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">cluster11_markers</span>, <span class="fl">10</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>                   p_val avg_log2FC pct.1 pct.2    p_val_adj
Trim54     1.414609e-100   5.153370 0.565 0.032 2.557331e-96
Igfn1       6.830745e-80   3.992798 0.681 0.065 1.234862e-75
Pcsk9       8.900274e-69   4.361957 0.406 0.024 1.608992e-64
Ndst4       2.113222e-64   3.517756 0.754 0.105 3.820283e-60
Ucn         2.245522e-49   6.573702 0.116 0.001 4.059455e-45
Slc30a3     2.348887e-43   2.962096 0.986 0.504 4.246318e-39
Ntf3        7.360843e-43   3.328010 0.406 0.041 1.330693e-38
Filip1      1.707933e-42   3.093626 0.667 0.122 3.087600e-38
AC154509.1  1.734661e-42   3.505531 0.391 0.038 3.135920e-38
Nwd2        3.501730e-42   2.531574 0.942 0.325 6.330427e-38</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb94"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Spatial plot of top marker</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">brain</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">cluster11_markers</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, pt.size.factor <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-9-3.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb95"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Addressing NA region</span></span>
<span><span class="co"># Subset "N/A" clusters</span></span>
<span><span class="va">na_clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">brain</span>, <span class="va">brain_region</span> <span class="op">==</span> <span class="st">"N/A"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Find markers for "N/A" clusters</span></span>
<span><span class="va">na_markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html">FindAllMarkers</a></span><span class="op">(</span><span class="va">na_clusters</span>, only.pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">na_markers</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>data frame with 0 columns and 0 rows</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb97"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Check spatial distribution</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialDimPlot</a></span><span class="op">(</span><span class="va">na_clusters</span>, pt.size.factor <span class="op">=</span> <span class="fl">3</span>, image.alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-9-4.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb98"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Test if cluster markers are enriched in specific brain regions</span></span>
<span><span class="va">cluster0_markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindMarkers.html">FindMarkers</a></span><span class="op">(</span><span class="va">brain</span>, ident.1 <span class="op">=</span> <span class="fl">0</span>, only.pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">top_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">cluster0_markers</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span>  <span class="co"># Top 50 markers</span></span>
<span></span>
<span></span>
<span><span class="co"># Quality Control of Annotation</span></span>
<span><span class="co"># 1️⃣ Check if any region is overrepresented or missing</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/proportions.html">prop.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">brain</span><span class="op">$</span><span class="va">brain_region</span><span class="op">)</span><span class="op">)</span> <span class="op">*</span> <span class="fl">100</span>  <span class="co"># Get percentage of spots per region</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>
                    Allocortex                 Caudatoputamen
                     12.898443                       3.817643
                  Fiber tracts             Hippocampal region
                     23.498888                      11.008154
                  Hypothalamus                            N/A
                      9.896219                       2.223870
                     Neocortex Striatum-like amygdalar nuclei
                     25.759822                       1.408451
                      Thalamus
                      9.488510 </code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb100"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># 2️⃣ Check expression of key marker genes per region</span></span>
<span></span>
<span><span class="co"># Neocortex region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># Excitatory neurons: Slc17a7, Rorb, Foxp2, Cux1</span></span>
<span><span class="co"># Inhibitory neurons: Gad1, Gad2, Pvalb, Sst, Vip</span></span>
<span><span class="co"># Glial cells: Gfap, Aqp4, Aldh1l1</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html">FeaturePlot</a></span><span class="op">(</span><span class="va">brain</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Pvalb"</span>, <span class="st">"Gfap"</span>, <span class="st">"Aldh1l1"</span><span class="op">)</span>, reduction <span class="op">=</span> <span class="st">"umap"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-9-5.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb101"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Hippocampal region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># CA1 pyramidal neurons: Wfs1, Nr3c2, Rgs14</span></span>
<span><span class="co"># CA3 pyramidal neurons: Slc17a7, Grik1, Prkcd</span></span>
<span><span class="co"># Dentate gyrus (granule cells): Prox1, C1ql2, Neurod1</span></span>
<span><span class="co"># FeaturePlot(brain, features = c("Foxp2", "Gad1", "Neurod1"), reduction = "umap")</span></span>
<span></span>
<span><span class="co"># Thalamus region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># Calb1 (Calbindin)</span></span>
<span><span class="co"># Vglut2 (Slc17a6) – Excitatory thalamic neurons</span></span>
<span><span class="co"># Foxp2 – Sensory relay neurons</span></span>
<span><span class="co"># Tbr1 – Thalamocortical projection neurons</span></span>
<span><span class="co"># FeaturePlot(brain, features = c("Tbr1", "Slc17a6", "Calb1"), reduction = "umap")</span></span>
<span></span>
<span><span class="co"># Hypothalamus region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># Oxt (Oxytocin)</span></span>
<span><span class="co"># Avp (Vasopressin)</span></span>
<span><span class="co"># Cartpt (Cocaine- and amphetamine-regulated transcript)</span></span>
<span><span class="co"># Hypothalamic nuclei: Pomc, Npy, Hcrt (Orexin), Crh (Corticotropin-releasing hormone)</span></span>
<span><span class="co"># FeaturePlot(brain, features = c("Tbr1", "Slc17a6", "Calb1"), reduction = "umap")</span></span>
<span></span>
<span><span class="co"># Caudatoputamen (Striatum) region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># Drd1 (D1 dopamine receptor, direct pathway)</span></span>
<span><span class="co"># Drd2 (D2 dopamine receptor, indirect pathway)</span></span>
<span><span class="co"># Ppp1r1b (DARPP-32, marker of medium spiny neurons)</span></span>
<span><span class="co"># Foxp2 (Regulates motor function in striatal neurons)</span></span>
<span><span class="co"># FeaturePlot(brain, features = c("Tbr1", "Slc17a6", "Calb1"), reduction = "umap")</span></span>
<span></span>
<span><span class="co"># Allocortex region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># Rasgrf2 (Piriform cortex)</span></span>
<span><span class="co"># Calb2 (Layer-specific cortical neurons)</span></span>
<span><span class="co"># Pcp4 (Parvalbumin interneurons)</span></span>
<span><span class="co"># FeaturePlot(brain, features = c("Tbr1", "Slc17a6", "Calb1"), reduction = "umap")</span></span>
<span></span>
<span><span class="co"># Striatum-like Amygdalar Nuclei region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># Cplx3, Tac2 (Central amygdala)</span></span>
<span><span class="co"># Foxp2 (Basolateral amygdala)</span></span>
<span><span class="co"># Drd1, Drd2 (Dopamine receptor expression in striatum-like regions)</span></span>
<span><span class="co"># Pvalb, Npy (Interneurons in the amygdala)</span></span>
<span><span class="co"># FeaturePlot(brain, features = c("Tbr1", "Slc17a6", "Calb1"), reduction = "umap")</span></span>
<span></span>
<span><span class="co"># Fiber Tracts region:</span></span>
<span><span class="co"># Marker Genes:</span></span>
<span><span class="co"># Mbp (Myelin basic protein, oligodendrocytes)</span></span>
<span><span class="co"># Plp1, Cnp (Myelination markers)</span></span>
<span><span class="co"># Olig2, Sox10 (Oligodendrocyte precursor cells)</span></span>
<span><span class="co"># Cldn11 (Tight junctions in myelin sheaths)</span></span>
<span><span class="co"># FeaturePlot(brain, features = c("Tbr1", "Slc17a6", "Calb1"), reduction = "umap")</span></span>
<span></span>
<span><span class="co">########## Flextable - Brain regions ########## </span></span>
<span></span>
<span><span class="co"># Create frequency table of region counts</span></span>
<span><span class="va">region_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">brain</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">brain_region</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert to data frame and calculate percentages</span></span>
<span><span class="va">region_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">region_counts</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>Region <span class="op">=</span> <span class="va">Var1</span>, `Amount of Spots` <span class="op">=</span> <span class="va">Freq</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span></span>
<span>    `% from total` <span class="op">=</span> <span class="va">`Amount of Spots`</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">`Amount of Spots`</span><span class="op">)</span>,</span>
<span>    `% from total` <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/percent_format.html">percent</a></span><span class="op">(</span><span class="va">`% from total`</span>, accuracy <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Add total row</span></span>
<span><span class="va">total_spots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">region_table</span><span class="op">$</span><span class="va">`Amount of Spots`</span><span class="op">)</span></span>
<span><span class="va">total_row</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span></span>
<span>  Region <span class="op">=</span> <span class="st">"Total"</span>,</span>
<span>  `Amount of Spots` <span class="op">=</span> <span class="va">total_spots</span>,</span>
<span>  `% from total` <span class="op">=</span> <span class="st">"100.0%"</span>,</span>
<span>  check.names <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Combine with original data</span></span>
<span><span class="va">region_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">region_table</span>, <span class="va">total_row</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create flextable</span></span>
<span><span class="va">ft_region</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/flextable.html">flextable</a></span><span class="op">(</span><span class="va">region_table</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/theme_vanilla.html">theme_vanilla</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/set_header_labels.html">set_header_labels</a></span><span class="op">(</span></span>
<span>    Region <span class="op">=</span> <span class="st">"Brain region"</span>,</span>
<span>    `Amount of Spots` <span class="op">=</span> <span class="st">"Amount of Spots"</span>,</span>
<span>    `% from total` <span class="op">=</span> <span class="st">"% from total"</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/add_header_lines.html">add_header_lines</a></span><span class="op">(</span>values <span class="op">=</span> <span class="st">"Regional transcriptomic representation of mouse brain section (coronal)*"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/add_footer_lines.html">add_footer_lines</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"* Source: https://www.10xgenomics.com/datasets/mouse-brain-section-coronal-1-standard-1-0-0"</span>, </span>
<span>                            <span class="st">"* Platform: Visium Spatial 10X Genomics"</span>,</span>
<span>                            <span class="st">"* Preservation method: Fresh Frozen"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/align.html">align</a></span><span class="op">(</span>align <span class="op">=</span> <span class="st">"center"</span>, part <span class="op">=</span> <span class="st">"header"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/align.html">align</a></span><span class="op">(</span>align <span class="op">=</span> <span class="st">"center"</span>, part <span class="op">=</span> <span class="st">"body"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/align.html">align</a></span><span class="op">(</span>align <span class="op">=</span> <span class="st">"left"</span>, part <span class="op">=</span> <span class="st">"footer"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/bold.html">bold</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Region</span> <span class="op">==</span> <span class="st">"Total"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/bg.html">bg</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Region</span> <span class="op">==</span> <span class="st">"Total"</span>, bg <span class="op">=</span> <span class="st">"#2D2D2D"</span>, part <span class="op">=</span> <span class="st">"body"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>  </span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/color.html">color</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Region</span> <span class="op">==</span> <span class="st">"Total"</span>, color <span class="op">=</span> <span class="st">"white"</span>, part <span class="op">=</span> <span class="st">"body"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/border_inner_h.html">border_inner_h</a></span><span class="op">(</span>border <span class="op">=</span> <span class="fu"><a href="https://davidgohel.github.io/officer/reference/fp_border.html">fp_border</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"#D3D3D3"</span>, width <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/border_outer.html">border_outer</a></span><span class="op">(</span>border <span class="op">=</span> <span class="fu"><a href="https://davidgohel.github.io/officer/reference/fp_border.html">fp_border</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"#D3D3D3"</span>, width <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/fontsize.html">fontsize</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span>, part <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/font.html">font</a></span><span class="op">(</span>fontname <span class="op">=</span> <span class="st">"Arial"</span>, part <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/autofit.html">autofit</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># print(ft_region)</span></span>
<span></span>
<span><span class="co"># Save 'ft_region'</span></span>
<span><span class="co"># Convert flextable to a grob</span></span>
<span><span class="co"># ft_grob2 &lt;- flextable::gen_grob(ft_region)</span></span>
<span><span class="co"># ggsave("~path/to/your/data/ft_region_brainregions_descriptive_table.png", ft_grob2, width = 10, height = 8, dpi = 600)</span></span>
<span></span>
<span><span class="co"># view(brain@meta.data)</span></span>
<span></span>
<span><span class="co"># Render flextable for regions</span></span>
<span><span class="fu">flextable</span><span class="fu">::</span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/htmltools_value.html">htmltools_value</a></span><span class="op">(</span><span class="va">ft_region</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<style></style>
<div class="tabwid">
<style>.cl-39be6d56{}.cl-39b14bf8{font-family:'Arial';font-size:16pt;font-weight:bold;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-39b14c2a{font-family:'Arial';font-size:16pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-39b14c34{font-family:'Arial';font-size:16pt;font-weight:bold;font-style:normal;text-decoration:none;color:rgba(255, 255, 255, 1.00);background-color:transparent;}.cl-39b73afe{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-39b73b12{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-39b77884{width:3.262in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0.5pt solid rgba(211, 211, 211, 1.00);border-right: 0.5pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b7788e{width:2.075in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b77898{width:1.569in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0.5pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778a2{width:3.262in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0.5pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778ac{width:2.075in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778ad{width:1.569in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0.5pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778b6{width:3.262in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0.5pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778b7{width:2.075in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778c0{width:1.569in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0.5pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778c1{width:3.262in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0.5pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778ca{width:2.075in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778cb{width:1.569in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0.5pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778d4{width:3.262in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0.5pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778d5{width:2.075in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778d6{width:1.569in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0.5pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778de{width:3.262in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0.5pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778e8{width:2.075in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778e9{width:1.569in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0.5pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778ea{width:3.262in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0.5pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778f2{width:2.075in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778fc{width:1.569in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0.5pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b778fd{width:3.262in;background-color:rgba(45, 45, 45, 1.00);vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0.5pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b77906{width:2.075in;background-color:rgba(45, 45, 45, 1.00);vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b77907{width:1.569in;background-color:rgba(45, 45, 45, 1.00);vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0.5pt solid rgba(211, 211, 211, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0.5pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b77910{width:3.262in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0.5pt solid rgba(211, 211, 211, 1.00);border-right: 0.5pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b77911{width:2.075in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b7791a{width:1.569in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(255, 255, 255, 0.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 0.5pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b7791b{width:3.262in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0.5pt solid rgba(211, 211, 211, 1.00);border-right: 0.5pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b77924{width:2.075in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-39b77925{width:1.569in;background-color:transparent;vertical-align: middle;border-bottom: 0.5pt solid rgba(211, 211, 211, 1.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 0.5pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style>
<table data-quarto-disable-processing="true" class="cl-39be6d56">
<thead>
<tr style="overflow-wrap:break-word;"><th colspan="3" class="cl-39b77884"><p class="cl-39b73afe"><span class="cl-39b14bf8">Regional transcriptomic representation of mouse brain section (coronal)*</span></p></th></tr>
<tr style="overflow-wrap:break-word;">
<th class="cl-39b778a2"><p class="cl-39b73afe"><span class="cl-39b14bf8">Brain region</span></p></th>
<th class="cl-39b778ac"><p class="cl-39b73afe"><span class="cl-39b14bf8">Amount of Spots</span></p></th>
<th class="cl-39b778ad"><p class="cl-39b73afe"><span class="cl-39b14bf8">% from total</span></p></th>
</tr>
</thead>
<tbody>
<tr style="overflow-wrap:break-word;">
<td class="cl-39b778b6"><p class="cl-39b73afe"><span class="cl-39b14c2a">Allocortex</span></p></td>
<td class="cl-39b778b7"><p class="cl-39b73afe"><span class="cl-39b14c2a">348</span></p></td>
<td class="cl-39b778c0"><p class="cl-39b73afe"><span class="cl-39b14c2a">12.9%</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-39b778c1"><p class="cl-39b73afe"><span class="cl-39b14c2a">Caudatoputamen</span></p></td>
<td class="cl-39b778ca"><p class="cl-39b73afe"><span class="cl-39b14c2a">103</span></p></td>
<td class="cl-39b778cb"><p class="cl-39b73afe"><span class="cl-39b14c2a">3.8%</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-39b778d4"><p class="cl-39b73afe"><span class="cl-39b14c2a">Fiber tracts</span></p></td>
<td class="cl-39b778d5"><p class="cl-39b73afe"><span class="cl-39b14c2a">634</span></p></td>
<td class="cl-39b778d6"><p class="cl-39b73afe"><span class="cl-39b14c2a">23.5%</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-39b778de"><p class="cl-39b73afe"><span class="cl-39b14c2a">Hippocampal region</span></p></td>
<td class="cl-39b778e8"><p class="cl-39b73afe"><span class="cl-39b14c2a">297</span></p></td>
<td class="cl-39b778e9"><p class="cl-39b73afe"><span class="cl-39b14c2a">11.0%</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-39b778de"><p class="cl-39b73afe"><span class="cl-39b14c2a">Hypothalamus</span></p></td>
<td class="cl-39b778e8"><p class="cl-39b73afe"><span class="cl-39b14c2a">267</span></p></td>
<td class="cl-39b778e9"><p class="cl-39b73afe"><span class="cl-39b14c2a">9.9%</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-39b778d4"><p class="cl-39b73afe"><span class="cl-39b14c2a">N/A</span></p></td>
<td class="cl-39b778d5"><p class="cl-39b73afe"><span class="cl-39b14c2a">60</span></p></td>
<td class="cl-39b778d6"><p class="cl-39b73afe"><span class="cl-39b14c2a">2.2%</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-39b778d4"><p class="cl-39b73afe"><span class="cl-39b14c2a">Neocortex</span></p></td>
<td class="cl-39b778d5"><p class="cl-39b73afe"><span class="cl-39b14c2a">695</span></p></td>
<td class="cl-39b778d6"><p class="cl-39b73afe"><span class="cl-39b14c2a">25.8%</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-39b778ea"><p class="cl-39b73afe"><span class="cl-39b14c2a">Striatum-like amygdalar nuclei</span></p></td>
<td class="cl-39b778f2"><p class="cl-39b73afe"><span class="cl-39b14c2a">38</span></p></td>
<td class="cl-39b778fc"><p class="cl-39b73afe"><span class="cl-39b14c2a">1.4%</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-39b778d4"><p class="cl-39b73afe"><span class="cl-39b14c2a">Thalamus</span></p></td>
<td class="cl-39b778d5"><p class="cl-39b73afe"><span class="cl-39b14c2a">256</span></p></td>
<td class="cl-39b778d6"><p class="cl-39b73afe"><span class="cl-39b14c2a">9.5%</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-39b778fd"><p class="cl-39b73afe"><span class="cl-39b14c34">Total</span></p></td>
<td class="cl-39b77906"><p class="cl-39b73afe"><span class="cl-39b14c34">2,698</span></p></td>
<td class="cl-39b77907"><p class="cl-39b73afe"><span class="cl-39b14c34">100.0%</span></p></td>
</tr>
</tbody>
<tfoot>
<tr style="overflow-wrap:break-word;"><td colspan="3" class="cl-39b77910"><p class="cl-39b73b12"><span class="cl-39b14c2a">* Source: https://www.10xgenomics.com/datasets/mouse-brain-section-coronal-1-standard-1-0-0</span></p></td></tr>
<tr style="overflow-wrap:break-word;"><td colspan="3" class="cl-39b77910"><p class="cl-39b73b12"><span class="cl-39b14c2a">* Platform: Visium Spatial 10X Genomics</span></p></td></tr>
<tr style="overflow-wrap:break-word;"><td colspan="3" class="cl-39b7791b"><p class="cl-39b73b12"><span class="cl-39b14c2a">* Preservation method: Fresh Frozen</span></p></td></tr>
</tfoot>
</table>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb102"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#################</span></span>
<span><span class="co">#################</span></span>
<span><span class="co"># Identify the top markers for your specific dataset</span></span>
<span><span class="co"># 1. ROC Analysis (AUC calculation)</span></span>
<span><span class="va">top_markers_roc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html">FindAllMarkers</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>,</span>
<span>  group.by <span class="op">=</span> <span class="st">"brain_region"</span>,</span>
<span>  test.use <span class="op">=</span> <span class="st">"roc"</span>,            <span class="co"># ROC test for AUC values</span></span>
<span>  logfc.threshold <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>  only.pos <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  min.pct <span class="op">=</span> <span class="fl">0.1</span>                <span class="co"># Gene detected in ≥10% of spots in either region</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Get top 5 markers per region (sorted by AUC power)</span></span>
<span><span class="va">top5_markers</span> <span class="op">&lt;-</span> <span class="va">top_markers_roc</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>        <span class="co"># "cluster" = brain_region</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span><span class="va">power</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>     <span class="co"># "power" = AUC (0.5 = random, 1 = perfect)</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice_head</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>            <span class="co"># Top 5 per region</span></span>
<span></span>
<span><span class="co"># View results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">top5_markers</span>, n <span class="op">=</span> <span class="fl">45</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code># A tibble: 45 × 8
# Groups:   cluster [9]
   myAUC avg_diff power avg_log2FC pct.1 pct.2 cluster                     gene
   &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;fct&gt;                       &lt;chr&gt;
 1 0.905    1.23  0.81       1.92  1     0.925 Allocortex                  Nptxr
 2 0.839    1.14  0.678      2.21  0.925 0.455 Allocortex                  Slc3…
 3 0.835    0.650 0.67       1.05  1     0.917 Allocortex                  Hpca…
 4 0.829    1.54  0.658      2.75  0.951 0.619 Allocortex                  Lypd1
 5 0.825    1.03  0.65       2.00  0.914 0.571 Allocortex                  Lmo3
 6 0.897    1.75  0.794      2.54  1     0.998 Fiber tracts                Mbp
 7 0.89     1.80  0.78       2.70  1     0.908 Fiber tracts                Mobp
 8 0.889    1.31  0.778      2.10  0.997 0.94  Fiber tracts                Qdpr
 9 0.887    1.30  0.774      2.06  0.998 0.913 Fiber tracts                Plek…
10 0.884    0.782 0.768      1.17  1     1     Fiber tracts                Scd2
11 0.934    1.22  0.868      2.03  0.999 0.798 Neocortex                   Mef2c
12 0.929    1.20  0.858      1.77  1     0.964 Neocortex                   Nrgn
13 0.928    1.01  0.856      1.60  1     0.99  Neocortex                   1110…
14 0.918    1.11  0.836      2.02  0.996 0.669 Neocortex                   Stx1a
15 0.916    1.07  0.832      1.57  1     1     Neocortex                   Camk…
16 0.978    2.62  0.956      4.71  1     0.408 N/A                         Mgp
17 0.976    2.88  0.952      4.20  1     0.99  N/A                         Ptgds
18 0.964    1.63  0.928      4.66  0.967 0.149 N/A                         Col1…
19 0.94     1.38  0.88       4.99  0.917 0.089 N/A                         Slc6…
20 0.94     1.27  0.88       2.29  1     0.695 N/A                         Id3
21 0.963    1.56  0.926      2.46  1     0.952 Hypothalamus                Resp…
22 0.962    0.609 0.924      0.897 1     1     Hypothalamus                Rpl3…
23 0.959    1.32  0.918      2.04  1     0.979 Hypothalamus                6330…
24 0.956    0.526 0.912      0.774 1     1     Hypothalamus                Rpl37
25 0.956    0.578 0.912      0.852 1     1     Hypothalamus                Rpl9
26 0.994    1.84  0.988      3.02  1     0.7   Striatum-like amygdalar nu… Hap1
27 0.983    1.21  0.966      1.87  1     0.995 Striatum-like amygdalar nu… Gaa
28 0.983    1.50  0.966      2.42  1     0.886 Striatum-like amygdalar nu… Ahi1
29 0.98     1.09  0.96       1.65  1     0.988 Striatum-like amygdalar nu… Ly6h
30 0.979    1.04  0.958      1.60  1     0.983 Striatum-like amygdalar nu… Gap43
31 0.899    0.995 0.798      1.57  1     0.976 Hippocampal region          Cnih2
32 0.89     1.28  0.78       3.25  0.862 0.287 Hippocampal region          Cabp7
33 0.874    0.763 0.748      1.20  1     0.945 Hippocampal region          Psd
34 0.873    1.76  0.746      2.72  1     0.939 Hippocampal region          Hpca
35 0.871    0.986 0.742      1.59  1     0.821 Hippocampal region          Ddn
36 0.994    1.74  0.988      2.99  1     0.811 Thalamus                    Adar…
37 0.993    1.87  0.986      3.37  1     0.671 Thalamus                    Rora
38 0.992    2.77  0.984      4.90  0.996 0.41  Thalamus                    Prkcd
39 0.981    1.57  0.962      3.34  0.996 0.408 Thalamus                    Amot…
40 0.978    1.83  0.956      2.72  1     0.988 Thalamus                    Pcp4
41 0.896    1.91  0.792      3.42  0.932 0.585 Caudatoputamen              Ppp1…
42 0.893    0.388 0.786      0.568 1     1     Caudatoputamen              Rpl41
43 0.888    4.58  0.776      6.70  1     0.983 Caudatoputamen              Ttr
44 0.878    0.981 0.756      1.82  0.961 0.755 Caudatoputamen              Id4
45 0.855    0.959 0.71       1.70  0.981 0.755 Caudatoputamen              Pcp4…</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb104"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Dot plot of top markers (expression + percentage)</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DotPlot.html">DotPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">top5_markers</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span>,</span>
<span>  group.by <span class="op">=</span> <span class="st">"brain_region"</span>,</span>
<span>  dot.scale <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  col.min <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  col.max <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html">RotatedAxis</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_color_viridis_c</a></span><span class="op">(</span>option <span class="op">=</span> <span class="st">"magma"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span></span>
<span>    axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"right"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="st">"Genes"</span>,</span>
<span>    y <span class="op">=</span> <span class="st">"Brain Region"</span>,</span>
<span>    color <span class="op">=</span> <span class="st">"Avg. Expression"</span>,</span>
<span>    size <span class="op">=</span> <span class="st">"% Spots Expressed"</span></span>
<span>  <span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-9-6.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb105"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># DEGs between Neocortex &amp; Thalamus</span></span>
<span><span class="va">neocortex_vs_thalamus</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindMarkers.html">FindMarkers</a></span><span class="op">(</span></span>
<span>  <span class="va">brain</span>, </span>
<span>  ident.1 <span class="op">=</span> <span class="st">"Neocortex"</span>, </span>
<span>  ident.2 <span class="op">=</span> <span class="st">"Thalamus"</span>, </span>
<span>  group.by <span class="op">=</span> <span class="st">"brain_region"</span>, </span>
<span>  logfc.threshold <span class="op">=</span> <span class="fl">0.25</span>, </span>
<span>  test.use <span class="op">=</span> <span class="st">"wilcox"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">neocortex_vs_thalamus</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>                p_val avg_log2FC pct.1 pct.2     p_val_adj
Shox2   1.630478e-169  -7.034157 0.024 0.914 2.947578e-165
Synpo2  5.193043e-163  -6.003112 0.066 0.941 9.387983e-159
Tcf7l2  2.113409e-157  -5.714385 0.181 1.000 3.820620e-153
Tnnt1   2.267419e-156  -6.200552 0.114 0.961 4.099040e-152
Zic1    1.066981e-154  -5.287955 0.131 0.973 1.928887e-150
Plekhg1 1.973419e-149  -5.379145 0.125 0.949 3.567548e-145</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb107"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># library(clusterProfiler)</span></span>
<span><span class="co"># library(org.Mm.eg.db)  # Mouse gene annotations</span></span>
<span></span>
<span><span class="co"># Convert top genes to ENTREZ IDs</span></span>
<span><span class="va">genes_entrez</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">mapIds</a></span><span class="op">(</span></span>
<span>  <span class="va">org.Mm.eg.db</span>, </span>
<span>  keys <span class="op">=</span> <span class="va">top5_markers</span><span class="op">$</span><span class="va">gene</span>, </span>
<span>  column <span class="op">=</span> <span class="st">"ENTREZID"</span>, </span>
<span>  keytype <span class="op">=</span> <span class="st">"SYMBOL"</span>,</span>
<span>  multiVals <span class="op">=</span> <span class="st">"first"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform GO enrichment (Biological Process)</span></span>
<span><span class="va">go_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html">enrichGO</a></span><span class="op">(</span></span>
<span>  gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html">na.omit</a></span><span class="op">(</span><span class="va">genes_entrez</span><span class="op">)</span>, </span>
<span>  OrgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>, </span>
<span>  keyType <span class="op">=</span> <span class="st">"ENTREZID"</span>, </span>
<span>  ont <span class="op">=</span> <span class="st">"BP"</span>, </span>
<span>  pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot top GO terms</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="va">go_results</span>, showCategory <span class="op">=</span> <span class="fl">10</span>, title <span class="op">=</span> <span class="st">"GO Enrichment: Brain Regions"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="260225_Spatial_Mouse_Brain_Coronal_10x_files/figure-html/unnamed-chunk-9-7.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb108"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">######### Flextable of highly expressed genes per brain region ###########</span></span>
<span></span>
<span><span class="co"># Prepare table</span></span>
<span><span class="va">marker_table</span> <span class="op">&lt;-</span> <span class="va">top5_markers</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">gene</span>, <span class="va">cluster</span>, <span class="va">power</span>, <span class="va">avg_log2FC</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span></span>
<span>    Gene <span class="op">=</span> <span class="va">gene</span>,</span>
<span>    `Brain Region` <span class="op">=</span> <span class="va">cluster</span>,</span>
<span>    `AUC Power` <span class="op">=</span> <span class="va">power</span>,</span>
<span>    `Log2 FC` <span class="op">=</span> <span class="va">avg_log2FC</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>`AUC Power` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">`AUC Power`</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>  <span class="co"># Round AUC to 2 decimals</span></span>
<span></span>
<span><span class="co"># Create flextable</span></span>
<span><span class="va">ft_markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/flextable.html">flextable</a></span><span class="op">(</span><span class="va">marker_table</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/theme_vanilla.html">theme_vanilla</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/set_header_labels.html">set_header_labels</a></span><span class="op">(</span></span>
<span>    Gene <span class="op">=</span> <span class="st">"Gene"</span>,</span>
<span>    `Brain Region` <span class="op">=</span> <span class="st">"Brain Region"</span>,</span>
<span>    `AUC Power` <span class="op">=</span> <span class="st">"AUC Power"</span>,</span>
<span>    `Log2 FC` <span class="op">=</span> <span class="st">"Log2 Fold Change"</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/add_header_lines.html">add_header_lines</a></span><span class="op">(</span>values <span class="op">=</span> <span class="st">"Top 5 Predictive Genes per Brain Region (ROC Analysis)*"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/add_footer_lines.html">add_footer_lines</a></span><span class="op">(</span>values <span class="op">=</span> <span class="st">"*AUC Power: 1 = perfect classifier, 0.5 = random."</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/align.html">align</a></span><span class="op">(</span>align <span class="op">=</span> <span class="st">"center"</span>, part <span class="op">=</span> <span class="st">"header"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/align.html">align</a></span><span class="op">(</span>align <span class="op">=</span> <span class="st">"center"</span>, part <span class="op">=</span> <span class="st">"body"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/bg.html">bg</a></span><span class="op">(</span><span class="op">~</span> <span class="va">`AUC Power`</span> <span class="op">&gt;</span> <span class="fl">0.9</span>, bg <span class="op">=</span> <span class="st">"#E6F3FF"</span>, part <span class="op">=</span> <span class="st">"body"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>  <span class="co"># Highlight high AUC</span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/border_outer.html">border_outer</a></span><span class="op">(</span>border <span class="op">=</span> <span class="fu"><a href="https://davidgohel.github.io/officer/reference/fp_border.html">fp_border</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"#D3D3D3"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/font.html">font</a></span><span class="op">(</span>fontname <span class="op">=</span> <span class="st">"Arial"</span>, part <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://davidgohel.github.io/flextable/reference/autofit.html">autofit</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># print(ft_markers)</span></span>
<span></span>
<span><span class="co"># Save 'ft_markers'</span></span>
<span><span class="co"># Convert flextable to a grob</span></span>
<span><span class="co"># ft_grob2 &lt;- flextable::gen_grob(ft_markers)</span></span>
<span><span class="co"># ggsave("~path/to/your/data/ft_markers_brainregions_Top5_predictive_genes_table.png", ft_grob2, width = 10, height = 8, dpi = 600)</span></span>
<span></span>
<span><span class="co"># Render flextable for markers</span></span>
<span><span class="fu">flextable</span><span class="fu">::</span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/htmltools_value.html">htmltools_value</a></span><span class="op">(</span><span class="va">ft_markers</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<style></style>
<div class="tabwid">
<style>.cl-7caa704a{}.cl-7c9993b0{font-family:'Arial';font-size:11pt;font-weight:bold;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-7c9993c4{font-family:'Arial';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-7c9f5340{margin:0;text-align:center;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-7c9f534a{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-7ca001fa{width:1.372in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(211, 211, 211, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0020e{width:2.331in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(211, 211, 211, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0020f{width:1.117in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(211, 211, 211, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00218{width:1.618in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(211, 211, 211, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00219{width:1.372in;background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(211, 211, 211, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00222{width:2.331in;background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(211, 211, 211, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0022c{width:1.117in;background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(211, 211, 211, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0022d{width:1.618in;background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(211, 211, 211, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0022e{width:1.372in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00236{width:2.331in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00240{width:1.117in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00241{width:1.618in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00242{width:1.372in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0024a{width:2.331in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00254{width:1.117in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00255{width:1.618in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0025e{width:1.372in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0025f{width:2.331in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00268{width:1.117in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00269{width:1.618in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00272{width:1.372in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0029a{width:2.331in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002a4{width:1.117in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002ae{width:1.618in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002af{width:1.372in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002b0{width:2.331in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002b8{width:1.117in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002b9{width:1.618in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002c2{width:1.372in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002c3{width:2.331in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002cc{width:1.117in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002cd{width:1.618in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002d6{width:1.372in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002d7{width:2.331in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002e0{width:1.117in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002e1{width:1.618in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002ea{width:1.372in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002eb{width:2.331in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002f4{width:1.117in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002f5{width:1.618in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002fe{width:1.372in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca002ff{width:2.331in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00308{width:1.117in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00309{width:1.618in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00312{width:1.372in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00313{width:2.331in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0031c{width:1.117in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0031d{width:1.618in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0031e{width:1.372in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00326{width:2.331in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00330{width:1.117in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00331{width:1.618in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00332{width:1.372in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0033a{width:2.331in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0033b{width:1.117in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00344{width:1.618in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00345{width:1.372in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0034e{width:2.331in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0034f{width:1.117in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00358{width:1.618in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00362{width:1.372in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00363{width:2.331in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0036c{width:1.117in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0036d{width:1.618in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0036e{width:1.372in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00376{width:2.331in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00380{width:1.117in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00381{width:1.618in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0038a{width:1.372in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0038b{width:2.331in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00394{width:1.117in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00395{width:1.618in;background-color:rgba(230, 243, 255, 1.00);vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca00396{width:1.372in;background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(211, 211, 211, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0039e{width:2.331in;background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(211, 211, 211, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca0039f{width:1.117in;background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(211, 211, 211, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca003a8{width:1.618in;background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(211, 211, 211, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca003a9{width:1.372in;background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(211, 211, 211, 1.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 1pt solid rgba(211, 211, 211, 1.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca003b2{width:2.331in;background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(211, 211, 211, 1.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca003b3{width:1.117in;background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(211, 211, 211, 1.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 0 solid rgba(255, 255, 255, 0.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-7ca003bc{width:1.618in;background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(211, 211, 211, 1.00);border-top: 0 solid rgba(255, 255, 255, 0.00);border-left: 0 solid rgba(255, 255, 255, 0.00);border-right: 1pt solid rgba(211, 211, 211, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style>
<table data-quarto-disable-processing="true" class="cl-7caa704a">
<thead>
<tr style="overflow-wrap:break-word;"><th colspan="4" class="cl-7ca001fa"><p class="cl-7c9f5340"><span class="cl-7c9993b0">Top 5 Predictive Genes per Brain Region (ROC Analysis)*</span></p></th></tr>
<tr style="overflow-wrap:break-word;">
<th class="cl-7ca00219"><p class="cl-7c9f5340"><span class="cl-7c9993b0">Gene</span></p></th>
<th class="cl-7ca00222"><p class="cl-7c9f5340"><span class="cl-7c9993b0">Brain Region</span></p></th>
<th class="cl-7ca0022c"><p class="cl-7c9f5340"><span class="cl-7c9993b0">AUC Power</span></p></th>
<th class="cl-7ca0022d"><p class="cl-7c9f5340"><span class="cl-7c9993b0">Log2 Fold Change</span></p></th>
</tr>
</thead>
<tbody>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca0022e"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Nptxr</span></p></td>
<td class="cl-7ca00236"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Allocortex</span></p></td>
<td class="cl-7ca00240"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.81</span></p></td>
<td class="cl-7ca00241"><p class="cl-7c9f5340"><span class="cl-7c9993c4">1.9173356</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00242"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Slc30a3</span></p></td>
<td class="cl-7ca0024a"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Allocortex</span></p></td>
<td class="cl-7ca00254"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.68</span></p></td>
<td class="cl-7ca00255"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.2138359</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca0025e"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Hpcal4</span></p></td>
<td class="cl-7ca0025f"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Allocortex</span></p></td>
<td class="cl-7ca00268"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.67</span></p></td>
<td class="cl-7ca00269"><p class="cl-7c9f5340"><span class="cl-7c9993c4">1.0487553</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00272"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Lypd1</span></p></td>
<td class="cl-7ca0029a"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Allocortex</span></p></td>
<td class="cl-7ca002a4"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.66</span></p></td>
<td class="cl-7ca002ae"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.7473660</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca002af"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Lmo3</span></p></td>
<td class="cl-7ca002b0"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Allocortex</span></p></td>
<td class="cl-7ca002b8"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.65</span></p></td>
<td class="cl-7ca002b9"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.0038215</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca0025e"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Mbp</span></p></td>
<td class="cl-7ca0025f"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Fiber tracts</span></p></td>
<td class="cl-7ca00268"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.79</span></p></td>
<td class="cl-7ca00269"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.5416660</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca0025e"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Mobp</span></p></td>
<td class="cl-7ca0025f"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Fiber tracts</span></p></td>
<td class="cl-7ca00268"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.78</span></p></td>
<td class="cl-7ca00269"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.7035539</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca002c2"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Qdpr</span></p></td>
<td class="cl-7ca002c3"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Fiber tracts</span></p></td>
<td class="cl-7ca002cc"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.78</span></p></td>
<td class="cl-7ca002cd"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.0972666</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca002af"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Plekhb1</span></p></td>
<td class="cl-7ca002b0"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Fiber tracts</span></p></td>
<td class="cl-7ca002b8"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.77</span></p></td>
<td class="cl-7ca002b9"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.0567973</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca002d6"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Scd2</span></p></td>
<td class="cl-7ca002d7"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Fiber tracts</span></p></td>
<td class="cl-7ca002e0"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.77</span></p></td>
<td class="cl-7ca002e1"><p class="cl-7c9f5340"><span class="cl-7c9993c4">1.1712759</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca002ea"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Mef2c</span></p></td>
<td class="cl-7ca002eb"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Neocortex</span></p></td>
<td class="cl-7ca002f4"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.87</span></p></td>
<td class="cl-7ca002f5"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.0321465</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca002fe"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Nrgn</span></p></td>
<td class="cl-7ca002ff"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Neocortex</span></p></td>
<td class="cl-7ca00308"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.86</span></p></td>
<td class="cl-7ca00309"><p class="cl-7c9f5340"><span class="cl-7c9993c4">1.7740790</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca002af"><p class="cl-7c9f5340"><span class="cl-7c9993c4">1110008P14Rik</span></p></td>
<td class="cl-7ca002b0"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Neocortex</span></p></td>
<td class="cl-7ca002b8"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.86</span></p></td>
<td class="cl-7ca002b9"><p class="cl-7c9f5340"><span class="cl-7c9993c4">1.6033309</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca002d6"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Stx1a</span></p></td>
<td class="cl-7ca002d7"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Neocortex</span></p></td>
<td class="cl-7ca002e0"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.84</span></p></td>
<td class="cl-7ca002e1"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.0241399</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca002d6"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Camk2n1</span></p></td>
<td class="cl-7ca002d7"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Neocortex</span></p></td>
<td class="cl-7ca002e0"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.83</span></p></td>
<td class="cl-7ca002e1"><p class="cl-7c9f5340"><span class="cl-7c9993c4">1.5744997</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00312"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Mgp</span></p></td>
<td class="cl-7ca00313"><p class="cl-7c9f5340"><span class="cl-7c9993c4">N/A</span></p></td>
<td class="cl-7ca0031c"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.96</span></p></td>
<td class="cl-7ca0031d"><p class="cl-7c9f5340"><span class="cl-7c9993c4">4.7124062</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00312"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Ptgds</span></p></td>
<td class="cl-7ca00313"><p class="cl-7c9f5340"><span class="cl-7c9993c4">N/A</span></p></td>
<td class="cl-7ca0031c"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.95</span></p></td>
<td class="cl-7ca0031d"><p class="cl-7c9f5340"><span class="cl-7c9993c4">4.1988364</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca0031e"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Col1a2</span></p></td>
<td class="cl-7ca00326"><p class="cl-7c9f5340"><span class="cl-7c9993c4">N/A</span></p></td>
<td class="cl-7ca00330"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.93</span></p></td>
<td class="cl-7ca00331"><p class="cl-7c9f5340"><span class="cl-7c9993c4">4.6622934</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00242"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Slc6a13</span></p></td>
<td class="cl-7ca0024a"><p class="cl-7c9f5340"><span class="cl-7c9993c4">N/A</span></p></td>
<td class="cl-7ca00254"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.88</span></p></td>
<td class="cl-7ca00255"><p class="cl-7c9f5340"><span class="cl-7c9993c4">4.9912122</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca002d6"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Id3</span></p></td>
<td class="cl-7ca002d7"><p class="cl-7c9f5340"><span class="cl-7c9993c4">N/A</span></p></td>
<td class="cl-7ca002e0"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.88</span></p></td>
<td class="cl-7ca002e1"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.2926602</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00312"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Resp18</span></p></td>
<td class="cl-7ca00313"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Hypothalamus</span></p></td>
<td class="cl-7ca0031c"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.93</span></p></td>
<td class="cl-7ca0031d"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.4563417</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00312"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Rpl35a</span></p></td>
<td class="cl-7ca00313"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Hypothalamus</span></p></td>
<td class="cl-7ca0031c"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.92</span></p></td>
<td class="cl-7ca0031d"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.8974079</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00312"><p class="cl-7c9f5340"><span class="cl-7c9993c4">6330403K07Rik</span></p></td>
<td class="cl-7ca00313"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Hypothalamus</span></p></td>
<td class="cl-7ca0031c"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.92</span></p></td>
<td class="cl-7ca0031d"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.0360217</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00312"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Rpl37</span></p></td>
<td class="cl-7ca00313"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Hypothalamus</span></p></td>
<td class="cl-7ca0031c"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.91</span></p></td>
<td class="cl-7ca0031d"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.7738712</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00312"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Rpl9</span></p></td>
<td class="cl-7ca00313"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Hypothalamus</span></p></td>
<td class="cl-7ca0031c"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.91</span></p></td>
<td class="cl-7ca0031d"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.8519425</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00332"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Hap1</span></p></td>
<td class="cl-7ca0033a"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Striatum-like amygdalar nuclei</span></p></td>
<td class="cl-7ca0033b"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.99</span></p></td>
<td class="cl-7ca00344"><p class="cl-7c9f5340"><span class="cl-7c9993c4">3.0180617</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00332"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Gaa</span></p></td>
<td class="cl-7ca0033a"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Striatum-like amygdalar nuclei</span></p></td>
<td class="cl-7ca0033b"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.97</span></p></td>
<td class="cl-7ca00344"><p class="cl-7c9f5340"><span class="cl-7c9993c4">1.8670969</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00332"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Ahi1</span></p></td>
<td class="cl-7ca0033a"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Striatum-like amygdalar nuclei</span></p></td>
<td class="cl-7ca0033b"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.97</span></p></td>
<td class="cl-7ca00344"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.4165205</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00332"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Ly6h</span></p></td>
<td class="cl-7ca0033a"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Striatum-like amygdalar nuclei</span></p></td>
<td class="cl-7ca0033b"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.96</span></p></td>
<td class="cl-7ca00344"><p class="cl-7c9f5340"><span class="cl-7c9993c4">1.6508453</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00332"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Gap43</span></p></td>
<td class="cl-7ca0033a"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Striatum-like amygdalar nuclei</span></p></td>
<td class="cl-7ca0033b"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.96</span></p></td>
<td class="cl-7ca00344"><p class="cl-7c9f5340"><span class="cl-7c9993c4">1.6048253</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca002fe"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Cnih2</span></p></td>
<td class="cl-7ca002ff"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Hippocampal region</span></p></td>
<td class="cl-7ca00308"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.80</span></p></td>
<td class="cl-7ca00309"><p class="cl-7c9f5340"><span class="cl-7c9993c4">1.5733312</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00345"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Cabp7</span></p></td>
<td class="cl-7ca0034e"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Hippocampal region</span></p></td>
<td class="cl-7ca0034f"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.78</span></p></td>
<td class="cl-7ca00358"><p class="cl-7c9f5340"><span class="cl-7c9993c4">3.2522224</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca002fe"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Psd</span></p></td>
<td class="cl-7ca002ff"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Hippocampal region</span></p></td>
<td class="cl-7ca00308"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.75</span></p></td>
<td class="cl-7ca00309"><p class="cl-7c9f5340"><span class="cl-7c9993c4">1.2039348</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca002fe"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Hpca</span></p></td>
<td class="cl-7ca002ff"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Hippocampal region</span></p></td>
<td class="cl-7ca00308"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.75</span></p></td>
<td class="cl-7ca00309"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.7224833</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca002fe"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Ddn</span></p></td>
<td class="cl-7ca002ff"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Hippocampal region</span></p></td>
<td class="cl-7ca00308"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.74</span></p></td>
<td class="cl-7ca00309"><p class="cl-7c9f5340"><span class="cl-7c9993c4">1.5901991</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00362"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Adarb1</span></p></td>
<td class="cl-7ca00363"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Thalamus</span></p></td>
<td class="cl-7ca0036c"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.99</span></p></td>
<td class="cl-7ca0036d"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.9863463</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00362"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Rora</span></p></td>
<td class="cl-7ca00363"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Thalamus</span></p></td>
<td class="cl-7ca0036c"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.99</span></p></td>
<td class="cl-7ca0036d"><p class="cl-7c9f5340"><span class="cl-7c9993c4">3.3680673</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca0036e"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Prkcd</span></p></td>
<td class="cl-7ca00376"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Thalamus</span></p></td>
<td class="cl-7ca00380"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.98</span></p></td>
<td class="cl-7ca00381"><p class="cl-7c9f5340"><span class="cl-7c9993c4">4.8971991</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00362"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Amotl1</span></p></td>
<td class="cl-7ca00363"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Thalamus</span></p></td>
<td class="cl-7ca0036c"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.96</span></p></td>
<td class="cl-7ca0036d"><p class="cl-7c9f5340"><span class="cl-7c9993c4">3.3424724</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca0038a"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Pcp4</span></p></td>
<td class="cl-7ca0038b"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Thalamus</span></p></td>
<td class="cl-7ca00394"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.96</span></p></td>
<td class="cl-7ca00395"><p class="cl-7c9f5340"><span class="cl-7c9993c4">2.7165748</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00345"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Ppp1r1b</span></p></td>
<td class="cl-7ca0034e"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Caudatoputamen</span></p></td>
<td class="cl-7ca0034f"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.79</span></p></td>
<td class="cl-7ca00358"><p class="cl-7c9f5340"><span class="cl-7c9993c4">3.4172123</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00345"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Rpl41</span></p></td>
<td class="cl-7ca0034e"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Caudatoputamen</span></p></td>
<td class="cl-7ca0034f"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.79</span></p></td>
<td class="cl-7ca00358"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.5678003</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00345"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Ttr</span></p></td>
<td class="cl-7ca0034e"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Caudatoputamen</span></p></td>
<td class="cl-7ca0034f"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.78</span></p></td>
<td class="cl-7ca00358"><p class="cl-7c9f5340"><span class="cl-7c9993c4">6.7005990</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00345"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Id4</span></p></td>
<td class="cl-7ca0034e"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Caudatoputamen</span></p></td>
<td class="cl-7ca0034f"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.76</span></p></td>
<td class="cl-7ca00358"><p class="cl-7c9f5340"><span class="cl-7c9993c4">1.8237540</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-7ca00396"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Pcp4l1</span></p></td>
<td class="cl-7ca0039e"><p class="cl-7c9f5340"><span class="cl-7c9993c4">Caudatoputamen</span></p></td>
<td class="cl-7ca0039f"><p class="cl-7c9f5340"><span class="cl-7c9993c4">0.71</span></p></td>
<td class="cl-7ca003a8"><p class="cl-7c9f5340"><span class="cl-7c9993c4">1.6998321</span></p></td>
</tr>
</tbody>
<tfoot><tr style="overflow-wrap:break-word;"><td colspan="4" class="cl-7ca003a9"><p class="cl-7c9f534a"><span class="cl-7c9993c4">*AUC Power: 1 = perfect classifier, 0.5 = random.</span></p></td></tr></tfoot>
</table>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb109"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># view(brain@meta.data)</span></span>
<span></span>
<span><span class="co"># Save Seurat object 'with markers'brain'</span></span>
<span><span class="co"># saveRDS(brain, file = "~/path/to/your/data/Spatial_transcript_Visium_mouse_brain_regions.rds")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="converting-seurat-object-brain-into-h5ad-format" class="level2" data-number="10"><h2 data-number="10" class="anchored" data-anchor-id="converting-seurat-object-brain-into-h5ad-format">
<span class="header-section-number">10</span> <em>Converting Seurat object “brain” into H5AD format</em>
</h2>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb110"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># # Activate the package</span></span>
<span><span class="co"># library(SeuratDisk)</span></span>
<span><span class="co"># library(reticulate)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># library(Seurat) # &lt;-- Critical! This provides the method for Seurat → SingleCellExperiment</span></span>
<span><span class="co"># library(SingleCellExperiment)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # Start conversion to H5AD:</span></span>
<span><span class="co"># # 1. Remove problematic layers (scale.data causes HDF5 issues)</span></span>
<span><span class="co"># pbmc_postinfusion_idecel[["RNA"]] &lt;- subset(</span></span>
<span><span class="co">#   pbmc_postinfusion_idecel[["RNA"]],</span></span>
<span><span class="co">#   layers = c("counts", "data")  # Keep only counts/data</span></span>
<span><span class="co"># )</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # 2. Convert to SingleCellExperiment and to H5AD</span></span>
<span><span class="co"># sce &lt;- as.SingleCellExperiment(pbmc_postinfusion_idecel)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># library(zellkonverter)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># writeH5AD(sce, file = "./Data_Grieb_CAR_Zenodo/pbmc_postinfusion_idecel.h5ad")</span></span>
<span></span>
<span><span class="va">brain</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>An object of class Seurat
49131 features across 2698 samples within 2 assays
Active assay: SCT (18078 features, 3000 variable features)
 3 layers present: counts, data, scale.data
 1 other assay present: Spatial
 2 dimensional reductions calculated: pca, umap
 1 spatial field of view present: slice1</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb112"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">########### METHOD 1 #############</span></span>
<span><span class="co"># Load required packages</span></span>
<span><span class="co"># library(Seurat)</span></span>
<span><span class="co"># library(SingleCellExperiment)</span></span>
<span><span class="co"># library(zellkonverter)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # Remove problematic layers (scale.data can cause HDF5 issues)</span></span>
<span><span class="co"># brain[["SCT"]] &lt;- subset(</span></span>
<span><span class="co">#   brain[["SCT"]], </span></span>
<span><span class="co">#   layers = c("counts", "data")  # Keep only essential layers</span></span>
<span><span class="co"># )</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # Convert to SingleCellExperiment</span></span>
<span><span class="co"># sce &lt;- as.SingleCellExperiment(brain)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # Save to H5AD (AnnData format)</span></span>
<span><span class="co"># writeH5AD(</span></span>
<span><span class="co">#   sce, </span></span>
<span><span class="co">#   file = "~path/to/your/data/Spatial_Data_MouseBrainCoronal/brain_metadata.h5ad",</span></span>
<span><span class="co">#   verbose = TRUE  # Monitor conversion progress</span></span>
<span><span class="co"># )</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># ########### METHOD 2 #############</span></span>
<span><span class="co"># # Load required packages</span></span>
<span><span class="co"># library(Seurat)</span></span>
<span><span class="co"># library(SingleCellExperiment)</span></span>
<span><span class="co"># library(zellkonverter)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # Clean up layers properly</span></span>
<span><span class="co"># brain &lt;- DietSeurat(</span></span>
<span><span class="co">#   brain,</span></span>
<span><span class="co">#   layers = c("counts", "data"),</span></span>
<span><span class="co">#   assays = c("SCT", "Spatial")</span></span>
<span><span class="co"># )</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # Convert to SingleCellExperiment</span></span>
<span><span class="co"># sce &lt;- as.SingleCellExperiment(brain)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # Add spatial coordinates</span></span>
<span><span class="co"># spatial_coords &lt;- GetTissueCoordinates(brain)</span></span>
<span><span class="co"># colData(sce)$imagecol &lt;- spatial_coords$imagecol</span></span>
<span><span class="co"># colData(sce)$imagerow &lt;- spatial_coords$imagerow</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # Save to H5AD</span></span>
<span><span class="co"># writeH5AD(sce, file = "~/path/to/your/data/brain_metadata_corrected.h5ad", verbose = TRUE)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # Check metadata transfer</span></span>
<span><span class="co"># head(colData(sce))  # Should match brain@meta.data</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # Check spatial coordinates</span></span>
<span><span class="co"># head(colData(sce)$imagecol)</span></span>
<span></span>
<span></span>
<span><span class="co">########### METHOD 3: THE CORRECT WAY TO SAVE SPATIAL METADATA AS H5AD FORMAT ##############</span></span>
<span><span class="co">########### THIS METHOD INCLUDES THE INCORPORATION OF SPATIAL COORDINATES ########### </span></span>
<span><span class="co"># Load required packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drighelli/SpatialExperiment">SpatialExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/theislab/zellkonverter">zellkonverter</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Extract spatial coordinates</span></span>
<span><span class="co"># spatial_coords &lt;- GetTissueCoordinates(brain)  # Columns are x, y, cell</span></span>
<span></span>
<span><span class="co"># Convert to SpatialExperiment</span></span>
<span><span class="co"># spe &lt;- SpatialExperiment(</span></span>
<span><span class="co">#   assays = list(</span></span>
<span><span class="co">#     counts = LayerData(brain, assay = "SCT", layer = "counts"),  # SCT counts</span></span>
<span><span class="co">#     logcounts = LayerData(brain, assay = "SCT", layer = "data")   # SCT normalized data</span></span>
<span><span class="co">#   ),</span></span>
<span><span class="co">#   colData = brain@meta.data,  # Transfer metadata</span></span>
<span><span class="co">#   spatialCoords = as.matrix(spatial_coords[, c("x", "y")])  # Use x and y columns</span></span>
<span><span class="co"># )</span></span>
<span></span>
<span><span class="co"># ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span></span>
<span><span class="co"># KEY FIX: Add coordinates to reducedDims for proper H5AD mapping</span></span>
<span><span class="co"># ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span></span>
<span><span class="co"># AnnData expects spatial coordinates in .obsm['spatial'], which maps to reducedDims in SingleCellExperiment</span></span>
<span><span class="co"># reducedDim(spe, "spatial") &lt;- as.matrix(spatial_coords[, c("x", "y")])</span></span>
<span></span>
<span><span class="co"># Verify spatial coordinates</span></span>
<span><span class="co"># head(spatialCoords(spe))  # Check SpatialExperiment slot</span></span>
<span><span class="co"># Should show x and y</span></span>
<span><span class="co"># Output:</span></span>
<span><span class="co">#            x     y</span></span>
<span><span class="co"># AAACAAGTATCTCCCA-1 7237  8230</span></span>
<span><span class="co"># AAACAATCTACTAGCA-1 1611  4170</span></span>
<span><span class="co"># ...  </span></span>
<span><span class="co"># head(reducedDim(spe, "spatial"))  # Check reducedDims for H5AD</span></span>
<span></span>
<span><span class="co"># Save to H5AD</span></span>
<span><span class="co"># writeH5AD(spe, file = "~/path/to/your/data/brain_metadata_corrected3_withCoordinatesRegions.h5ad", verbose = TRUE)</span></span>
<span></span>
<span></span>
<span><span class="co"># Explanation:</span></span>
<span><span class="co"># reducedDim(spe, "spatial"):</span></span>
<span><span class="co"># Explicitly stores coordinates in the reducedDims slot of the SpatialExperiment object.</span></span>
<span><span class="co"># When writing to H5AD, zellkonverter maps reducedDims to .obsm in AnnData.</span></span>
<span><span class="co"># This ensures coordinates appear in adata.obsm['spatial'] in Python.</span></span>
<span><span class="co"># Why This Works:</span></span>
<span><span class="co"># .obsm in AnnData is designed for multi-dimensional annotations (e.g., spatial coordinates).</span></span>
<span><span class="co"># The reducedDims slot in SingleCellExperiment/SpatialExperiment directly translates to .obsm in H5AD.</span></span>
<span></span>
<span><span class="co"># VERIFICATION OF H5AD: Read H5AD back into R (optional)</span></span>
<span><span class="co"># library(zellkonverter)</span></span>
<span><span class="co"># spe_check &lt;- readH5AD("~/path/to/your/data/brain_metadata_corrected3_withCoordinatesRegions.h5ad")</span></span>
<span></span>
<span><span class="co"># Confirm coordinates are in reducedDims</span></span>
<span><span class="co"># head(reducedDim(spe_check, "spatial"))  # Should match original x/y</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<!-- -->

</section></main><!-- /main column --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button,
        { trigger: "manual",
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        }
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config);
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined;
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            }
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          }
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb113" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Spatial Tutorial: Mouse Brain Section (Coronal) – 10x Genomics"</span></span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Frano Malinarich G."</span></span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a><span class="an">Company:</span><span class="co"> "... GmbH"</span></span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "`r Sys.Date()`"</span></span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-depth: 4</span></span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a><span class="co">    number-sections: true</span></span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-width: 8</span></span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-height: 6</span></span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb113-14"><a href="#cb113-14" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb113-15"><a href="#cb113-15" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true</span></span>
<span id="cb113-16"><a href="#cb113-16" aria-hidden="true" tabindex="-1"></a><span class="co">    code-download: true</span></span>
<span id="cb113-17"><a href="#cb113-17" aria-hidden="true" tabindex="-1"></a><span class="co">    css: </span></span>
<span id="cb113-18"><a href="#cb113-18" aria-hidden="true" tabindex="-1"></a><span class="co">      - styles.css</span></span>
<span id="cb113-19"><a href="#cb113-19" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb113-20"><a href="#cb113-20" aria-hidden="true" tabindex="-1"></a><span class="an">resources:</span></span>
<span id="cb113-21"><a href="#cb113-21" aria-hidden="true" tabindex="-1"></a><span class="co">  - logo.png</span></span>
<span id="cb113-22"><a href="#cb113-22" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb113-23"><a href="#cb113-23" aria-hidden="true" tabindex="-1"></a><span class="co">  warning: false</span></span>
<span id="cb113-24"><a href="#cb113-24" aria-hidden="true" tabindex="-1"></a><span class="co">  message: false</span></span>
<span id="cb113-25"><a href="#cb113-25" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb113-26"><a href="#cb113-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-27"><a href="#cb113-27" aria-hidden="true" tabindex="-1"></a><span class="fu">## Spatial transcriptomics</span></span>
<span id="cb113-28"><a href="#cb113-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-29"><a href="#cb113-29" aria-hidden="true" tabindex="-1"></a>**Tutorials:**</span>
<span id="cb113-30"><a href="#cb113-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-31"><a href="#cb113-31" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://satijalab.org/seurat/articles/spatial_vignette&gt;</span></span>
<span id="cb113-32"><a href="#cb113-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-33"><a href="#cb113-33" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://www.youtube.com/watch?v=L_7VdCeJ4Z8&amp;list=PLOLdjuxsfI4MXCyKZDY0jKU-K6Uo4o1_d&amp;index=2.&gt;</span></span>
<span id="cb113-34"><a href="#cb113-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-35"><a href="#cb113-35" aria-hidden="true" tabindex="-1"></a>**Data source: 10XGenomics, Mouse brain section (coronal)** <span class="ot">&lt;https://www.10xgenomics.com/datasets/mouse-brain-section-coronal-1-standard-1-0-0&gt;</span></span>
<span id="cb113-36"><a href="#cb113-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-37"><a href="#cb113-37" aria-hidden="true" tabindex="-1"></a><span class="fu">## *Loading / Installing package*</span></span>
<span id="cb113-38"><a href="#cb113-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-41"><a href="#cb113-41" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb113-42"><a href="#cb113-42" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false   # Suppresses "Loading package" messages</span></span>
<span id="cb113-43"><a href="#cb113-43" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false   # Suppresses version mismatch warnings</span></span>
<span id="cb113-44"><a href="#cb113-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-45"><a href="#cb113-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Installing necessary packages</span></span>
<span id="cb113-46"><a href="#cb113-46" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("extrafont")</span></span>
<span id="cb113-47"><a href="#cb113-47" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("SpatialExperiment")</span></span>
<span id="cb113-48"><a href="#cb113-48" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages('Rfast2')</span></span>
<span id="cb113-49"><a href="#cb113-49" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("Boruta")</span></span>
<span id="cb113-50"><a href="#cb113-50" aria-hidden="true" tabindex="-1"></a><span class="co"># Install Bioconductor manager (if not already installed)</span></span>
<span id="cb113-51"><a href="#cb113-51" aria-hidden="true" tabindex="-1"></a><span class="co"># Install mouse annotation database</span></span>
<span id="cb113-52"><a href="#cb113-52" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("org.Mm.eg.db")</span></span>
<span id="cb113-53"><a href="#cb113-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-54"><a href="#cb113-54" aria-hidden="true" tabindex="-1"></a><span class="co"># Load packages</span></span>
<span id="cb113-55"><a href="#cb113-55" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(conflicted)</span>
<span id="cb113-56"><a href="#cb113-56" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb113-57"><a href="#cb113-57" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb113-58"><a href="#cb113-58" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb113-59"><a href="#cb113-59" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb113-60"><a href="#cb113-60" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(flextable)</span>
<span id="cb113-61"><a href="#cb113-61" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(extrafont)</span>
<span id="cb113-62"><a href="#cb113-62" aria-hidden="true" tabindex="-1"></a><span class="co"># font_import()</span></span>
<span id="cb113-63"><a href="#cb113-63" aria-hidden="true" tabindex="-1"></a><span class="co"># loadfonts(device = "all")</span></span>
<span id="cb113-64"><a href="#cb113-64" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(officer)</span>
<span id="cb113-65"><a href="#cb113-65" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot) <span class="co"># For plot arrangement</span></span>
<span id="cb113-66"><a href="#cb113-66" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gridExtra) <span class="co"># For grid-based layout</span></span>
<span id="cb113-67"><a href="#cb113-67" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(grid)</span>
<span id="cb113-68"><a href="#cb113-68" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Rfast2)</span>
<span id="cb113-69"><a href="#cb113-69" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Mm.eg.db)  <span class="co"># For Mus musculus (mouse)</span></span>
<span id="cb113-70"><a href="#cb113-70" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb113-71"><a href="#cb113-71" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleCellExperiment)</span>
<span id="cb113-72"><a href="#cb113-72" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SpatialExperiment)</span>
<span id="cb113-73"><a href="#cb113-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-74"><a href="#cb113-74" aria-hidden="true" tabindex="-1"></a><span class="co"># Resolve conflicts</span></span>
<span id="cb113-75"><a href="#cb113-75" aria-hidden="true" tabindex="-1"></a><span class="fu">conflict_prefer</span>(<span class="st">"filter"</span>, <span class="st">"dplyr"</span>)</span>
<span id="cb113-76"><a href="#cb113-76" aria-hidden="true" tabindex="-1"></a><span class="fu">conflict_prefer</span>(<span class="st">"combine"</span>, <span class="st">"dplyr"</span>)</span>
<span id="cb113-77"><a href="#cb113-77" aria-hidden="true" tabindex="-1"></a><span class="fu">conflict_prefer</span>(<span class="st">"setdiff"</span>, <span class="st">"base"</span>)  <span class="co"># Specify base::setdiff to avoid errors</span></span>
<span id="cb113-78"><a href="#cb113-78" aria-hidden="true" tabindex="-1"></a><span class="fu">conflict_prefer</span>(<span class="st">"rename"</span>, <span class="st">"dplyr"</span>)</span>
<span id="cb113-79"><a href="#cb113-79" aria-hidden="true" tabindex="-1"></a><span class="fu">conflict_prefer</span>(<span class="st">"select"</span>, <span class="st">"dplyr"</span>)</span>
<span id="cb113-80"><a href="#cb113-80" aria-hidden="true" tabindex="-1"></a><span class="fu">conflict_prefer</span>(<span class="st">"width"</span>, <span class="st">"flextable"</span>)</span>
<span id="cb113-81"><a href="#cb113-81" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb113-82"><a href="#cb113-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-83"><a href="#cb113-83" aria-hidden="true" tabindex="-1"></a><span class="fu">## *Load spatial RNA sequencing data*</span></span>
<span id="cb113-84"><a href="#cb113-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-85"><a href="#cb113-85" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, echo=FALSE}</span></span>
<span id="cb113-86"><a href="#cb113-86" aria-hidden="true" tabindex="-1"></a><span class="in"># Load data</span></span>
<span id="cb113-87"><a href="#cb113-87" aria-hidden="true" tabindex="-1"></a><span class="in">data_dir = "~/path/to/your/data/"</span></span>
<span id="cb113-88"><a href="#cb113-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-89"><a href="#cb113-89" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb113-90"><a href="#cb113-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-91"><a href="#cb113-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-94"><a href="#cb113-94" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb113-95"><a href="#cb113-95" aria-hidden="true" tabindex="-1"></a><span class="co"># Load data</span></span>
<span id="cb113-96"><a href="#cb113-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-97"><a href="#cb113-97" aria-hidden="true" tabindex="-1"></a><span class="co"># data_dir = "~/path/to/your/data/"</span></span>
<span id="cb113-98"><a href="#cb113-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-99"><a href="#cb113-99" aria-hidden="true" tabindex="-1"></a>brain <span class="ot">&lt;-</span> <span class="fu">Load10X_Spatial</span>(<span class="at">data.dir =</span> data_dir,</span>
<span id="cb113-100"><a href="#cb113-100" aria-hidden="true" tabindex="-1"></a>                         <span class="at">filename =</span> <span class="st">"V1_Adult_Mouse_Brain_filtered_feature_bc_matrix.h5"</span>,</span>
<span id="cb113-101"><a href="#cb113-101" aria-hidden="true" tabindex="-1"></a>                         <span class="at">assay =</span> <span class="st">"Spatial"</span>,</span>
<span id="cb113-102"><a href="#cb113-102" aria-hidden="true" tabindex="-1"></a>                         <span class="at">slice =</span> <span class="st">"slice1"</span>,</span>
<span id="cb113-103"><a href="#cb113-103" aria-hidden="true" tabindex="-1"></a>                         <span class="at">filter.matrix =</span> <span class="cn">TRUE</span>,</span>
<span id="cb113-104"><a href="#cb113-104" aria-hidden="true" tabindex="-1"></a>                         <span class="at">to.upper =</span> <span class="cn">FALSE</span>,</span>
<span id="cb113-105"><a href="#cb113-105" aria-hidden="true" tabindex="-1"></a>                         <span class="at">image =</span> <span class="cn">NULL</span>)</span>
<span id="cb113-106"><a href="#cb113-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-107"><a href="#cb113-107" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt; brain</span></span>
<span id="cb113-108"><a href="#cb113-108" aria-hidden="true" tabindex="-1"></a><span class="co"># An object of class Seurat </span></span>
<span id="cb113-109"><a href="#cb113-109" aria-hidden="true" tabindex="-1"></a><span class="co"># 31053 features across 2698 samples within 1 assay </span></span>
<span id="cb113-110"><a href="#cb113-110" aria-hidden="true" tabindex="-1"></a><span class="co"># Active assay: Spatial (31053 features, 0 variable features)</span></span>
<span id="cb113-111"><a href="#cb113-111" aria-hidden="true" tabindex="-1"></a><span class="co">#  1 layer present: counts</span></span>
<span id="cb113-112"><a href="#cb113-112" aria-hidden="true" tabindex="-1"></a><span class="co">#  1 spatial field of view present: slice1</span></span>
<span id="cb113-113"><a href="#cb113-113" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb113-114"><a href="#cb113-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-115"><a href="#cb113-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-116"><a href="#cb113-116" aria-hidden="true" tabindex="-1"></a><span class="fu">## *QC analysis*</span></span>
<span id="cb113-117"><a href="#cb113-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-120"><a href="#cb113-120" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb113-121"><a href="#cb113-121" aria-hidden="true" tabindex="-1"></a><span class="co"># View first few rows of metadata</span></span>
<span id="cb113-122"><a href="#cb113-122" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(brain<span class="sc">@</span>meta.data)</span>
<span id="cb113-123"><a href="#cb113-123" aria-hidden="true" tabindex="-1"></a><span class="co"># view(brain@meta.data)</span></span>
<span id="cb113-124"><a href="#cb113-124" aria-hidden="true" tabindex="-1"></a><span class="co"># View(brain) #'View' and 'view' cannot be rendered in HTML so comment out, replace with 'head()'</span></span>
<span id="cb113-125"><a href="#cb113-125" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(brain<span class="sc">@</span>meta.data) </span>
<span id="cb113-126"><a href="#cb113-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-127"><a href="#cb113-127" aria-hidden="true" tabindex="-1"></a><span class="co"># Mitochondrial Gene Percentage incorporated into brain@meta.data</span></span>
<span id="cb113-128"><a href="#cb113-128" aria-hidden="true" tabindex="-1"></a>brain[[<span class="st">"percent.mt"</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(brain, <span class="at">pattern =</span> <span class="st">"^mt-"</span>) </span>
<span id="cb113-129"><a href="#cb113-129" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(brain<span class="sc">@</span>meta.data)</span>
<span id="cb113-130"><a href="#cb113-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-131"><a href="#cb113-131" aria-hidden="true" tabindex="-1"></a><span class="co"># Violin plot for spatial counts</span></span>
<span id="cb113-132"><a href="#cb113-132" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(brain, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"nCount_Spatial"</span>, <span class="st">"nFeature_Spatial"</span>, <span class="st">"percent.mt"</span>), <span class="at">pt.size =</span> <span class="fl">0.1</span>)   <span class="sc">+</span> <span class="fu">NoLegend</span>()</span>
<span id="cb113-133"><a href="#cb113-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-134"><a href="#cb113-134" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter low-quality spots</span></span>
<span id="cb113-135"><a href="#cb113-135" aria-hidden="true" tabindex="-1"></a><span class="co"># brain &lt;- subset(</span></span>
<span id="cb113-136"><a href="#cb113-136" aria-hidden="true" tabindex="-1"></a><span class="co">#   brain,</span></span>
<span id="cb113-137"><a href="#cb113-137" aria-hidden="true" tabindex="-1"></a><span class="co">#   nCount_Spatial &gt; 500 &amp; </span></span>
<span id="cb113-138"><a href="#cb113-138" aria-hidden="true" tabindex="-1"></a><span class="co">#   nFeature_Spatial &gt; 300 &amp; </span></span>
<span id="cb113-139"><a href="#cb113-139" aria-hidden="true" tabindex="-1"></a><span class="co">#   percent.mt &lt; 20</span></span>
<span id="cb113-140"><a href="#cb113-140" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb113-141"><a href="#cb113-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-142"><a href="#cb113-142" aria-hidden="true" tabindex="-1"></a><span class="co"># Ribosomal Protein Gene Percentage </span></span>
<span id="cb113-143"><a href="#cb113-143" aria-hidden="true" tabindex="-1"></a>brain[[<span class="st">"percent.rb"</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(brain, <span class="at">pattern =</span> <span class="st">"^Rp[sl]"</span>)  <span class="co"># Elevated ribosomal RNA can indicate residual cytoplasmic RNA or stress.</span></span>
<span id="cb113-144"><a href="#cb113-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-145"><a href="#cb113-145" aria-hidden="true" tabindex="-1"></a><span class="co"># High percent.rb + low nCount_Spatial → Technical artifact (filter)</span></span>
<span id="cb113-146"><a href="#cb113-146" aria-hidden="true" tabindex="-1"></a><span class="co"># High percent.rb + high percent.mt → Cell stress (consider filtering).</span></span>
<span id="cb113-147"><a href="#cb113-147" aria-hidden="true" tabindex="-1"></a><span class="fu">FeatureScatter</span>(brain, <span class="at">feature1 =</span> <span class="st">"percent.rb"</span>, <span class="at">feature2 =</span> <span class="st">"nCount_Spatial"</span>) <span class="sc">+</span></span>
<span id="cb113-148"><a href="#cb113-148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FeatureScatter</span>(brain, <span class="at">feature1 =</span> <span class="st">"percent.rb"</span>, <span class="at">feature2 =</span> <span class="st">"percent.mt"</span>) </span>
<span id="cb113-149"><a href="#cb113-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-150"><a href="#cb113-150" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify spots with high ribosomal gene expression</span></span>
<span id="cb113-151"><a href="#cb113-151" aria-hidden="true" tabindex="-1"></a><span class="fu">WhichCells</span>(brain, <span class="at">expression =</span> percent.rb <span class="sc">&gt;</span> <span class="dv">19</span>) <span class="co"># only one spot "CTCTGCAGGCATTCTT-1"</span></span>
<span id="cb113-152"><a href="#cb113-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-153"><a href="#cb113-153" aria-hidden="true" tabindex="-1"></a><span class="co"># Summary statistics for key QC metrics</span></span>
<span id="cb113-154"><a href="#cb113-154" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(brain<span class="sc">$</span>nCount_Spatial)  <span class="co"># Confirm min &gt;500</span></span>
<span id="cb113-155"><a href="#cb113-155" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(brain<span class="sc">$</span>nFeature_Spatial)  <span class="co"># Confirm min &gt;300</span></span>
<span id="cb113-156"><a href="#cb113-156" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(brain<span class="sc">$</span>percent.mt)  <span class="co"># Confirm max &lt;20%</span></span>
<span id="cb113-157"><a href="#cb113-157" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(brain<span class="sc">$</span>percent.rb) <span class="co"># Confim &gt; 20</span></span>
<span id="cb113-158"><a href="#cb113-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-159"><a href="#cb113-159" aria-hidden="true" tabindex="-1"></a><span class="co"># The spot with percent.rb &gt; 20 does not show low nCount_Spatial or high percent.mt.</span></span>
<span id="cb113-160"><a href="#cb113-160" aria-hidden="true" tabindex="-1"></a><span class="co"># Therefore, this elevated ribosomal signal may not indicate a technical artifact or stress.</span></span>
<span id="cb113-161"><a href="#cb113-161" aria-hidden="true" tabindex="-1"></a><span class="co"># Filtering may be skipped in this case.</span></span>
<span id="cb113-162"><a href="#cb113-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-163"><a href="#cb113-163" aria-hidden="true" tabindex="-1"></a><span class="co"># Violin plot for spatial counts +  Ribosomal Protein Gene Percentage (percent.rb)</span></span>
<span id="cb113-164"><a href="#cb113-164" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(brain, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"nCount_Spatial"</span>, <span class="st">"nFeature_Spatial"</span>, <span class="st">"percent.mt"</span>, <span class="st">"percent.rb"</span>), <span class="at">pt.size =</span> <span class="fl">0.1</span>)   <span class="sc">+</span> <span class="fu">NoLegend</span>()</span>
<span id="cb113-165"><a href="#cb113-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-166"><a href="#cb113-166" aria-hidden="true" tabindex="-1"></a><span class="co"># Spatial visualization of percent.rb across the brain section</span></span>
<span id="cb113-167"><a href="#cb113-167" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">SpatialFeaturePlot</span>(brain, <span class="at">features =</span> </span>
<span id="cb113-168"><a href="#cb113-168" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"percent.rb"</span>, <span class="at">alpha =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="dv">2</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>)</span>
<span id="cb113-169"><a href="#cb113-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-170"><a href="#cb113-170" aria-hidden="true" tabindex="-1"></a><span class="co"># Spatial visualization of nCount_Spatial across the brain section</span></span>
<span id="cb113-171"><a href="#cb113-171" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">SpatialFeaturePlot</span>(brain, <span class="at">features =</span> </span>
<span id="cb113-172"><a href="#cb113-172" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"nCount_Spatial"</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>)</span>
<span id="cb113-173"><a href="#cb113-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-174"><a href="#cb113-174" aria-hidden="true" tabindex="-1"></a><span class="co"># Spatial visualization of nFeature_Spatial across the brain section</span></span>
<span id="cb113-175"><a href="#cb113-175" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">SpatialFeaturePlot</span>(brain, <span class="at">features =</span> </span>
<span id="cb113-176"><a href="#cb113-176" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"nFeature_Spatial"</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>)</span>
<span id="cb113-177"><a href="#cb113-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-178"><a href="#cb113-178" aria-hidden="true" tabindex="-1"></a><span class="co"># Spatial visualization of nFeature_Spatial across the brain section</span></span>
<span id="cb113-179"><a href="#cb113-179" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">SpatialFeaturePlot</span>(brain, <span class="at">features =</span> </span>
<span id="cb113-180"><a href="#cb113-180" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"percent.mt"</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>)</span>
<span id="cb113-181"><a href="#cb113-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-182"><a href="#cb113-182" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p1 <span class="sc">+</span> p2 <span class="sc">+</span> p3 <span class="sc">+</span> p4)</span>
<span id="cb113-183"><a href="#cb113-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-184"><a href="#cb113-184" aria-hidden="true" tabindex="-1"></a><span class="co"># Range of counts and features</span></span>
<span id="cb113-185"><a href="#cb113-185" aria-hidden="true" tabindex="-1"></a><span class="fu">range</span>(brain<span class="sc">@</span>meta.data<span class="sc">$</span>nCount_Spatial) <span class="co"># 558 149086</span></span>
<span id="cb113-186"><a href="#cb113-186" aria-hidden="true" tabindex="-1"></a><span class="fu">range</span>(brain<span class="sc">@</span>meta.data<span class="sc">$</span>nFeature_Spatial) <span class="co"># 403 10691</span></span>
<span id="cb113-187"><a href="#cb113-187" aria-hidden="true" tabindex="-1"></a><span class="fu">range</span>(brain<span class="sc">@</span>meta.data<span class="sc">$</span>percent.mt) <span class="co"># 5.726924 44.657763</span></span>
<span id="cb113-188"><a href="#cb113-188" aria-hidden="true" tabindex="-1"></a><span class="fu">range</span>(brain<span class="sc">@</span>meta.data<span class="sc">$</span>percent.rb) <span class="co"># 3.518981 23.129093</span></span>
<span id="cb113-189"><a href="#cb113-189" aria-hidden="true" tabindex="-1"></a><span class="co"># normally, the brain has a high % of ribosome</span></span>
<span id="cb113-190"><a href="#cb113-190" aria-hidden="true" tabindex="-1"></a><span class="co"># 23.1%: Moderately high but plausible for brain regions with active protein synthesis (e.g., hippocampus).</span></span>
<span id="cb113-191"><a href="#cb113-191" aria-hidden="true" tabindex="-1"></a><span class="co"># Do NOT filter solely on ribosomal % unless: Spots are outliers (e.g., &gt;25%) AND correlate with low RNA quality.</span></span>
<span id="cb113-192"><a href="#cb113-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-193"><a href="#cb113-193" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb113-194"><a href="#cb113-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-195"><a href="#cb113-195" aria-hidden="true" tabindex="-1"></a><span class="fu">## *Data pre-processing: normalization, scaling and FindSpatiallyVariableFeatures*</span></span>
<span id="cb113-196"><a href="#cb113-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-199"><a href="#cb113-199" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb113-200"><a href="#cb113-200" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(brain<span class="sc">@</span>meta.data,<span class="dv">3</span>)</span>
<span id="cb113-201"><a href="#cb113-201" aria-hidden="true" tabindex="-1"></a>brain<span class="sc">@</span>commands<span class="sc">$</span>SCTransform.Spatial</span>
<span id="cb113-202"><a href="#cb113-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-203"><a href="#cb113-203" aria-hidden="true" tabindex="-1"></a><span class="do">#########</span></span>
<span id="cb113-204"><a href="#cb113-204" aria-hidden="true" tabindex="-1"></a><span class="do">#########</span></span>
<span id="cb113-205"><a href="#cb113-205" aria-hidden="true" tabindex="-1"></a><span class="do">#########</span></span>
<span id="cb113-206"><a href="#cb113-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-207"><a href="#cb113-207" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. Preprocessing</span></span>
<span id="cb113-208"><a href="#cb113-208" aria-hidden="true" tabindex="-1"></a>brain <span class="ot">&lt;-</span> <span class="fu">SCTransform</span>(brain, <span class="at">assay =</span> <span class="st">"Spatial"</span>, <span class="at">new.assay.name =</span> <span class="st">"SCT"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb113-209"><a href="#cb113-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-210"><a href="#cb113-210" aria-hidden="true" tabindex="-1"></a>Seurat<span class="sc">::</span><span class="fu">Assays</span>(brain) <span class="co"># expected output: "Spatial" "SCT"</span></span>
<span id="cb113-211"><a href="#cb113-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-212"><a href="#cb113-212" aria-hidden="true" tabindex="-1"></a><span class="co"># The second step FindSpatiallyVariableFeatures() is OPTIONAL!</span></span>
<span id="cb113-213"><a href="#cb113-213" aria-hidden="true" tabindex="-1"></a><span class="co"># This step: </span></span>
<span id="cb113-214"><a href="#cb113-214" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. Used to find genes with spatial expression gradients (e.g., Moran's I).</span></span>
<span id="cb113-215"><a href="#cb113-215" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Not required for standard clustering/visualization (UMAP, spatial plots).</span></span>
<span id="cb113-216"><a href="#cb113-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-217"><a href="#cb113-217" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Set SCT as active assay and find spatially variable features</span></span>
<span id="cb113-218"><a href="#cb113-218" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(brain) <span class="ot">&lt;-</span> <span class="st">"SCT"</span></span>
<span id="cb113-219"><a href="#cb113-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-220"><a href="#cb113-220" aria-hidden="true" tabindex="-1"></a>brain <span class="ot">&lt;-</span> <span class="fu">FindSpatiallyVariableFeatures</span>(</span>
<span id="cb113-221"><a href="#cb113-221" aria-hidden="true" tabindex="-1"></a>  brain,</span>
<span id="cb113-222"><a href="#cb113-222" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">"SCT"</span>,</span>
<span id="cb113-223"><a href="#cb113-223" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="fu">VariableFeatures</span>(brain), <span class="co"># this argument is important!</span></span>
<span id="cb113-224"><a href="#cb113-224" aria-hidden="true" tabindex="-1"></a>  <span class="at">selection.method =</span> <span class="st">"moransi"</span>,</span>
<span id="cb113-225"><a href="#cb113-225" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span></span>
<span id="cb113-226"><a href="#cb113-226" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-227"><a href="#cb113-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-228"><a href="#cb113-228" aria-hidden="true" tabindex="-1"></a><span class="co"># Warning message:</span></span>
<span id="cb113-229"><a href="#cb113-229" aria-hidden="true" tabindex="-1"></a><span class="co"># In dist(x = pos) : NAs introduced by coercion</span></span>
<span id="cb113-230"><a href="#cb113-230" aria-hidden="true" tabindex="-1"></a><span class="co"># suggests that there are missing or non-numeric values in the spatial coordinates used for the Moran’s I computation.</span></span>
<span id="cb113-231"><a href="#cb113-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-232"><a href="#cb113-232" aria-hidden="true" tabindex="-1"></a><span class="co"># The "features = VariableFeatures(brain)" allows to get more biologically meaningful spatially variable genes.</span></span>
<span id="cb113-233"><a href="#cb113-233" aria-hidden="true" tabindex="-1"></a><span class="co"># Without "features = VariableFeatures(brain)" </span></span>
<span id="cb113-234"><a href="#cb113-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-235"><a href="#cb113-235" aria-hidden="true" tabindex="-1"></a><span class="co"># Verify the presence of NAs in 'coords'</span></span>
<span id="cb113-236"><a href="#cb113-236" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">is.na</span>(brain<span class="sc">@</span>images<span class="sc">$</span>slice1<span class="sc">$</span>centroids<span class="sc">@</span>coords)) <span class="co"># If you get 0, the coordinates are fine. There are no NAs.</span></span>
<span id="cb113-237"><a href="#cb113-237" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(brain<span class="sc">@</span>images<span class="sc">$</span>slice1<span class="sc">$</span>centroids<span class="sc">@</span>coords) <span class="co"># coordinates 'x' and 'y' are numeric "num [1:2698, 1:2] 7237 1611 8315 2927 6280 ..."</span></span>
<span id="cb113-238"><a href="#cb113-238" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(brain<span class="sc">@</span>images<span class="sc">$</span>slice1<span class="sc">$</span>centroids<span class="sc">@</span>coords)</span>
<span id="cb113-239"><a href="#cb113-239" aria-hidden="true" tabindex="-1"></a><span class="co"># brain@images$slice1$centroids@coords</span></span>
<span id="cb113-240"><a href="#cb113-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-241"><a href="#cb113-241" aria-hidden="true" tabindex="-1"></a><span class="co"># Check if Spatially Variable Features Exist</span></span>
<span id="cb113-242"><a href="#cb113-242" aria-hidden="true" tabindex="-1"></a><span class="co"># Check if Moran's I results are stored in metadata</span></span>
<span id="cb113-243"><a href="#cb113-243" aria-hidden="true" tabindex="-1"></a>svf_info <span class="ot">&lt;-</span> <span class="fu">SVFInfo</span>(brain, <span class="at">assay =</span> <span class="st">"SCT"</span>)</span>
<span id="cb113-244"><a href="#cb113-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-245"><a href="#cb113-245" aria-hidden="true" tabindex="-1"></a><span class="co"># If empty, check the assay's meta.features directly</span></span>
<span id="cb113-246"><a href="#cb113-246" aria-hidden="true" tabindex="-1"></a>moran_results <span class="ot">&lt;-</span> brain<span class="sc">@</span>assays<span class="sc">$</span>SCT<span class="sc">@</span>meta.features</span>
<span id="cb113-247"><a href="#cb113-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-248"><a href="#cb113-248" aria-hidden="true" tabindex="-1"></a><span class="co"># Look for columns like "MoransI_observed" or "moransi"</span></span>
<span id="cb113-249"><a href="#cb113-249" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">colnames</span>(moran_results))</span>
<span id="cb113-250"><a href="#cb113-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-251"><a href="#cb113-251" aria-hidden="true" tabindex="-1"></a><span class="co"># Expected results</span></span>
<span id="cb113-252"><a href="#cb113-252" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] "MoransI_observed"                "MoransI_p.value"                </span></span>
<span id="cb113-253"><a href="#cb113-253" aria-hidden="true" tabindex="-1"></a><span class="co"># [3] "moransi.spatially.variable"      "moransi.spatially.variable.rank"</span></span>
<span id="cb113-254"><a href="#cb113-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-255"><a href="#cb113-255" aria-hidden="true" tabindex="-1"></a><span class="co"># getting this output means that FindSpatiallyVariableFeatures() function worked, and results are stored in the SCT assay’s metadata.</span></span>
<span id="cb113-256"><a href="#cb113-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-257"><a href="#cb113-257" aria-hidden="true" tabindex="-1"></a><span class="co"># View(brain)</span></span>
<span id="cb113-258"><a href="#cb113-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-259"><a href="#cb113-259" aria-hidden="true" tabindex="-1"></a><span class="co"># Check How Many Genes Have Non-NA Moran’s I Values</span></span>
<span id="cb113-260"><a href="#cb113-260" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(moran_results<span class="sc">$</span>MoransI_observed)) <span class="co"># 3000</span></span>
<span id="cb113-261"><a href="#cb113-261" aria-hidden="true" tabindex="-1"></a><span class="co"># 1️⃣ There are 3000 genes with valid Moran’s I values, which is a normal number for spatially variable genes.</span></span>
<span id="cb113-262"><a href="#cb113-262" aria-hidden="true" tabindex="-1"></a><span class="co"># Why Are Many Other Genes NA? because only Highly Variable Genes Were Analyzed.</span></span>
<span id="cb113-263"><a href="#cb113-263" aria-hidden="true" tabindex="-1"></a><span class="co"># By default, FindSpatiallyVariableFeatures() only computes Moran’s I for genes in VariableFeatures(brain) (often 3000 genes).</span></span>
<span id="cb113-264"><a href="#cb113-264" aria-hidden="true" tabindex="-1"></a><span class="co"># Other genes were not tested, which is why they show NA.</span></span>
<span id="cb113-265"><a href="#cb113-265" aria-hidden="true" tabindex="-1"></a><span class="co"># 2️⃣  Genes with Low or No Expression</span></span>
<span id="cb113-266"><a href="#cb113-266" aria-hidden="true" tabindex="-1"></a><span class="co"># If a gene has low counts across spots, it gets excluded.</span></span>
<span id="cb113-267"><a href="#cb113-267" aria-hidden="true" tabindex="-1"></a><span class="co"># Moran’s I needs enough expression variation to be meaningful.</span></span>
<span id="cb113-268"><a href="#cb113-268" aria-hidden="true" tabindex="-1"></a><span class="co"># 3️⃣ Sparse or Noisy Data</span></span>
<span id="cb113-269"><a href="#cb113-269" aria-hidden="true" tabindex="-1"></a><span class="co"># Some genes might be filtered due to low-quality spots or technical artifacts.</span></span>
<span id="cb113-270"><a href="#cb113-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-271"><a href="#cb113-271" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(moran_results<span class="sc">$</span>MoransI_observed, <span class="at">breaks =</span> <span class="dv">50</span>, <span class="at">main =</span> <span class="st">"Distribution of Moran's I"</span>, <span class="at">xlab =</span> <span class="st">"Moran's I Score"</span>)</span>
<span id="cb113-272"><a href="#cb113-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-273"><a href="#cb113-273" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(moran_results<span class="sc">$</span>MoransI_observed)</span>
<span id="cb113-274"><a href="#cb113-274" aria-hidden="true" tabindex="-1"></a><span class="co"># Interpretation:</span></span>
<span id="cb113-275"><a href="#cb113-275" aria-hidden="true" tabindex="-1"></a><span class="co"># 14968 genes have NA values were ignored because they were not selected as highly variable genes (via " VariableFeatures(brain)")</span></span>
<span id="cb113-276"><a href="#cb113-276" aria-hidden="true" tabindex="-1"></a><span class="co"># 3000 genes were selected as the most spatial variable genes.</span></span>
<span id="cb113-277"><a href="#cb113-277" aria-hidden="true" tabindex="-1"></a><span class="co"># This filtering step reduces noise and focuses on biologically relevant genes. It's more meaningful.</span></span>
<span id="cb113-278"><a href="#cb113-278" aria-hidden="true" tabindex="-1"></a><span class="co"># Moran’s I values range from -0.003 to 0.685.</span></span>
<span id="cb113-279"><a href="#cb113-279" aria-hidden="true" tabindex="-1"></a><span class="co"># Median Moran’s I = 0.149 → most genes show moderate spatial autocorrelation.</span></span>
<span id="cb113-280"><a href="#cb113-280" aria-hidden="true" tabindex="-1"></a><span class="co"># Mean Moran’s I = 0.175, with a max value of 0.685 → some genes have strong spatial patterns.</span></span>
<span id="cb113-281"><a href="#cb113-281" aria-hidden="true" tabindex="-1"></a><span class="co"># 1️⃣ Your spatial analysis worked correctly → The non-NA genes have meaningful Moran’s I values.</span></span>
<span id="cb113-282"><a href="#cb113-282" aria-hidden="true" tabindex="-1"></a><span class="co"># 2️⃣ The distribution is reasonable → Many genes have moderate spatial variation, and some have strong spatial patterns.</span></span>
<span id="cb113-283"><a href="#cb113-283" aria-hidden="true" tabindex="-1"></a><span class="co"># 3️⃣ You can confidently use the top Moran’s I genes for visualization &amp; clustering.</span></span>
<span id="cb113-284"><a href="#cb113-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-285"><a href="#cb113-285" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-286"><a href="#cb113-286" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">VariableFeatures</span>(brain))</span>
<span id="cb113-287"><a href="#cb113-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-288"><a href="#cb113-288" aria-hidden="true" tabindex="-1"></a><span class="co"># How to use the Spatial Variable Genes (SVG)</span></span>
<span id="cb113-289"><a href="#cb113-289" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify Top Spatially Variable Genes</span></span>
<span id="cb113-290"><a href="#cb113-290" aria-hidden="true" tabindex="-1"></a>top_spatial_genes <span class="ot">&lt;-</span> moran_results <span class="sc">%&gt;%</span></span>
<span id="cb113-291"><a href="#cb113-291" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(moransi.spatially.variable) <span class="sc">%&gt;%</span></span>
<span id="cb113-292"><a href="#cb113-292" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(MoransI_observed)) <span class="sc">%&gt;%</span></span>
<span id="cb113-293"><a href="#cb113-293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="at">n =</span> <span class="dv">20</span>)</span>
<span id="cb113-294"><a href="#cb113-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-295"><a href="#cb113-295" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize Spatial Expression</span></span>
<span id="cb113-296"><a href="#cb113-296" aria-hidden="true" tabindex="-1"></a><span class="fu">SpatialFeaturePlot</span>(</span>
<span id="cb113-297"><a href="#cb113-297" aria-hidden="true" tabindex="-1"></a>  brain,</span>
<span id="cb113-298"><a href="#cb113-298" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="fu">rownames</span>(top_spatial_genes)[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>],  <span class="co"># Top 4 genes</span></span>
<span id="cb113-299"><a href="#cb113-299" aria-hidden="true" tabindex="-1"></a>  <span class="at">pt.size.factor =</span> <span class="fl">2.5</span>,</span>
<span id="cb113-300"><a href="#cb113-300" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="dv">2</span>)</span>
<span id="cb113-301"><a href="#cb113-301" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-302"><a href="#cb113-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-303"><a href="#cb113-303" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb113-304"><a href="#cb113-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-305"><a href="#cb113-305" aria-hidden="true" tabindex="-1"></a><span class="fu">## *Dimensional reduction, clustering and visualizations*</span></span>
<span id="cb113-306"><a href="#cb113-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-309"><a href="#cb113-309" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb113-310"><a href="#cb113-310" aria-hidden="true" tabindex="-1"></a><span class="co"># Similar to scRNAseq analysis</span></span>
<span id="cb113-311"><a href="#cb113-311" aria-hidden="true" tabindex="-1"></a>brain <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(brain, <span class="at">assay =</span> <span class="st">"SCT"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb113-312"><a href="#cb113-312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-313"><a href="#cb113-313" aria-hidden="true" tabindex="-1"></a><span class="co"># ElbowPlot to determine optimal PCs</span></span>
<span id="cb113-314"><a href="#cb113-314" aria-hidden="true" tabindex="-1"></a><span class="fu">ElbowPlot</span>(brain, <span class="at">ndims =</span> <span class="dv">50</span>) <span class="co"># compared to scRNAseq, the amount of PCs are typically 30-50. ndims --&gt; how many PCs to display in elbow plot. For spatial data (Visium): ndims = 50 is a good stating point</span></span>
<span id="cb113-315"><a href="#cb113-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-316"><a href="#cb113-316" aria-hidden="true" tabindex="-1"></a>brain <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(brain, <span class="at">reduction =</span> <span class="st">"pca"</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb113-317"><a href="#cb113-317" aria-hidden="true" tabindex="-1"></a>brain <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(brain, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb113-318"><a href="#cb113-318" aria-hidden="true" tabindex="-1"></a>brain <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(brain, <span class="at">reduction =</span> <span class="st">"pca"</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb113-319"><a href="#cb113-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-320"><a href="#cb113-320" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt; brain</span></span>
<span id="cb113-321"><a href="#cb113-321" aria-hidden="true" tabindex="-1"></a><span class="co"># An object of class Seurat </span></span>
<span id="cb113-322"><a href="#cb113-322" aria-hidden="true" tabindex="-1"></a><span class="co"># 49131 features across 2698 samples within 2 assays </span></span>
<span id="cb113-323"><a href="#cb113-323" aria-hidden="true" tabindex="-1"></a><span class="co"># Active assay: SCT (18078 features, 3000 variable features)</span></span>
<span id="cb113-324"><a href="#cb113-324" aria-hidden="true" tabindex="-1"></a><span class="co">#  3 layers present: counts, data, scale.data</span></span>
<span id="cb113-325"><a href="#cb113-325" aria-hidden="true" tabindex="-1"></a><span class="co">#  1 other assay present: Spatial</span></span>
<span id="cb113-326"><a href="#cb113-326" aria-hidden="true" tabindex="-1"></a><span class="co">#  2 dimensional reductions calculated: pca, umap</span></span>
<span id="cb113-327"><a href="#cb113-327" aria-hidden="true" tabindex="-1"></a><span class="co">#  1 spatial field of view present: slice1</span></span>
<span id="cb113-328"><a href="#cb113-328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-329"><a href="#cb113-329" aria-hidden="true" tabindex="-1"></a>p5 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(brain, </span>
<span id="cb113-330"><a href="#cb113-330" aria-hidden="true" tabindex="-1"></a>              <span class="at">reduction =</span> <span class="st">"umap"</span>, </span>
<span id="cb113-331"><a href="#cb113-331" aria-hidden="true" tabindex="-1"></a>              <span class="at">label =</span> <span class="cn">TRUE</span>, </span>
<span id="cb113-332"><a href="#cb113-332" aria-hidden="true" tabindex="-1"></a>              <span class="at">pt.size =</span> <span class="fl">2.5</span>) <span class="sc">+</span> </span>
<span id="cb113-333"><a href="#cb113-333" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"UMAP spots representation"</span>) <span class="sc">+</span></span>
<span id="cb113-334"><a href="#cb113-334" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb113-335"><a href="#cb113-335" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>, <span class="at">family =</span> <span class="st">"Arial"</span>),</span>
<span id="cb113-336"><a href="#cb113-336" aria-hidden="true" tabindex="-1"></a>    <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">family =</span> <span class="st">"Arial"</span>))</span>
<span id="cb113-337"><a href="#cb113-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-338"><a href="#cb113-338" aria-hidden="true" tabindex="-1"></a>p6 <span class="ot">&lt;-</span> <span class="fu">SpatialDimPlot</span>(</span>
<span id="cb113-339"><a href="#cb113-339" aria-hidden="true" tabindex="-1"></a>  brain,</span>
<span id="cb113-340"><a href="#cb113-340" aria-hidden="true" tabindex="-1"></a>  <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb113-341"><a href="#cb113-341" aria-hidden="true" tabindex="-1"></a>  <span class="at">label.size =</span> <span class="dv">5</span>,</span>
<span id="cb113-342"><a href="#cb113-342" aria-hidden="true" tabindex="-1"></a>  <span class="at">pt.size.factor =</span> <span class="fl">2.5</span>,</span>
<span id="cb113-343"><a href="#cb113-343" aria-hidden="true" tabindex="-1"></a>  <span class="at">image.alpha =</span> <span class="dv">0</span>  <span class="co"># Make spatial image transparent</span></span>
<span id="cb113-344"><a href="#cb113-344" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb113-345"><a href="#cb113-345" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Spatial spots representation"</span>) <span class="sc">+</span></span>
<span id="cb113-346"><a href="#cb113-346" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb113-347"><a href="#cb113-347" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>, <span class="at">family =</span> <span class="st">"sans-serif"</span>, <span class="at">face =</span><span class="st">"bold"</span>),</span>
<span id="cb113-348"><a href="#cb113-348" aria-hidden="true" tabindex="-1"></a>    <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">family =</span> <span class="st">"sans-serif"</span>),</span>
<span id="cb113-349"><a href="#cb113-349" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"grey"</span>),</span>
<span id="cb113-350"><a href="#cb113-350" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),   <span class="co"># Increase cluster number font size</span></span>
<span id="cb113-351"><a href="#cb113-351" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">"cm"</span>),</span>
<span id="cb113-352"><a href="#cb113-352" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="st">"white"</span>),  <span class="co"># Plot area</span></span>
<span id="cb113-353"><a href="#cb113-353" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="st">"white"</span>)    <span class="co"># Entire background</span></span>
<span id="cb113-354"><a href="#cb113-354" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb113-355"><a href="#cb113-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-356"><a href="#cb113-356" aria-hidden="true" tabindex="-1"></a>p5 <span class="sc">+</span> p6</span>
<span id="cb113-357"><a href="#cb113-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-358"><a href="#cb113-358" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb113-359"><a href="#cb113-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-360"><a href="#cb113-360" aria-hidden="true" tabindex="-1"></a><span class="fu">## *Summary table of spots identifiers and spot amount*</span></span>
<span id="cb113-361"><a href="#cb113-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-364"><a href="#cb113-364" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb113-365"><a href="#cb113-365" aria-hidden="true" tabindex="-1"></a><span class="co"># view(brain@meta.data)</span></span>
<span id="cb113-366"><a href="#cb113-366" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(brain<span class="sc">@</span>meta.data)</span>
<span id="cb113-367"><a href="#cb113-367" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(brain<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters)</span>
<span id="cb113-368"><a href="#cb113-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-369"><a href="#cb113-369" aria-hidden="true" tabindex="-1"></a><span class="co"># Install if needed</span></span>
<span id="cb113-370"><a href="#cb113-370" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("flextable")</span></span>
<span id="cb113-371"><a href="#cb113-371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-372"><a href="#cb113-372" aria-hidden="true" tabindex="-1"></a><span class="co"># Create frequency table of cluster counts</span></span>
<span id="cb113-373"><a href="#cb113-373" aria-hidden="true" tabindex="-1"></a>cluster_counts <span class="ot">&lt;-</span> <span class="fu">table</span>(brain<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters)</span>
<span id="cb113-374"><a href="#cb113-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-375"><a href="#cb113-375" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and calculate percentages</span></span>
<span id="cb113-376"><a href="#cb113-376" aria-hidden="true" tabindex="-1"></a>spot_table <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(cluster_counts) <span class="sc">%&gt;%</span></span>
<span id="cb113-377"><a href="#cb113-377" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Spot =</span> Var1, <span class="st">`</span><span class="at">Amount of Spots</span><span class="st">`</span> <span class="ot">=</span> Freq) <span class="sc">%&gt;%</span></span>
<span id="cb113-378"><a href="#cb113-378" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb113-379"><a href="#cb113-379" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">% from total</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">Amount of Spots</span><span class="st">`</span> <span class="sc">/</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">Amount of Spots</span><span class="st">`</span>),</span>
<span id="cb113-380"><a href="#cb113-380" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">% from total</span><span class="st">`</span> <span class="ot">=</span> scales<span class="sc">::</span><span class="fu">percent</span>(<span class="st">`</span><span class="at">% from total</span><span class="st">`</span>, <span class="at">accuracy =</span> <span class="fl">0.1</span>)</span>
<span id="cb113-381"><a href="#cb113-381" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb113-382"><a href="#cb113-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-383"><a href="#cb113-383" aria-hidden="true" tabindex="-1"></a><span class="co"># Add total row</span></span>
<span id="cb113-384"><a href="#cb113-384" aria-hidden="true" tabindex="-1"></a>total_spots <span class="ot">&lt;-</span> <span class="fu">sum</span>(spot_table<span class="sc">$</span><span class="st">`</span><span class="at">Amount of Spots</span><span class="st">`</span>)</span>
<span id="cb113-385"><a href="#cb113-385" aria-hidden="true" tabindex="-1"></a>total_row <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb113-386"><a href="#cb113-386" aria-hidden="true" tabindex="-1"></a>  <span class="at">Spot =</span> <span class="st">"Total"</span>,</span>
<span id="cb113-387"><a href="#cb113-387" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">Amount of Spots</span><span class="st">`</span> <span class="ot">=</span> total_spots,</span>
<span id="cb113-388"><a href="#cb113-388" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">% from total</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"100.0%"</span>,</span>
<span id="cb113-389"><a href="#cb113-389" aria-hidden="true" tabindex="-1"></a>  <span class="at">check.names =</span> <span class="cn">FALSE</span></span>
<span id="cb113-390"><a href="#cb113-390" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-391"><a href="#cb113-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-392"><a href="#cb113-392" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine with original data</span></span>
<span id="cb113-393"><a href="#cb113-393" aria-hidden="true" tabindex="-1"></a>spot_table <span class="ot">&lt;-</span> <span class="fu">rbind</span>(spot_table, total_row)</span>
<span id="cb113-394"><a href="#cb113-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-395"><a href="#cb113-395" aria-hidden="true" tabindex="-1"></a><span class="co"># Create flextable</span></span>
<span id="cb113-396"><a href="#cb113-396" aria-hidden="true" tabindex="-1"></a>ft <span class="ot">&lt;-</span> <span class="fu">flextable</span>(spot_table) <span class="sc">%&gt;%</span></span>
<span id="cb113-397"><a href="#cb113-397" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_vanilla</span>() <span class="sc">%&gt;%</span>  <span class="co"># Apply theme FIRST</span></span>
<span id="cb113-398"><a href="#cb113-398" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_header_labels</span>(</span>
<span id="cb113-399"><a href="#cb113-399" aria-hidden="true" tabindex="-1"></a>    <span class="at">Spot =</span> <span class="st">"Cluster identifier"</span>,</span>
<span id="cb113-400"><a href="#cb113-400" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">Amount of Spots</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Amount of Spots"</span>,</span>
<span id="cb113-401"><a href="#cb113-401" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">% from total</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"% from total"</span></span>
<span id="cb113-402"><a href="#cb113-402" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb113-403"><a href="#cb113-403" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_header_lines</span>(<span class="at">values =</span> <span class="st">"Cluster representation and Spot amount from mouse brain section (coronal)*"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-404"><a href="#cb113-404" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_footer_lines</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"* Source: https://www.10xgenomics.com/datasets/mouse-brain-section-coronal-1-standard-1-0-0"</span>, <span class="st">"*Platform: Visium Spatial 10X Genomics."</span>, <span class="st">"*Preservation method: Fresh Frozen"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb113-405"><a href="#cb113-405" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Alignment</span></span>
<span id="cb113-406"><a href="#cb113-406" aria-hidden="true" tabindex="-1"></a>  <span class="fu">align</span>(<span class="at">align =</span> <span class="st">"center"</span>, <span class="at">part =</span> <span class="st">"header"</span>) <span class="sc">%&gt;%</span>     <span class="co"># Center header (including title line)</span></span>
<span id="cb113-407"><a href="#cb113-407" aria-hidden="true" tabindex="-1"></a>  <span class="fu">align</span>(<span class="at">align =</span> <span class="st">"center"</span>, <span class="at">part =</span> <span class="st">"body"</span>) <span class="sc">%&gt;%</span>       <span class="co"># Center all body cells</span></span>
<span id="cb113-408"><a href="#cb113-408" aria-hidden="true" tabindex="-1"></a>  <span class="fu">align</span>(<span class="at">align =</span> <span class="st">"left"</span>, <span class="at">part =</span> <span class="st">"footer"</span>) <span class="sc">%&gt;%</span>       <span class="co"># Left-align footnote</span></span>
<span id="cb113-409"><a href="#cb113-409" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Total row styling</span></span>
<span id="cb113-410"><a href="#cb113-410" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bold</span>(<span class="sc">~</span> Spot <span class="sc">==</span> <span class="st">"Total"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-411"><a href="#cb113-411" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bg</span>(<span class="sc">~</span> Spot <span class="sc">==</span> <span class="st">"Total"</span>, <span class="at">bg =</span> <span class="st">"#2D2D2D"</span>, <span class="at">part =</span> <span class="st">"body"</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb113-412"><a href="#cb113-412" aria-hidden="true" tabindex="-1"></a>  <span class="fu">color</span>(<span class="sc">~</span> Spot <span class="sc">==</span> <span class="st">"Total"</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">part =</span> <span class="st">"body"</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb113-413"><a href="#cb113-413" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Borders</span></span>
<span id="cb113-414"><a href="#cb113-414" aria-hidden="true" tabindex="-1"></a>  <span class="fu">border_inner_h</span>(<span class="at">border =</span> <span class="fu">fp_border</span>(<span class="at">color =</span> <span class="st">"#D3D3D3"</span>, <span class="at">width =</span> <span class="fl">0.5</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb113-415"><a href="#cb113-415" aria-hidden="true" tabindex="-1"></a>  <span class="fu">border_outer</span>(<span class="at">border =</span> <span class="fu">fp_border</span>(<span class="at">color =</span> <span class="st">"#D3D3D3"</span>, <span class="at">width =</span> <span class="fl">0.5</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb113-416"><a href="#cb113-416" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Font</span></span>
<span id="cb113-417"><a href="#cb113-417" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fontsize</span>(<span class="at">size =</span> <span class="dv">11</span>, <span class="at">part =</span> <span class="st">"all"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-418"><a href="#cb113-418" aria-hidden="true" tabindex="-1"></a>  <span class="fu">font</span>(<span class="at">fontname =</span> <span class="st">"Arial"</span>, <span class="at">part =</span> <span class="st">"all"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb113-419"><a href="#cb113-419" aria-hidden="true" tabindex="-1"></a>  <span class="fu">autofit</span>()</span>
<span id="cb113-420"><a href="#cb113-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-421"><a href="#cb113-421" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(ft)</span>
<span id="cb113-422"><a href="#cb113-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-423"><a href="#cb113-423" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert flextable to a grob</span></span>
<span id="cb113-424"><a href="#cb113-424" aria-hidden="true" tabindex="-1"></a>ft_grob <span class="ot">&lt;-</span> flextable<span class="sc">::</span><span class="fu">gen_grob</span>(ft)</span>
<span id="cb113-425"><a href="#cb113-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-426"><a href="#cb113-426" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a vertical layout for p5 and p6</span></span>
<span id="cb113-427"><a href="#cb113-427" aria-hidden="true" tabindex="-1"></a>left_column <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(p5, p6, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb113-428"><a href="#cb113-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-429"><a href="#cb113-429" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine left_column and ft_grob horizontally</span></span>
<span id="cb113-430"><a href="#cb113-430" aria-hidden="true" tabindex="-1"></a>final_layout <span class="ot">&lt;-</span> <span class="fu">grid.arrange</span>(</span>
<span id="cb113-431"><a href="#cb113-431" aria-hidden="true" tabindex="-1"></a>  left_column, </span>
<span id="cb113-432"><a href="#cb113-432" aria-hidden="true" tabindex="-1"></a>  ft_grob,</span>
<span id="cb113-433"><a href="#cb113-433" aria-hidden="true" tabindex="-1"></a>  <span class="at">nrow =</span> <span class="dv">1</span>,</span>
<span id="cb113-434"><a href="#cb113-434" aria-hidden="true" tabindex="-1"></a>  <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1.2</span>),  <span class="co"># Adjust widths (left plots vs. table)</span></span>
<span id="cb113-435"><a href="#cb113-435" aria-hidden="true" tabindex="-1"></a>  <span class="at">padding =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">"cm"</span>)  <span class="co"># Add spacing between elements</span></span>
<span id="cb113-436"><a href="#cb113-436" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-437"><a href="#cb113-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-438"><a href="#cb113-438" aria-hidden="true" tabindex="-1"></a><span class="co"># Save</span></span>
<span id="cb113-439"><a href="#cb113-439" aria-hidden="true" tabindex="-1"></a><span class="co"># ggsave("~/path/to/your/data/Spatial_transcript_Visium_mouse_brain.png", final_layout, width = 16, height = 8, dpi = 600)</span></span>
<span id="cb113-440"><a href="#cb113-440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-441"><a href="#cb113-441" aria-hidden="true" tabindex="-1"></a><span class="co"># view(brain)</span></span>
<span id="cb113-442"><a href="#cb113-442" aria-hidden="true" tabindex="-1"></a>brain</span>
<span id="cb113-443"><a href="#cb113-443" aria-hidden="true" tabindex="-1"></a><span class="co"># View(brain)</span></span>
<span id="cb113-444"><a href="#cb113-444" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(brain<span class="sc">@</span>images<span class="sc">$</span>slice1<span class="sc">@</span>boundaries<span class="sc">$</span>centroids<span class="sc">@</span>coords)</span>
<span id="cb113-445"><a href="#cb113-445" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(brain<span class="sc">@</span>images<span class="sc">$</span>slice1<span class="sc">@</span>boundaries<span class="sc">$</span>centroids<span class="sc">@</span>cells)</span>
<span id="cb113-446"><a href="#cb113-446" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(brain<span class="sc">@</span>images<span class="sc">$</span>slice1<span class="sc">@</span>scale.factors)</span>
<span id="cb113-447"><a href="#cb113-447" aria-hidden="true" tabindex="-1"></a><span class="co"># Show only the first 6 genes (rows) and first 6 spots (columns)</span></span>
<span id="cb113-448"><a href="#cb113-448" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">as.matrix</span>(<span class="fu">head</span>(brain<span class="sc">@</span>assays<span class="sc">$</span>SCT<span class="sc">@</span>data[, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>])), <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb113-449"><a href="#cb113-449" aria-hidden="true" tabindex="-1"></a><span class="co"># head(brain@assays$SCT@data)</span></span>
<span id="cb113-450"><a href="#cb113-450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-451"><a href="#cb113-451" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb113-452"><a href="#cb113-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-453"><a href="#cb113-453" aria-hidden="true" tabindex="-1"></a><span class="fu">## *Differential Expression Gene (DEG) analysis across spatial clusters*</span></span>
<span id="cb113-454"><a href="#cb113-454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-457"><a href="#cb113-457" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb113-458"><a href="#cb113-458" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(brain) <span class="ot">&lt;-</span> <span class="st">"seurat_clusters"</span>  <span class="co"># Use your cluster column from metadata</span></span>
<span id="cb113-459"><a href="#cb113-459" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(brain<span class="sc">@</span>active.ident)  <span class="co"># Verify cluster levels (0, 1, 2, ...)</span></span>
<span id="cb113-460"><a href="#cb113-460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-461"><a href="#cb113-461" aria-hidden="true" tabindex="-1"></a><span class="co"># # Identify DEGs for all clusters (Wilcoxon test by default)</span></span>
<span id="cb113-462"><a href="#cb113-462" aria-hidden="true" tabindex="-1"></a>all_markers <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(</span>
<span id="cb113-463"><a href="#cb113-463" aria-hidden="true" tabindex="-1"></a>  brain,</span>
<span id="cb113-464"><a href="#cb113-464" aria-hidden="true" tabindex="-1"></a>  <span class="at">logfc.threshold =</span> <span class="fl">0.25</span>,       <span class="co"># Minimum log2 fold change</span></span>
<span id="cb113-465"><a href="#cb113-465" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.1</span>,                <span class="co"># Gene detected in at least 10% of cells in either cluster</span></span>
<span id="cb113-466"><a href="#cb113-466" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> <span class="cn">TRUE</span>,              <span class="co"># Report only positive markers</span></span>
<span id="cb113-467"><a href="#cb113-467" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"wilcox"</span>,          <span class="co"># Default test (alternatives: "MAST", "DESeq2", etc.)</span></span>
<span id="cb113-468"><a href="#cb113-468" aria-hidden="true" tabindex="-1"></a>  <span class="at">return.thresh =</span> <span class="fl">0.01</span>          <span class="co"># Return genes with adjusted p-value &lt; 0.01</span></span>
<span id="cb113-469"><a href="#cb113-469" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-470"><a href="#cb113-470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-471"><a href="#cb113-471" aria-hidden="true" tabindex="-1"></a><span class="co"># View top 5 markers per cluster</span></span>
<span id="cb113-472"><a href="#cb113-472" aria-hidden="true" tabindex="-1"></a>top_markers <span class="ot">&lt;-</span> all_markers <span class="sc">%&gt;%</span></span>
<span id="cb113-473"><a href="#cb113-473" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb113-474"><a href="#cb113-474" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="at">n =</span> <span class="dv">5</span>, <span class="at">order_by =</span> avg_log2FC)</span>
<span id="cb113-475"><a href="#cb113-475" aria-hidden="true" tabindex="-1"></a><span class="co"># View(top_markers) # View cannot be rendered in HTML so comment out</span></span>
<span id="cb113-476"><a href="#cb113-476" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(brain)</span>
<span id="cb113-477"><a href="#cb113-477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-478"><a href="#cb113-478" aria-hidden="true" tabindex="-1"></a><span class="co"># Comparing cluster 0 vs cluster 1</span></span>
<span id="cb113-479"><a href="#cb113-479" aria-hidden="true" tabindex="-1"></a><span class="co"># cluster0_vs_1 &lt;- FindMarkers(</span></span>
<span id="cb113-480"><a href="#cb113-480" aria-hidden="true" tabindex="-1"></a><span class="co">#   brain,</span></span>
<span id="cb113-481"><a href="#cb113-481" aria-hidden="true" tabindex="-1"></a><span class="co">#   ident.1 = 0,</span></span>
<span id="cb113-482"><a href="#cb113-482" aria-hidden="true" tabindex="-1"></a><span class="co">#   ident.2 = 1,</span></span>
<span id="cb113-483"><a href="#cb113-483" aria-hidden="true" tabindex="-1"></a><span class="co">#   logfc.threshold = 0.25,</span></span>
<span id="cb113-484"><a href="#cb113-484" aria-hidden="true" tabindex="-1"></a><span class="co">#   min.pct = 0.1</span></span>
<span id="cb113-485"><a href="#cb113-485" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb113-486"><a href="#cb113-486" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-487"><a href="#cb113-487" aria-hidden="true" tabindex="-1"></a><span class="co"># Interpreting results</span></span>
<span id="cb113-488"><a href="#cb113-488" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter significant markers (adjust p-value cutoff)</span></span>
<span id="cb113-489"><a href="#cb113-489" aria-hidden="true" tabindex="-1"></a>significant_markers <span class="ot">&lt;-</span> all_markers[all_markers<span class="sc">$</span>p_val_adj <span class="sc">&lt;</span> <span class="fl">0.01</span>, ]</span>
<span id="cb113-490"><a href="#cb113-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-491"><a href="#cb113-491" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap of top markers</span></span>
<span id="cb113-492"><a href="#cb113-492" aria-hidden="true" tabindex="-1"></a><span class="fu">DoHeatmap</span>(</span>
<span id="cb113-493"><a href="#cb113-493" aria-hidden="true" tabindex="-1"></a>  brain,</span>
<span id="cb113-494"><a href="#cb113-494" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> top_markers<span class="sc">$</span>gene,</span>
<span id="cb113-495"><a href="#cb113-495" aria-hidden="true" tabindex="-1"></a>  <span class="at">group.colors =</span> scales<span class="sc">::</span><span class="fu">hue_pal</span>()(<span class="fu">length</span>(<span class="fu">levels</span>(brain))),  <span class="co"># Cluster colors</span></span>
<span id="cb113-496"><a href="#cb113-496" aria-hidden="true" tabindex="-1"></a>  <span class="at">size =</span> <span class="dv">3</span></span>
<span id="cb113-497"><a href="#cb113-497" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>))</span>
<span id="cb113-498"><a href="#cb113-498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-499"><a href="#cb113-499" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize Marker Expression</span></span>
<span id="cb113-500"><a href="#cb113-500" aria-hidden="true" tabindex="-1"></a><span class="co"># Feature plot for a top gene (e.g., "Gfap")</span></span>
<span id="cb113-501"><a href="#cb113-501" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(</span>
<span id="cb113-502"><a href="#cb113-502" aria-hidden="true" tabindex="-1"></a>  brain,</span>
<span id="cb113-503"><a href="#cb113-503" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="st">"Tnnt1"</span>,           <span class="co"># Replace with your gene of interest</span></span>
<span id="cb113-504"><a href="#cb113-504" aria-hidden="true" tabindex="-1"></a>  <span class="at">reduction =</span> <span class="st">"umap"</span>,          <span class="co"># Or "spatial" for spatial visualization</span></span>
<span id="cb113-505"><a href="#cb113-505" aria-hidden="true" tabindex="-1"></a>  <span class="at">pt.size =</span> <span class="fl">1.5</span></span>
<span id="cb113-506"><a href="#cb113-506" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-507"><a href="#cb113-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-508"><a href="#cb113-508" aria-hidden="true" tabindex="-1"></a><span class="co"># Violin plot</span></span>
<span id="cb113-509"><a href="#cb113-509" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(</span>
<span id="cb113-510"><a href="#cb113-510" aria-hidden="true" tabindex="-1"></a>  brain,</span>
<span id="cb113-511"><a href="#cb113-511" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="st">"Tnnt1"</span>,</span>
<span id="cb113-512"><a href="#cb113-512" aria-hidden="true" tabindex="-1"></a>  <span class="at">pt.size =</span> <span class="dv">0</span>,</span>
<span id="cb113-513"><a href="#cb113-513" aria-hidden="true" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">1</span></span>
<span id="cb113-514"><a href="#cb113-514" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-515"><a href="#cb113-515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-516"><a href="#cb113-516" aria-hidden="true" tabindex="-1"></a><span class="co"># Save DEG results</span></span>
<span id="cb113-517"><a href="#cb113-517" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(all_markers, "~/path/to/your/data/Spatial_Data_MouseBrainCoronal/Spatial_mouse_brain_DEG_all_clusters.csv")</span></span>
<span id="cb113-518"><a href="#cb113-518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-519"><a href="#cb113-519" aria-hidden="true" tabindex="-1"></a><span class="co"># Save Seurat object with markers</span></span>
<span id="cb113-520"><a href="#cb113-520" aria-hidden="true" tabindex="-1"></a><span class="co"># saveRDS(brain, file = "brain_with_markers.rds")</span></span>
<span id="cb113-521"><a href="#cb113-521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-522"><a href="#cb113-522" aria-hidden="true" tabindex="-1"></a><span class="co"># Spatial Feature Plots</span></span>
<span id="cb113-523"><a href="#cb113-523" aria-hidden="true" tabindex="-1"></a><span class="co"># Example: Plot expression of gene "Tnnt1" on spatial coordinates, highly expressed in spot #3</span></span>
<span id="cb113-524"><a href="#cb113-524" aria-hidden="true" tabindex="-1"></a><span class="fu">SpatialFeaturePlot</span>(</span>
<span id="cb113-525"><a href="#cb113-525" aria-hidden="true" tabindex="-1"></a>  brain,</span>
<span id="cb113-526"><a href="#cb113-526" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="st">"Tnnt1"</span>,          <span class="co"># Replace with your gene of interest</span></span>
<span id="cb113-527"><a href="#cb113-527" aria-hidden="true" tabindex="-1"></a>  <span class="at">pt.size.factor =</span> <span class="fl">2.5</span>,        <span class="co"># Adjust spot size (matching your SpatialDimPlot)</span></span>
<span id="cb113-528"><a href="#cb113-528" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="dv">2</span>),           <span class="co"># Adjust transparency (min and max expression)</span></span>
<span id="cb113-529"><a href="#cb113-529" aria-hidden="true" tabindex="-1"></a>  <span class="at">image.alpha =</span> <span class="dv">0</span>,             <span class="co"># Hide the tissue image (optional)</span></span>
<span id="cb113-530"><a href="#cb113-530" aria-hidden="true" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">1</span>                     <span class="co"># Number of columns (if plotting multiple genes)</span></span>
<span id="cb113-531"><a href="#cb113-531" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb113-532"><a href="#cb113-532" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Spatial expression of Tnnt1, highly expressed in cluster 3"</span>) <span class="sc">+</span></span>
<span id="cb113-533"><a href="#cb113-533" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb113-534"><a href="#cb113-534" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>, <span class="at">family =</span> <span class="st">"sans-serif"</span>, <span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb113-535"><a href="#cb113-535" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)</span>
<span id="cb113-536"><a href="#cb113-536" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb113-537"><a href="#cb113-537" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_c</span>(<span class="at">option =</span> <span class="st">"magma"</span>)  <span class="co"># Use viridis/magma for better contrast</span></span>
<span id="cb113-538"><a href="#cb113-538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-539"><a href="#cb113-539" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot two genes in a single row</span></span>
<span id="cb113-540"><a href="#cb113-540" aria-hidden="true" tabindex="-1"></a>cluster3_cluster1_highestDEG <span class="ot">&lt;-</span> <span class="fu">SpatialFeaturePlot</span>(</span>
<span id="cb113-541"><a href="#cb113-541" aria-hidden="true" tabindex="-1"></a>  brain,</span>
<span id="cb113-542"><a href="#cb113-542" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"Tnnt1"</span>, <span class="st">"Lypd1"</span>),  <span class="co"># Replace with your genes of interest</span></span>
<span id="cb113-543"><a href="#cb113-543" aria-hidden="true" tabindex="-1"></a>  <span class="at">pt.size.factor =</span> <span class="fl">2.5</span>,           <span class="co"># Match your SpatialDimPlot settings</span></span>
<span id="cb113-544"><a href="#cb113-544" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="dv">3</span>),              <span class="co"># Transparency range (0.1 for low, 1 for high)</span></span>
<span id="cb113-545"><a href="#cb113-545" aria-hidden="true" tabindex="-1"></a>  <span class="at">image.alpha =</span> <span class="dv">0</span>,                <span class="co"># Hide tissue image (optional)</span></span>
<span id="cb113-546"><a href="#cb113-546" aria-hidden="true" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">2</span>                        <span class="co"># Arrange plots in 2 columns</span></span>
<span id="cb113-547"><a href="#cb113-547" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb113-548"><a href="#cb113-548" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(</span>
<span id="cb113-549"><a href="#cb113-549" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Spatial Expression of Tnnt1 (cluster 3, left) and Lypd1 (cluster 1, right)"</span></span>
<span id="cb113-550"><a href="#cb113-550" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb113-551"><a href="#cb113-551" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb113-552"><a href="#cb113-552" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(</span>
<span id="cb113-553"><a href="#cb113-553" aria-hidden="true" tabindex="-1"></a>      <span class="at">size =</span> <span class="dv">30</span>,                   <span class="co"># Larger font size</span></span>
<span id="cb113-554"><a href="#cb113-554" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> <span class="st">"sans-serif"</span>,       <span class="co"># Font family</span></span>
<span id="cb113-555"><a href="#cb113-555" aria-hidden="true" tabindex="-1"></a>      <span class="at">face =</span> <span class="st">"bold"</span>,               <span class="co"># Bold text</span></span>
<span id="cb113-556"><a href="#cb113-556" aria-hidden="true" tabindex="-1"></a>      <span class="at">hjust =</span> <span class="fl">0.5</span>,                <span class="co"># Center-align title</span></span>
<span id="cb113-557"><a href="#cb113-557" aria-hidden="true" tabindex="-1"></a>      <span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">20</span>)     <span class="co"># Add top margin to prevent cutoff</span></span>
<span id="cb113-558"><a href="#cb113-558" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb113-559"><a href="#cb113-559" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb113-560"><a href="#cb113-560" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_c</span>(<span class="at">option =</span> <span class="st">"magma"</span>)  <span class="co"># Consistent color scale</span></span>
<span id="cb113-561"><a href="#cb113-561" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-562"><a href="#cb113-562" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(cluster3_cluster1_highestDEG)</span>
<span id="cb113-563"><a href="#cb113-563" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-564"><a href="#cb113-564" aria-hidden="true" tabindex="-1"></a><span class="co"># Save</span></span>
<span id="cb113-565"><a href="#cb113-565" aria-hidden="true" tabindex="-1"></a><span class="co"># ggsave("~/path/to/your/data/Spatial_highest_DEG_Visium_mouse_brain.png", cluster3_cluster1_highestDEG, width = 16, height = 8, dpi = 600)</span></span>
<span id="cb113-566"><a href="#cb113-566" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-567"><a href="#cb113-567" aria-hidden="true" tabindex="-1"></a><span class="co"># Display a safe summary of the Seurat object</span></span>
<span id="cb113-568"><a href="#cb113-568" aria-hidden="true" tabindex="-1"></a><span class="co"># str(brain, max.level = 2)  # Show top-level structure only</span></span>
<span id="cb113-569"><a href="#cb113-569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-570"><a href="#cb113-570" aria-hidden="true" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb113-571"><a href="#cb113-571" aria-hidden="true" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb113-572"><a href="#cb113-572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-573"><a href="#cb113-573" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize Spatial Expression according to SVG</span></span>
<span id="cb113-574"><a href="#cb113-574" aria-hidden="true" tabindex="-1"></a><span class="co"># If these genes show clear gradients or regional enrichment</span></span>
<span id="cb113-575"><a href="#cb113-575" aria-hidden="true" tabindex="-1"></a><span class="fu">SpatialFeaturePlot</span>(</span>
<span id="cb113-576"><a href="#cb113-576" aria-hidden="true" tabindex="-1"></a>  brain,</span>
<span id="cb113-577"><a href="#cb113-577" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"Sparc"</span>, <span class="st">"Camk2n1"</span>, <span class="st">"Pmch"</span>, <span class="st">"Nrgn"</span>),</span>
<span id="cb113-578"><a href="#cb113-578" aria-hidden="true" tabindex="-1"></a>  <span class="at">pt.size.factor =</span> <span class="fl">2.5</span>,</span>
<span id="cb113-579"><a href="#cb113-579" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="dv">3</span>),</span>
<span id="cb113-580"><a href="#cb113-580" aria-hidden="true" tabindex="-1"></a>  <span class="at">image.alpha =</span> <span class="fl">0.25</span></span>
<span id="cb113-581"><a href="#cb113-581" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-582"><a href="#cb113-582" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-583"><a href="#cb113-583" aria-hidden="true" tabindex="-1"></a><span class="co"># GO Enrichment analysis</span></span>
<span id="cb113-584"><a href="#cb113-584" aria-hidden="true" tabindex="-1"></a>enrich_result <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(</span>
<span id="cb113-585"><a href="#cb113-585" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene =</span> <span class="fu">rownames</span>(top_spatial_genes),</span>
<span id="cb113-586"><a href="#cb113-586" aria-hidden="true" tabindex="-1"></a>  <span class="at">OrgDb =</span> org.Mm.eg.db,     <span class="co"># org.Hs.eg.db (human)</span></span>
<span id="cb113-587"><a href="#cb113-587" aria-hidden="true" tabindex="-1"></a>  <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>,       <span class="co"># or "ENSEMBL" if genes are Ensembl IDs</span></span>
<span id="cb113-588"><a href="#cb113-588" aria-hidden="true" tabindex="-1"></a>  <span class="at">ont =</span> <span class="st">"BP"</span>               <span class="co"># Biological Process ontology</span></span>
<span id="cb113-589"><a href="#cb113-589" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-590"><a href="#cb113-590" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize results</span></span>
<span id="cb113-591"><a href="#cb113-591" aria-hidden="true" tabindex="-1"></a>go_plot <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(enrich_result, <span class="at">showCategory =</span> <span class="dv">10</span>,</span>
<span id="cb113-592"><a href="#cb113-592" aria-hidden="true" tabindex="-1"></a>        <span class="at">orderBy =</span> <span class="st">"GeneRatio"</span>, <span class="at">title =</span> <span class="st">"Top Enriched GO Terms (Biological Process)"</span>) </span>
<span id="cb113-593"><a href="#cb113-593" aria-hidden="true" tabindex="-1"></a><span class="co"># top 10 most significantly enriched GO terms based on adjusted p-value or other ranking metrics.</span></span>
<span id="cb113-594"><a href="#cb113-594" aria-hidden="true" tabindex="-1"></a><span class="co"># Gene Ratio: 0.15 means p = 1e-10</span></span>
<span id="cb113-595"><a href="#cb113-595" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-596"><a href="#cb113-596" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(go_plot)</span>
<span id="cb113-597"><a href="#cb113-597" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-598"><a href="#cb113-598" aria-hidden="true" tabindex="-1"></a><span class="co"># ggsave(</span></span>
<span id="cb113-599"><a href="#cb113-599" aria-hidden="true" tabindex="-1"></a><span class="co">#   filename = "~/path/to/your/data/Spatial_brain_mouse_go_enrichment_highres.png",</span></span>
<span id="cb113-600"><a href="#cb113-600" aria-hidden="true" tabindex="-1"></a><span class="co">#   plot = go_plot,</span></span>
<span id="cb113-601"><a href="#cb113-601" aria-hidden="true" tabindex="-1"></a><span class="co">#   width = 12,                      # Wider for better label readability</span></span>
<span id="cb113-602"><a href="#cb113-602" aria-hidden="true" tabindex="-1"></a><span class="co">#   height = 8,</span></span>
<span id="cb113-603"><a href="#cb113-603" aria-hidden="true" tabindex="-1"></a><span class="co">#   dpi = 600,                       # Higher resolution for publications</span></span>
<span id="cb113-604"><a href="#cb113-604" aria-hidden="true" tabindex="-1"></a><span class="co">#   bg = "white"                     # Set background to white</span></span>
<span id="cb113-605"><a href="#cb113-605" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb113-606"><a href="#cb113-606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-607"><a href="#cb113-607" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb113-608"><a href="#cb113-608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-609"><a href="#cb113-609" aria-hidden="true" tabindex="-1"></a><span class="fu">## *Brain region annotation / Summary table*</span></span>
<span id="cb113-610"><a href="#cb113-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-613"><a href="#cb113-613" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb113-614"><a href="#cb113-614" aria-hidden="true" tabindex="-1"></a><span class="co"># Install if not installed</span></span>
<span id="cb113-615"><a href="#cb113-615" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("ABAData") --&gt; </span><span class="al">DEPRECATED</span></span>
<span id="cb113-616"><a href="#cb113-616" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("ABAbilaR") --&gt; </span><span class="al">DEPRECATED</span></span>
<span id="cb113-617"><a href="#cb113-617" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("SingleR")</span></span>
<span id="cb113-618"><a href="#cb113-618" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("spatialLIBD")</span></span>
<span id="cb113-619"><a href="#cb113-619" aria-hidden="true" tabindex="-1"></a><span class="co"># remotes::install_github("benfulcher/allen_clr") --&gt; </span><span class="al">DEPRECATED</span></span>
<span id="cb113-620"><a href="#cb113-620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-621"><a href="#cb113-621" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb113-622"><a href="#cb113-622" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleR)</span>
<span id="cb113-623"><a href="#cb113-623" aria-hidden="true" tabindex="-1"></a><span class="co"># library(ABAData)  # Allen Brain Atlas --&gt; </span><span class="al">DEPRECATED</span></span>
<span id="cb113-624"><a href="#cb113-624" aria-hidden="true" tabindex="-1"></a><span class="co"># library(ABAbilaR) # Interface with Allen Brain Atlas --&gt; </span><span class="al">DEPRECATED</span></span>
<span id="cb113-625"><a href="#cb113-625" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(spatialLIBD)</span>
<span id="cb113-626"><a href="#cb113-626" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-627"><a href="#cb113-627" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign broad brain regions manually based on spatial clustering</span></span>
<span id="cb113-628"><a href="#cb113-628" aria-hidden="true" tabindex="-1"></a>brain<span class="sc">$</span>brain_region <span class="ot">&lt;-</span> <span class="fu">case_when</span>(</span>
<span id="cb113-629"><a href="#cb113-629" aria-hidden="true" tabindex="-1"></a>  brain<span class="sc">$</span>seurat_clusters <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="dv">6</span>) <span class="sc">~</span> <span class="st">"Neocortex"</span>, <span class="do">##</span></span>
<span id="cb113-630"><a href="#cb113-630" aria-hidden="true" tabindex="-1"></a>  brain<span class="sc">$</span>seurat_clusters <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">16</span>, <span class="dv">4</span>, <span class="dv">17</span>) <span class="sc">~</span> <span class="st">"Hippocampal region"</span>,</span>
<span id="cb113-631"><a href="#cb113-631" aria-hidden="true" tabindex="-1"></a>  brain<span class="sc">$</span>seurat_clusters <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">3</span>) <span class="sc">~</span> <span class="st">"Thalamus"</span>, <span class="do">##</span></span>
<span id="cb113-632"><a href="#cb113-632" aria-hidden="true" tabindex="-1"></a>  brain<span class="sc">$</span>seurat_clusters <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">2</span>) <span class="sc">~</span> <span class="st">"Hypothalamus"</span>, <span class="do">##</span></span>
<span id="cb113-633"><a href="#cb113-633" aria-hidden="true" tabindex="-1"></a>  brain<span class="sc">$</span>seurat_clusters <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">13</span>, <span class="dv">14</span>) <span class="sc">~</span> <span class="st">"Caudatoputamen"</span>,</span>
<span id="cb113-634"><a href="#cb113-634" aria-hidden="true" tabindex="-1"></a>  brain<span class="sc">$</span>seurat_clusters <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">11</span>) <span class="sc">~</span> <span class="st">"Allocortex"</span>,</span>
<span id="cb113-635"><a href="#cb113-635" aria-hidden="true" tabindex="-1"></a>  brain<span class="sc">$</span>seurat_clusters <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">15</span>) <span class="sc">~</span> <span class="st">"Striatum-like amygdalar nuclei"</span>, <span class="do">##</span></span>
<span id="cb113-636"><a href="#cb113-636" aria-hidden="true" tabindex="-1"></a>  brain<span class="sc">$</span>seurat_clusters <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">9</span>, <span class="dv">10</span>, <span class="dv">5</span>, <span class="dv">7</span>) <span class="sc">~</span> <span class="st">"Fiber tracts"</span>, <span class="do">##</span></span>
<span id="cb113-637"><a href="#cb113-637" aria-hidden="true" tabindex="-1"></a>  <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"N/A"</span></span>
<span id="cb113-638"><a href="#cb113-638" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-639"><a href="#cb113-639" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-640"><a href="#cb113-640" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(brain<span class="sc">$</span>brain_region)  <span class="co"># Count spots per region</span></span>
<span id="cb113-641"><a href="#cb113-641" aria-hidden="true" tabindex="-1"></a><span class="co"># head(brain@meta.data)  # View metadata</span></span>
<span id="cb113-642"><a href="#cb113-642" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(brain<span class="sc">@</span>meta.data)</span>
<span id="cb113-643"><a href="#cb113-643" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-644"><a href="#cb113-644" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-645"><a href="#cb113-645" aria-hidden="true" tabindex="-1"></a>p7 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(</span>
<span id="cb113-646"><a href="#cb113-646" aria-hidden="true" tabindex="-1"></a>  brain, </span>
<span id="cb113-647"><a href="#cb113-647" aria-hidden="true" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">"brain_region"</span>, </span>
<span id="cb113-648"><a href="#cb113-648" aria-hidden="true" tabindex="-1"></a>  <span class="at">label =</span> <span class="cn">FALSE</span>, </span>
<span id="cb113-649"><a href="#cb113-649" aria-hidden="true" tabindex="-1"></a>  <span class="at">repel =</span> <span class="cn">TRUE</span>,            <span class="co"># Automatically separates labels</span></span>
<span id="cb113-650"><a href="#cb113-650" aria-hidden="true" tabindex="-1"></a>  <span class="at">pt.size =</span> <span class="fl">2.5</span>,           <span class="co"># Size of points on UMAP</span></span>
<span id="cb113-651"><a href="#cb113-651" aria-hidden="true" tabindex="-1"></a>  <span class="at">label.size =</span> <span class="dv">5</span>,          <span class="co"># Font size of cluster labels</span></span>
<span id="cb113-652"><a href="#cb113-652" aria-hidden="true" tabindex="-1"></a>  <span class="at">label.box =</span> <span class="cn">TRUE</span>,        <span class="co"># Adds background box to labels</span></span>
<span id="cb113-653"><a href="#cb113-653" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fl">0.8</span></span>
<span id="cb113-654"><a href="#cb113-654" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb113-655"><a href="#cb113-655" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"UMAP Brain Region Annotation"</span>) <span class="sc">+</span></span>
<span id="cb113-656"><a href="#cb113-656" aria-hidden="true" tabindex="-1"></a>  <span class="fu">NoAxes</span>() <span class="sc">+</span></span>
<span id="cb113-657"><a href="#cb113-657" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb113-658"><a href="#cb113-658" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Use `color` instead of `fill` for DimPlot legends</span></span>
<span id="cb113-659"><a href="#cb113-659" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="fu">guide_legend</span>(</span>
<span id="cb113-660"><a href="#cb113-660" aria-hidden="true" tabindex="-1"></a>      <span class="at">title =</span> <span class="st">"Brain Region"</span>,</span>
<span id="cb113-661"><a href="#cb113-661" aria-hidden="true" tabindex="-1"></a>      <span class="at">override.aes =</span> <span class="fu">list</span>(</span>
<span id="cb113-662"><a href="#cb113-662" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">6</span>,          <span class="co"># Size of color dots in legend</span></span>
<span id="cb113-663"><a href="#cb113-663" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="dv">1</span>          <span class="co"># Ensure full opacity for legend keys</span></span>
<span id="cb113-664"><a href="#cb113-664" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb113-665"><a href="#cb113-665" aria-hidden="true" tabindex="-1"></a>      <span class="at">ncol =</span> <span class="dv">1</span>             <span class="co"># Single column for vertical legend</span></span>
<span id="cb113-666"><a href="#cb113-666" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb113-667"><a href="#cb113-667" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb113-668"><a href="#cb113-668" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb113-669"><a href="#cb113-669" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>, <span class="at">family =</span> <span class="st">"sans-serif"</span>, <span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb113-670"><a href="#cb113-670" aria-hidden="true" tabindex="-1"></a>    <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">family =</span> <span class="st">"sans-serif"</span>),</span>
<span id="cb113-671"><a href="#cb113-671" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),  <span class="co"># Legend title font size</span></span>
<span id="cb113-672"><a href="#cb113-672" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),   <span class="co"># Legend text font size</span></span>
<span id="cb113-673"><a href="#cb113-673" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"cm"</span>),         <span class="co"># Size of legend key boxes</span></span>
<span id="cb113-674"><a href="#cb113-674" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.spacing.y =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">"cm"</span>)       <span class="co"># Vertical spacing between legend items</span></span>
<span id="cb113-675"><a href="#cb113-675" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb113-676"><a href="#cb113-676" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-677"><a href="#cb113-677" aria-hidden="true" tabindex="-1"></a>p8 <span class="ot">&lt;-</span> <span class="fu">SpatialDimPlot</span>(</span>
<span id="cb113-678"><a href="#cb113-678" aria-hidden="true" tabindex="-1"></a>  brain,</span>
<span id="cb113-679"><a href="#cb113-679" aria-hidden="true" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">"brain_region"</span>,</span>
<span id="cb113-680"><a href="#cb113-680" aria-hidden="true" tabindex="-1"></a>  <span class="at">label =</span> <span class="cn">TRUE</span>,    <span class="co"># Labels on (TRUE) or off (FALSE) on the brain regions</span></span>
<span id="cb113-681"><a href="#cb113-681" aria-hidden="true" tabindex="-1"></a>  <span class="at">repel =</span> <span class="cn">TRUE</span>,    <span class="co"># Prevents overlap of labels</span></span>
<span id="cb113-682"><a href="#cb113-682" aria-hidden="true" tabindex="-1"></a>  <span class="at">label.size =</span> <span class="dv">6</span>,</span>
<span id="cb113-683"><a href="#cb113-683" aria-hidden="true" tabindex="-1"></a>  <span class="at">pt.size.factor =</span> <span class="fl">3.5</span>,</span>
<span id="cb113-684"><a href="#cb113-684" aria-hidden="true" tabindex="-1"></a>  <span class="at">image.alpha =</span> <span class="fl">0.5</span>,</span>
<span id="cb113-685"><a href="#cb113-685" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fl">0.8</span></span>
<span id="cb113-686"><a href="#cb113-686" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb113-687"><a href="#cb113-687" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Spatial Brain Region Annotation"</span>) <span class="sc">+</span></span>
<span id="cb113-688"><a href="#cb113-688" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Customize legend title and key size</span></span>
<span id="cb113-689"><a href="#cb113-689" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(</span>
<span id="cb113-690"><a href="#cb113-690" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">guide_legend</span>(</span>
<span id="cb113-691"><a href="#cb113-691" aria-hidden="true" tabindex="-1"></a>      <span class="at">title =</span> <span class="st">"Brain Region"</span>,  <span class="co"># New legend title</span></span>
<span id="cb113-692"><a href="#cb113-692" aria-hidden="true" tabindex="-1"></a>      <span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="dv">6</span>)  <span class="co"># Increase color circle size</span></span>
<span id="cb113-693"><a href="#cb113-693" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb113-694"><a href="#cb113-694" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb113-695"><a href="#cb113-695" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb113-696"><a href="#cb113-696" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>, <span class="at">family =</span> <span class="st">"sans-serif"</span>, <span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb113-697"><a href="#cb113-697" aria-hidden="true" tabindex="-1"></a>    <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">family =</span> <span class="st">"sans-serif"</span>),</span>
<span id="cb113-698"><a href="#cb113-698" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"grey"</span>),</span>
<span id="cb113-699"><a href="#cb113-699" aria-hidden="true" tabindex="-1"></a>    <span class="co"># legend.position = "none",  # Remove legend from the right side</span></span>
<span id="cb113-700"><a href="#cb113-700" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>),  <span class="co"># Legend title font size</span></span>
<span id="cb113-701"><a href="#cb113-701" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),    <span class="co"># Legend text font size</span></span>
<span id="cb113-702"><a href="#cb113-702" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"cm"</span>),        <span class="co"># Overall legend key box size</span></span>
<span id="cb113-703"><a href="#cb113-703" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="st">"white"</span>), <span class="co"># Plot area</span></span>
<span id="cb113-704"><a href="#cb113-704" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="st">"white"</span>)   <span class="co"># Entire background</span></span>
<span id="cb113-705"><a href="#cb113-705" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb113-706"><a href="#cb113-706" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-707"><a href="#cb113-707" aria-hidden="true" tabindex="-1"></a>p9 <span class="ot">&lt;-</span> <span class="fu">SpatialDimPlot</span>(</span>
<span id="cb113-708"><a href="#cb113-708" aria-hidden="true" tabindex="-1"></a>  brain,</span>
<span id="cb113-709"><a href="#cb113-709" aria-hidden="true" tabindex="-1"></a>  <span class="at">pt.size.factor =</span> <span class="fl">3.5</span>,</span>
<span id="cb113-710"><a href="#cb113-710" aria-hidden="true" tabindex="-1"></a>  <span class="at">image.alpha =</span> <span class="dv">1</span>,</span>
<span id="cb113-711"><a href="#cb113-711" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fl">0.0</span></span>
<span id="cb113-712"><a href="#cb113-712" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb113-713"><a href="#cb113-713" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"H&amp;E Brain Region Coronal"</span>) <span class="sc">+</span></span>
<span id="cb113-714"><a href="#cb113-714" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Semi-transparent white overlay</span></span>
<span id="cb113-715"><a href="#cb113-715" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(</span>
<span id="cb113-716"><a href="#cb113-716" aria-hidden="true" tabindex="-1"></a>    <span class="st">"rect"</span>, </span>
<span id="cb113-717"><a href="#cb113-717" aria-hidden="true" tabindex="-1"></a>    <span class="at">xmin =</span> <span class="sc">-</span><span class="cn">Inf</span>, <span class="at">xmax =</span> <span class="cn">Inf</span>, </span>
<span id="cb113-718"><a href="#cb113-718" aria-hidden="true" tabindex="-1"></a>    <span class="at">ymin =</span> <span class="sc">-</span><span class="cn">Inf</span>, <span class="at">ymax =</span> <span class="cn">Inf</span>, </span>
<span id="cb113-719"><a href="#cb113-719" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">alpha =</span> <span class="fl">0.2</span>  <span class="co"># Adjust alpha (0 = no overlay, 1 = fully white)</span></span>
<span id="cb113-720"><a href="#cb113-720" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb113-721"><a href="#cb113-721" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb113-722"><a href="#cb113-722" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>, <span class="at">family =</span> <span class="st">"sans-serif"</span>, <span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb113-723"><a href="#cb113-723" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span>,          <span class="co"># Remove any residual legend</span></span>
<span id="cb113-724"><a href="#cb113-724" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),      <span class="co"># Remove x/y axis titles</span></span>
<span id="cb113-725"><a href="#cb113-725" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_blank</span>(),       <span class="co"># Remove x/y axis text</span></span>
<span id="cb113-726"><a href="#cb113-726" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),      <span class="co"># Remove axis ticks</span></span>
<span id="cb113-727"><a href="#cb113-727" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),      <span class="co"># Remove grid lines</span></span>
<span id="cb113-728"><a href="#cb113-728" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="cn">NA</span>),</span>
<span id="cb113-729"><a href="#cb113-729" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="cn">NA</span>)</span>
<span id="cb113-730"><a href="#cb113-730" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb113-731"><a href="#cb113-731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-732"><a href="#cb113-732" aria-hidden="true" tabindex="-1"></a>patch1 <span class="ot">&lt;-</span> (p9 <span class="sc">|</span> p8) <span class="sc">/</span> p7</span>
<span id="cb113-733"><a href="#cb113-733" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(patch1)</span>
<span id="cb113-734"><a href="#cb113-734" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-735"><a href="#cb113-735" aria-hidden="true" tabindex="-1"></a><span class="co"># Save 'H&amp;E (p9)'</span></span>
<span id="cb113-736"><a href="#cb113-736" aria-hidden="true" tabindex="-1"></a><span class="co"># ggsave("~/path/to/your/data/Spatial_Visium_mouse_brain_H&amp;E.png", p9, width = 16, height = 8, dpi = 600)</span></span>
<span id="cb113-737"><a href="#cb113-737" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-738"><a href="#cb113-738" aria-hidden="true" tabindex="-1"></a><span class="co"># Save 'UMAP_regions (p7)'</span></span>
<span id="cb113-739"><a href="#cb113-739" aria-hidden="true" tabindex="-1"></a><span class="co"># ggsave("~/path/to/your/data/Spatial_Visium_mouse_brain_UMAP_regions.png", p7, width = 16, height = 8, dpi = 600)</span></span>
<span id="cb113-740"><a href="#cb113-740" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-741"><a href="#cb113-741" aria-hidden="true" tabindex="-1"></a><span class="co"># Save 'patch1'</span></span>
<span id="cb113-742"><a href="#cb113-742" aria-hidden="true" tabindex="-1"></a><span class="co"># ggsave("~/path/to/your/data/Spatial_Visium_mouse_brain_regions_patch.png", patch1, width = 16, height = 8, dpi = 600)</span></span>
<span id="cb113-743"><a href="#cb113-743" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-744"><a href="#cb113-744" aria-hidden="true" tabindex="-1"></a><span class="co"># Save 'Brain_regions (p8)'</span></span>
<span id="cb113-745"><a href="#cb113-745" aria-hidden="true" tabindex="-1"></a><span class="co"># ggsave("~/path/to/your/data/Spatial_Visium_mouse_brain_regions.png", p8, width = 16, height = 8, dpi = 600)</span></span>
<span id="cb113-746"><a href="#cb113-746" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-747"><a href="#cb113-747" aria-hidden="true" tabindex="-1"></a><span class="co"># Save 'Brain_regions (p8)'</span></span>
<span id="cb113-748"><a href="#cb113-748" aria-hidden="true" tabindex="-1"></a><span class="co"># ggsave("~path/to/your/data/Spatial_Visium_mouse_brain_regionsNoLegends.png", p8, width = 16, height = 8, dpi = 600)</span></span>
<span id="cb113-749"><a href="#cb113-749" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-750"><a href="#cb113-750" aria-hidden="true" tabindex="-1"></a><span class="co"># Define region-specific markers (expand with more genes)</span></span>
<span id="cb113-751"><a href="#cb113-751" aria-hidden="true" tabindex="-1"></a>region_markers <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb113-752"><a href="#cb113-752" aria-hidden="true" tabindex="-1"></a>  <span class="at">Neocortex =</span> <span class="fu">c</span>(<span class="st">"Slc17a7"</span>, <span class="st">"Rorb"</span>, <span class="st">"Foxp2"</span>, <span class="st">"Cux1"</span>,      <span class="co"># Excitatory neurons</span></span>
<span id="cb113-753"><a href="#cb113-753" aria-hidden="true" tabindex="-1"></a>                <span class="st">"Gad1"</span>,<span class="st">"Gad2"</span>, <span class="st">"Pvalb"</span>, <span class="st">"Sst"</span>, <span class="st">"Vip"</span>,    <span class="co"># Inhibitory neurons</span></span>
<span id="cb113-754"><a href="#cb113-754" aria-hidden="true" tabindex="-1"></a>                <span class="st">"Gfap"</span>, <span class="st">"Aqp4"</span>, <span class="st">"Aldh1l1"</span>, <span class="st">"Ptprc"</span>),     <span class="co"># Glial cells</span></span>
<span id="cb113-755"><a href="#cb113-755" aria-hidden="true" tabindex="-1"></a>  <span class="at">Hippocampus =</span> <span class="fu">c</span>(<span class="st">"Wfs1"</span>, <span class="st">"Nr3c2"</span>, <span class="st">"Rgs14"</span>,              <span class="co"># CA1</span></span>
<span id="cb113-756"><a href="#cb113-756" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Slc17a7"</span>, <span class="st">"Grik1"</span>, <span class="st">"Prkcd"</span>,           <span class="co"># CA3</span></span>
<span id="cb113-757"><a href="#cb113-757" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Prox1"</span>, <span class="st">"C1ql2"</span>, <span class="st">"Neurod1"</span>, <span class="st">"Hpca"</span>),  <span class="co"># Dentate Gyrus (granule cells)</span></span>
<span id="cb113-758"><a href="#cb113-758" aria-hidden="true" tabindex="-1"></a>  <span class="at">Thalamus =</span> <span class="fu">c</span>(<span class="st">"Gad2"</span>, <span class="st">"Npy"</span>, <span class="st">"Slc32a1"</span>,</span>
<span id="cb113-759"><a href="#cb113-759" aria-hidden="true" tabindex="-1"></a>               <span class="st">"Calb1"</span>, <span class="st">"Vglut2"</span>,<span class="st">"Foxp2"</span>, <span class="st">"Tbr1"</span>),  <span class="co"># Excitatory thalamic neurons, Sensory relay neurons, Thalamocortical projection neurons</span></span>
<span id="cb113-760"><a href="#cb113-760" aria-hidden="true" tabindex="-1"></a>  <span class="at">Hypothalamus =</span> <span class="fu">c</span>(<span class="st">"Oxt"</span>, <span class="st">"Avp"</span>,<span class="st">"Cartpt"</span> ,</span>
<span id="cb113-761"><a href="#cb113-761" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Pomc"</span>, <span class="st">"Agrp"</span>, <span class="st">"Npy"</span>, <span class="st">"Hcrt"</span>, <span class="st">"Crh"</span>),          <span class="co"># Arcuate nucleus</span></span>
<span id="cb113-762"><a href="#cb113-762" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb113-763"><a href="#cb113-763" aria-hidden="true" tabindex="-1"></a>  <span class="at">Caudoputamen_Striatum =</span> <span class="fu">c</span>(<span class="st">"Cplx3"</span>,<span class="st">"Drd1"</span>, <span class="st">"Drd2"</span>, <span class="st">"Ppp1r1b"</span>),           <span class="co"># Dopamine receptors</span></span>
<span id="cb113-764"><a href="#cb113-764" aria-hidden="true" tabindex="-1"></a>  <span class="at">Allocortex =</span> <span class="fu">c</span>(<span class="st">"Rasgrf2"</span>,<span class="st">"Calb2"</span>,<span class="st">"Pcp4"</span>),</span>
<span id="cb113-765"><a href="#cb113-765" aria-hidden="true" tabindex="-1"></a>  <span class="at">Striatum_like_Amygdalar_Nuclei =</span> <span class="fu">c</span>(<span class="st">"Cplx3"</span>,<span class="st">"Tac2"</span>,<span class="st">"Drd1"</span>, <span class="st">"Drd2"</span>,</span>
<span id="cb113-766"><a href="#cb113-766" aria-hidden="true" tabindex="-1"></a>                                     <span class="st">"Pvalb"</span>, <span class="st">"Npy"</span>),</span>
<span id="cb113-767"><a href="#cb113-767" aria-hidden="true" tabindex="-1"></a>  <span class="at">FiberTracts =</span> <span class="fu">c</span>(<span class="st">"Mbp"</span>, <span class="st">"Plp1"</span>, <span class="st">"Mog"</span>, <span class="st">"Olig2"</span>, <span class="st">"Sox10"</span>, <span class="st">"Cldn11"</span>)              <span class="co"># Oligodendrocytes</span></span>
<span id="cb113-768"><a href="#cb113-768" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-769"><a href="#cb113-769" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-770"><a href="#cb113-770" aria-hidden="true" tabindex="-1"></a><span class="co"># Neocortex region:</span></span>
<span id="cb113-771"><a href="#cb113-771" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-772"><a href="#cb113-772" aria-hidden="true" tabindex="-1"></a><span class="co"># Excitatory neurons: Slc17a7, Rorb, Foxp2, Cux1</span></span>
<span id="cb113-773"><a href="#cb113-773" aria-hidden="true" tabindex="-1"></a><span class="co"># Inhibitory neurons: Gad1, Gad2, Pvalb, Sst, Vip</span></span>
<span id="cb113-774"><a href="#cb113-774" aria-hidden="true" tabindex="-1"></a><span class="co"># Glial cells: Gfap, Aqp4, Aldh1l1</span></span>
<span id="cb113-775"><a href="#cb113-775" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-776"><a href="#cb113-776" aria-hidden="true" tabindex="-1"></a><span class="co"># Hippocampal region:</span></span>
<span id="cb113-777"><a href="#cb113-777" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-778"><a href="#cb113-778" aria-hidden="true" tabindex="-1"></a><span class="co"># CA1 pyramidal neurons: Wfs1, Nr3c2, Rgs14</span></span>
<span id="cb113-779"><a href="#cb113-779" aria-hidden="true" tabindex="-1"></a><span class="co"># CA3 pyramidal neurons: Slc17a7, Grik1, Prkcd</span></span>
<span id="cb113-780"><a href="#cb113-780" aria-hidden="true" tabindex="-1"></a><span class="co"># Dentate gyrus (granule cells): Prox1, C1ql2, Neurod1</span></span>
<span id="cb113-781"><a href="#cb113-781" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-782"><a href="#cb113-782" aria-hidden="true" tabindex="-1"></a><span class="co"># Thalamus region:</span></span>
<span id="cb113-783"><a href="#cb113-783" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-784"><a href="#cb113-784" aria-hidden="true" tabindex="-1"></a><span class="co"># Calb1 (Calbindin)</span></span>
<span id="cb113-785"><a href="#cb113-785" aria-hidden="true" tabindex="-1"></a><span class="co"># Vglut2 (Slc17a6) – Excitatory thalamic neurons</span></span>
<span id="cb113-786"><a href="#cb113-786" aria-hidden="true" tabindex="-1"></a><span class="co"># Foxp2 – Sensory relay neurons</span></span>
<span id="cb113-787"><a href="#cb113-787" aria-hidden="true" tabindex="-1"></a><span class="co"># Tbr1 – Thalamocortical projection neurons</span></span>
<span id="cb113-788"><a href="#cb113-788" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-789"><a href="#cb113-789" aria-hidden="true" tabindex="-1"></a><span class="co"># Hypothalamus region:</span></span>
<span id="cb113-790"><a href="#cb113-790" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-791"><a href="#cb113-791" aria-hidden="true" tabindex="-1"></a><span class="co"># Oxt (Oxytocin)</span></span>
<span id="cb113-792"><a href="#cb113-792" aria-hidden="true" tabindex="-1"></a><span class="co"># Avp (Vasopressin)</span></span>
<span id="cb113-793"><a href="#cb113-793" aria-hidden="true" tabindex="-1"></a><span class="co"># Cartpt (Cocaine- and amphetamine-regulated transcript)</span></span>
<span id="cb113-794"><a href="#cb113-794" aria-hidden="true" tabindex="-1"></a><span class="co"># Hypothalamic nuclei: Pomc, Npy, Hcrt (Orexin), Crh (Corticotropin-releasing hormone)</span></span>
<span id="cb113-795"><a href="#cb113-795" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-796"><a href="#cb113-796" aria-hidden="true" tabindex="-1"></a><span class="co"># Caudatoputamen (Striatum) region:</span></span>
<span id="cb113-797"><a href="#cb113-797" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-798"><a href="#cb113-798" aria-hidden="true" tabindex="-1"></a><span class="co"># Drd1 (D1 dopamine receptor, direct pathway)</span></span>
<span id="cb113-799"><a href="#cb113-799" aria-hidden="true" tabindex="-1"></a><span class="co"># Drd2 (D2 dopamine receptor, indirect pathway)</span></span>
<span id="cb113-800"><a href="#cb113-800" aria-hidden="true" tabindex="-1"></a><span class="co"># Ppp1r1b (DARPP-32, marker of medium spiny neurons)</span></span>
<span id="cb113-801"><a href="#cb113-801" aria-hidden="true" tabindex="-1"></a><span class="co"># Foxp2 (Regulates motor function in striatal neurons)</span></span>
<span id="cb113-802"><a href="#cb113-802" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-803"><a href="#cb113-803" aria-hidden="true" tabindex="-1"></a><span class="co"># Allocortex region:</span></span>
<span id="cb113-804"><a href="#cb113-804" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-805"><a href="#cb113-805" aria-hidden="true" tabindex="-1"></a><span class="co"># Rasgrf2 (Piriform cortex)</span></span>
<span id="cb113-806"><a href="#cb113-806" aria-hidden="true" tabindex="-1"></a><span class="co"># Calb2 (Layer-specific cortical neurons)</span></span>
<span id="cb113-807"><a href="#cb113-807" aria-hidden="true" tabindex="-1"></a><span class="co"># Pcp4 (Parvalbumin interneurons)</span></span>
<span id="cb113-808"><a href="#cb113-808" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-809"><a href="#cb113-809" aria-hidden="true" tabindex="-1"></a><span class="co"># Striatum-like Amygdalar Nuclei region:</span></span>
<span id="cb113-810"><a href="#cb113-810" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-811"><a href="#cb113-811" aria-hidden="true" tabindex="-1"></a><span class="co"># Cplx3, Tac2 (Central amygdala)</span></span>
<span id="cb113-812"><a href="#cb113-812" aria-hidden="true" tabindex="-1"></a><span class="co"># Foxp2 (Basolateral amygdala)</span></span>
<span id="cb113-813"><a href="#cb113-813" aria-hidden="true" tabindex="-1"></a><span class="co"># Drd1, Drd2 (Dopamine receptor expression in striatum-like regions)</span></span>
<span id="cb113-814"><a href="#cb113-814" aria-hidden="true" tabindex="-1"></a><span class="co"># Pvalb, Npy (Interneurons in the amygdala)</span></span>
<span id="cb113-815"><a href="#cb113-815" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-816"><a href="#cb113-816" aria-hidden="true" tabindex="-1"></a><span class="co"># Fiber Tracts region:</span></span>
<span id="cb113-817"><a href="#cb113-817" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-818"><a href="#cb113-818" aria-hidden="true" tabindex="-1"></a><span class="co"># Mbp (Myelin basic protein, oligodendrocytes)</span></span>
<span id="cb113-819"><a href="#cb113-819" aria-hidden="true" tabindex="-1"></a><span class="co"># Plp1, Cnp (Myelination markers)</span></span>
<span id="cb113-820"><a href="#cb113-820" aria-hidden="true" tabindex="-1"></a><span class="co"># Olig2, Sox10 (Oligodendrocyte precursor cells)</span></span>
<span id="cb113-821"><a href="#cb113-821" aria-hidden="true" tabindex="-1"></a><span class="co"># Cldn11 (Tight junctions in myelin sheaths)</span></span>
<span id="cb113-822"><a href="#cb113-822" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-823"><a href="#cb113-823" aria-hidden="true" tabindex="-1"></a><span class="co"># Dot plot to validate markers</span></span>
<span id="cb113-824"><a href="#cb113-824" aria-hidden="true" tabindex="-1"></a><span class="fu">DotPlot</span>(brain, <span class="at">features =</span> <span class="fu">unique</span>(<span class="fu">unlist</span>(region_markers)), <span class="at">group.by =</span> <span class="st">"brain_region"</span>) <span class="sc">+</span></span>
<span id="cb113-825"><a href="#cb113-825" aria-hidden="true" tabindex="-1"></a>  <span class="fu">RotatedAxis</span>() <span class="sc">+</span></span>
<span id="cb113-826"><a href="#cb113-826" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis_c</span>(<span class="at">option =</span> <span class="st">"magma"</span>)</span>
<span id="cb113-827"><a href="#cb113-827" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-828"><a href="#cb113-828" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-829"><a href="#cb113-829" aria-hidden="true" tabindex="-1"></a><span class="co"># The current mapping has potential overlaps (e.g., cluster 11 in both Neocortex and Allocortex). Investigate ambiguous clusters:</span></span>
<span id="cb113-830"><a href="#cb113-830" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset ambiguous clusters (e.g., cluster 11)</span></span>
<span id="cb113-831"><a href="#cb113-831" aria-hidden="true" tabindex="-1"></a>cluster11 <span class="ot">&lt;-</span> <span class="fu">subset</span>(brain, seurat_clusters <span class="sc">==</span> <span class="dv">11</span>)</span>
<span id="cb113-832"><a href="#cb113-832" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-833"><a href="#cb113-833" aria-hidden="true" tabindex="-1"></a><span class="co"># Find unique markers for cluster 11</span></span>
<span id="cb113-834"><a href="#cb113-834" aria-hidden="true" tabindex="-1"></a>cluster11_markers <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(brain, <span class="at">ident.1 =</span> <span class="dv">11</span>, <span class="at">logfc.threshold =</span> <span class="fl">0.5</span>)</span>
<span id="cb113-835"><a href="#cb113-835" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cluster11_markers, <span class="dv">10</span>)</span>
<span id="cb113-836"><a href="#cb113-836" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-837"><a href="#cb113-837" aria-hidden="true" tabindex="-1"></a><span class="co"># Spatial plot of top marker</span></span>
<span id="cb113-838"><a href="#cb113-838" aria-hidden="true" tabindex="-1"></a><span class="fu">SpatialFeaturePlot</span>(brain, <span class="at">features =</span> <span class="fu">rownames</span>(cluster11_markers)[<span class="dv">1</span>], <span class="at">pt.size.factor =</span> <span class="dv">3</span>)</span>
<span id="cb113-839"><a href="#cb113-839" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-840"><a href="#cb113-840" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-841"><a href="#cb113-841" aria-hidden="true" tabindex="-1"></a><span class="co"># Addressing NA region</span></span>
<span id="cb113-842"><a href="#cb113-842" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset "N/A" clusters</span></span>
<span id="cb113-843"><a href="#cb113-843" aria-hidden="true" tabindex="-1"></a>na_clusters <span class="ot">&lt;-</span> <span class="fu">subset</span>(brain, brain_region <span class="sc">==</span> <span class="st">"N/A"</span>)</span>
<span id="cb113-844"><a href="#cb113-844" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-845"><a href="#cb113-845" aria-hidden="true" tabindex="-1"></a><span class="co"># Find markers for "N/A" clusters</span></span>
<span id="cb113-846"><a href="#cb113-846" aria-hidden="true" tabindex="-1"></a>na_markers <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(na_clusters, <span class="at">only.pos =</span> <span class="cn">TRUE</span>)</span>
<span id="cb113-847"><a href="#cb113-847" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(na_markers)</span>
<span id="cb113-848"><a href="#cb113-848" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-849"><a href="#cb113-849" aria-hidden="true" tabindex="-1"></a><span class="co"># Check spatial distribution</span></span>
<span id="cb113-850"><a href="#cb113-850" aria-hidden="true" tabindex="-1"></a><span class="fu">SpatialDimPlot</span>(na_clusters, <span class="at">pt.size.factor =</span> <span class="dv">3</span>, <span class="at">image.alpha =</span> <span class="fl">0.5</span>)</span>
<span id="cb113-851"><a href="#cb113-851" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-852"><a href="#cb113-852" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-853"><a href="#cb113-853" aria-hidden="true" tabindex="-1"></a><span class="co"># Test if cluster markers are enriched in specific brain regions</span></span>
<span id="cb113-854"><a href="#cb113-854" aria-hidden="true" tabindex="-1"></a>cluster0_markers <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(brain, <span class="at">ident.1 =</span> <span class="dv">0</span>, <span class="at">only.pos =</span> <span class="cn">TRUE</span>)</span>
<span id="cb113-855"><a href="#cb113-855" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(cluster0_markers)[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]  <span class="co"># Top 50 markers</span></span>
<span id="cb113-856"><a href="#cb113-856" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-857"><a href="#cb113-857" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-858"><a href="#cb113-858" aria-hidden="true" tabindex="-1"></a><span class="co"># Quality Control of Annotation</span></span>
<span id="cb113-859"><a href="#cb113-859" aria-hidden="true" tabindex="-1"></a><span class="co"># 1️⃣ Check if any region is overrepresented or missing</span></span>
<span id="cb113-860"><a href="#cb113-860" aria-hidden="true" tabindex="-1"></a><span class="fu">prop.table</span>(<span class="fu">table</span>(brain<span class="sc">$</span>brain_region)) <span class="sc">*</span> <span class="dv">100</span>  <span class="co"># Get percentage of spots per region</span></span>
<span id="cb113-861"><a href="#cb113-861" aria-hidden="true" tabindex="-1"></a><span class="co"># 2️⃣ Check expression of key marker genes per region</span></span>
<span id="cb113-862"><a href="#cb113-862" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-863"><a href="#cb113-863" aria-hidden="true" tabindex="-1"></a><span class="co"># Neocortex region:</span></span>
<span id="cb113-864"><a href="#cb113-864" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-865"><a href="#cb113-865" aria-hidden="true" tabindex="-1"></a><span class="co"># Excitatory neurons: Slc17a7, Rorb, Foxp2, Cux1</span></span>
<span id="cb113-866"><a href="#cb113-866" aria-hidden="true" tabindex="-1"></a><span class="co"># Inhibitory neurons: Gad1, Gad2, Pvalb, Sst, Vip</span></span>
<span id="cb113-867"><a href="#cb113-867" aria-hidden="true" tabindex="-1"></a><span class="co"># Glial cells: Gfap, Aqp4, Aldh1l1</span></span>
<span id="cb113-868"><a href="#cb113-868" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(brain, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"Pvalb"</span>, <span class="st">"Gfap"</span>, <span class="st">"Aldh1l1"</span>), <span class="at">reduction =</span> <span class="st">"umap"</span>)</span>
<span id="cb113-869"><a href="#cb113-869" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-870"><a href="#cb113-870" aria-hidden="true" tabindex="-1"></a><span class="co"># Hippocampal region:</span></span>
<span id="cb113-871"><a href="#cb113-871" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-872"><a href="#cb113-872" aria-hidden="true" tabindex="-1"></a><span class="co"># CA1 pyramidal neurons: Wfs1, Nr3c2, Rgs14</span></span>
<span id="cb113-873"><a href="#cb113-873" aria-hidden="true" tabindex="-1"></a><span class="co"># CA3 pyramidal neurons: Slc17a7, Grik1, Prkcd</span></span>
<span id="cb113-874"><a href="#cb113-874" aria-hidden="true" tabindex="-1"></a><span class="co"># Dentate gyrus (granule cells): Prox1, C1ql2, Neurod1</span></span>
<span id="cb113-875"><a href="#cb113-875" aria-hidden="true" tabindex="-1"></a><span class="co"># FeaturePlot(brain, features = c("Foxp2", "Gad1", "Neurod1"), reduction = "umap")</span></span>
<span id="cb113-876"><a href="#cb113-876" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-877"><a href="#cb113-877" aria-hidden="true" tabindex="-1"></a><span class="co"># Thalamus region:</span></span>
<span id="cb113-878"><a href="#cb113-878" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-879"><a href="#cb113-879" aria-hidden="true" tabindex="-1"></a><span class="co"># Calb1 (Calbindin)</span></span>
<span id="cb113-880"><a href="#cb113-880" aria-hidden="true" tabindex="-1"></a><span class="co"># Vglut2 (Slc17a6) – Excitatory thalamic neurons</span></span>
<span id="cb113-881"><a href="#cb113-881" aria-hidden="true" tabindex="-1"></a><span class="co"># Foxp2 – Sensory relay neurons</span></span>
<span id="cb113-882"><a href="#cb113-882" aria-hidden="true" tabindex="-1"></a><span class="co"># Tbr1 – Thalamocortical projection neurons</span></span>
<span id="cb113-883"><a href="#cb113-883" aria-hidden="true" tabindex="-1"></a><span class="co"># FeaturePlot(brain, features = c("Tbr1", "Slc17a6", "Calb1"), reduction = "umap")</span></span>
<span id="cb113-884"><a href="#cb113-884" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-885"><a href="#cb113-885" aria-hidden="true" tabindex="-1"></a><span class="co"># Hypothalamus region:</span></span>
<span id="cb113-886"><a href="#cb113-886" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-887"><a href="#cb113-887" aria-hidden="true" tabindex="-1"></a><span class="co"># Oxt (Oxytocin)</span></span>
<span id="cb113-888"><a href="#cb113-888" aria-hidden="true" tabindex="-1"></a><span class="co"># Avp (Vasopressin)</span></span>
<span id="cb113-889"><a href="#cb113-889" aria-hidden="true" tabindex="-1"></a><span class="co"># Cartpt (Cocaine- and amphetamine-regulated transcript)</span></span>
<span id="cb113-890"><a href="#cb113-890" aria-hidden="true" tabindex="-1"></a><span class="co"># Hypothalamic nuclei: Pomc, Npy, Hcrt (Orexin), Crh (Corticotropin-releasing hormone)</span></span>
<span id="cb113-891"><a href="#cb113-891" aria-hidden="true" tabindex="-1"></a><span class="co"># FeaturePlot(brain, features = c("Tbr1", "Slc17a6", "Calb1"), reduction = "umap")</span></span>
<span id="cb113-892"><a href="#cb113-892" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-893"><a href="#cb113-893" aria-hidden="true" tabindex="-1"></a><span class="co"># Caudatoputamen (Striatum) region:</span></span>
<span id="cb113-894"><a href="#cb113-894" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-895"><a href="#cb113-895" aria-hidden="true" tabindex="-1"></a><span class="co"># Drd1 (D1 dopamine receptor, direct pathway)</span></span>
<span id="cb113-896"><a href="#cb113-896" aria-hidden="true" tabindex="-1"></a><span class="co"># Drd2 (D2 dopamine receptor, indirect pathway)</span></span>
<span id="cb113-897"><a href="#cb113-897" aria-hidden="true" tabindex="-1"></a><span class="co"># Ppp1r1b (DARPP-32, marker of medium spiny neurons)</span></span>
<span id="cb113-898"><a href="#cb113-898" aria-hidden="true" tabindex="-1"></a><span class="co"># Foxp2 (Regulates motor function in striatal neurons)</span></span>
<span id="cb113-899"><a href="#cb113-899" aria-hidden="true" tabindex="-1"></a><span class="co"># FeaturePlot(brain, features = c("Tbr1", "Slc17a6", "Calb1"), reduction = "umap")</span></span>
<span id="cb113-900"><a href="#cb113-900" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-901"><a href="#cb113-901" aria-hidden="true" tabindex="-1"></a><span class="co"># Allocortex region:</span></span>
<span id="cb113-902"><a href="#cb113-902" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-903"><a href="#cb113-903" aria-hidden="true" tabindex="-1"></a><span class="co"># Rasgrf2 (Piriform cortex)</span></span>
<span id="cb113-904"><a href="#cb113-904" aria-hidden="true" tabindex="-1"></a><span class="co"># Calb2 (Layer-specific cortical neurons)</span></span>
<span id="cb113-905"><a href="#cb113-905" aria-hidden="true" tabindex="-1"></a><span class="co"># Pcp4 (Parvalbumin interneurons)</span></span>
<span id="cb113-906"><a href="#cb113-906" aria-hidden="true" tabindex="-1"></a><span class="co"># FeaturePlot(brain, features = c("Tbr1", "Slc17a6", "Calb1"), reduction = "umap")</span></span>
<span id="cb113-907"><a href="#cb113-907" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-908"><a href="#cb113-908" aria-hidden="true" tabindex="-1"></a><span class="co"># Striatum-like Amygdalar Nuclei region:</span></span>
<span id="cb113-909"><a href="#cb113-909" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-910"><a href="#cb113-910" aria-hidden="true" tabindex="-1"></a><span class="co"># Cplx3, Tac2 (Central amygdala)</span></span>
<span id="cb113-911"><a href="#cb113-911" aria-hidden="true" tabindex="-1"></a><span class="co"># Foxp2 (Basolateral amygdala)</span></span>
<span id="cb113-912"><a href="#cb113-912" aria-hidden="true" tabindex="-1"></a><span class="co"># Drd1, Drd2 (Dopamine receptor expression in striatum-like regions)</span></span>
<span id="cb113-913"><a href="#cb113-913" aria-hidden="true" tabindex="-1"></a><span class="co"># Pvalb, Npy (Interneurons in the amygdala)</span></span>
<span id="cb113-914"><a href="#cb113-914" aria-hidden="true" tabindex="-1"></a><span class="co"># FeaturePlot(brain, features = c("Tbr1", "Slc17a6", "Calb1"), reduction = "umap")</span></span>
<span id="cb113-915"><a href="#cb113-915" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-916"><a href="#cb113-916" aria-hidden="true" tabindex="-1"></a><span class="co"># Fiber Tracts region:</span></span>
<span id="cb113-917"><a href="#cb113-917" aria-hidden="true" tabindex="-1"></a><span class="co"># Marker Genes:</span></span>
<span id="cb113-918"><a href="#cb113-918" aria-hidden="true" tabindex="-1"></a><span class="co"># Mbp (Myelin basic protein, oligodendrocytes)</span></span>
<span id="cb113-919"><a href="#cb113-919" aria-hidden="true" tabindex="-1"></a><span class="co"># Plp1, Cnp (Myelination markers)</span></span>
<span id="cb113-920"><a href="#cb113-920" aria-hidden="true" tabindex="-1"></a><span class="co"># Olig2, Sox10 (Oligodendrocyte precursor cells)</span></span>
<span id="cb113-921"><a href="#cb113-921" aria-hidden="true" tabindex="-1"></a><span class="co"># Cldn11 (Tight junctions in myelin sheaths)</span></span>
<span id="cb113-922"><a href="#cb113-922" aria-hidden="true" tabindex="-1"></a><span class="co"># FeaturePlot(brain, features = c("Tbr1", "Slc17a6", "Calb1"), reduction = "umap")</span></span>
<span id="cb113-923"><a href="#cb113-923" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-924"><a href="#cb113-924" aria-hidden="true" tabindex="-1"></a><span class="do">########## Flextable - Brain regions ########## </span></span>
<span id="cb113-925"><a href="#cb113-925" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-926"><a href="#cb113-926" aria-hidden="true" tabindex="-1"></a><span class="co"># Create frequency table of region counts</span></span>
<span id="cb113-927"><a href="#cb113-927" aria-hidden="true" tabindex="-1"></a>region_counts <span class="ot">&lt;-</span> <span class="fu">table</span>(brain<span class="sc">@</span>meta.data<span class="sc">$</span>brain_region)</span>
<span id="cb113-928"><a href="#cb113-928" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-929"><a href="#cb113-929" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and calculate percentages</span></span>
<span id="cb113-930"><a href="#cb113-930" aria-hidden="true" tabindex="-1"></a>region_table <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(region_counts) <span class="sc">%&gt;%</span></span>
<span id="cb113-931"><a href="#cb113-931" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Region =</span> Var1, <span class="st">`</span><span class="at">Amount of Spots</span><span class="st">`</span> <span class="ot">=</span> Freq) <span class="sc">%&gt;%</span></span>
<span id="cb113-932"><a href="#cb113-932" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb113-933"><a href="#cb113-933" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">% from total</span><span class="st">`</span> <span class="ot">=</span> <span class="st">`</span><span class="at">Amount of Spots</span><span class="st">`</span> <span class="sc">/</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">Amount of Spots</span><span class="st">`</span>),</span>
<span id="cb113-934"><a href="#cb113-934" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">% from total</span><span class="st">`</span> <span class="ot">=</span> scales<span class="sc">::</span><span class="fu">percent</span>(<span class="st">`</span><span class="at">% from total</span><span class="st">`</span>, <span class="at">accuracy =</span> <span class="fl">0.1</span>)</span>
<span id="cb113-935"><a href="#cb113-935" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb113-936"><a href="#cb113-936" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-937"><a href="#cb113-937" aria-hidden="true" tabindex="-1"></a><span class="co"># Add total row</span></span>
<span id="cb113-938"><a href="#cb113-938" aria-hidden="true" tabindex="-1"></a>total_spots <span class="ot">&lt;-</span> <span class="fu">sum</span>(region_table<span class="sc">$</span><span class="st">`</span><span class="at">Amount of Spots</span><span class="st">`</span>)</span>
<span id="cb113-939"><a href="#cb113-939" aria-hidden="true" tabindex="-1"></a>total_row <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb113-940"><a href="#cb113-940" aria-hidden="true" tabindex="-1"></a>  <span class="at">Region =</span> <span class="st">"Total"</span>,</span>
<span id="cb113-941"><a href="#cb113-941" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">Amount of Spots</span><span class="st">`</span> <span class="ot">=</span> total_spots,</span>
<span id="cb113-942"><a href="#cb113-942" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">% from total</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"100.0%"</span>,</span>
<span id="cb113-943"><a href="#cb113-943" aria-hidden="true" tabindex="-1"></a>  <span class="at">check.names =</span> <span class="cn">FALSE</span></span>
<span id="cb113-944"><a href="#cb113-944" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-945"><a href="#cb113-945" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-946"><a href="#cb113-946" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine with original data</span></span>
<span id="cb113-947"><a href="#cb113-947" aria-hidden="true" tabindex="-1"></a>region_table <span class="ot">&lt;-</span> <span class="fu">rbind</span>(region_table, total_row)</span>
<span id="cb113-948"><a href="#cb113-948" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-949"><a href="#cb113-949" aria-hidden="true" tabindex="-1"></a><span class="co"># Create flextable</span></span>
<span id="cb113-950"><a href="#cb113-950" aria-hidden="true" tabindex="-1"></a>ft_region <span class="ot">&lt;-</span> <span class="fu">flextable</span>(region_table) <span class="sc">%&gt;%</span></span>
<span id="cb113-951"><a href="#cb113-951" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_vanilla</span>() <span class="sc">%&gt;%</span></span>
<span id="cb113-952"><a href="#cb113-952" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_header_labels</span>(</span>
<span id="cb113-953"><a href="#cb113-953" aria-hidden="true" tabindex="-1"></a>    <span class="at">Region =</span> <span class="st">"Brain region"</span>,</span>
<span id="cb113-954"><a href="#cb113-954" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">Amount of Spots</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Amount of Spots"</span>,</span>
<span id="cb113-955"><a href="#cb113-955" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">% from total</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"% from total"</span></span>
<span id="cb113-956"><a href="#cb113-956" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb113-957"><a href="#cb113-957" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_header_lines</span>(<span class="at">values =</span> <span class="st">"Regional transcriptomic representation of mouse brain section (coronal)*"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-958"><a href="#cb113-958" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_footer_lines</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"* Source: https://www.10xgenomics.com/datasets/mouse-brain-section-coronal-1-standard-1-0-0"</span>, </span>
<span id="cb113-959"><a href="#cb113-959" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"* Platform: Visium Spatial 10X Genomics"</span>,</span>
<span id="cb113-960"><a href="#cb113-960" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"* Preservation method: Fresh Frozen"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb113-961"><a href="#cb113-961" aria-hidden="true" tabindex="-1"></a>  <span class="fu">align</span>(<span class="at">align =</span> <span class="st">"center"</span>, <span class="at">part =</span> <span class="st">"header"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-962"><a href="#cb113-962" aria-hidden="true" tabindex="-1"></a>  <span class="fu">align</span>(<span class="at">align =</span> <span class="st">"center"</span>, <span class="at">part =</span> <span class="st">"body"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-963"><a href="#cb113-963" aria-hidden="true" tabindex="-1"></a>  <span class="fu">align</span>(<span class="at">align =</span> <span class="st">"left"</span>, <span class="at">part =</span> <span class="st">"footer"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-964"><a href="#cb113-964" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bold</span>(<span class="sc">~</span> Region <span class="sc">==</span> <span class="st">"Total"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-965"><a href="#cb113-965" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bg</span>(<span class="sc">~</span> Region <span class="sc">==</span> <span class="st">"Total"</span>, <span class="at">bg =</span> <span class="st">"#2D2D2D"</span>, <span class="at">part =</span> <span class="st">"body"</span>) <span class="sc">%&gt;%</span>  </span>
<span id="cb113-966"><a href="#cb113-966" aria-hidden="true" tabindex="-1"></a>  <span class="fu">color</span>(<span class="sc">~</span> Region <span class="sc">==</span> <span class="st">"Total"</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">part =</span> <span class="st">"body"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-967"><a href="#cb113-967" aria-hidden="true" tabindex="-1"></a>  <span class="fu">border_inner_h</span>(<span class="at">border =</span> <span class="fu">fp_border</span>(<span class="at">color =</span> <span class="st">"#D3D3D3"</span>, <span class="at">width =</span> <span class="fl">0.5</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb113-968"><a href="#cb113-968" aria-hidden="true" tabindex="-1"></a>  <span class="fu">border_outer</span>(<span class="at">border =</span> <span class="fu">fp_border</span>(<span class="at">color =</span> <span class="st">"#D3D3D3"</span>, <span class="at">width =</span> <span class="fl">0.5</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb113-969"><a href="#cb113-969" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fontsize</span>(<span class="at">size =</span> <span class="dv">16</span>, <span class="at">part =</span> <span class="st">"all"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-970"><a href="#cb113-970" aria-hidden="true" tabindex="-1"></a>  <span class="fu">font</span>(<span class="at">fontname =</span> <span class="st">"Arial"</span>, <span class="at">part =</span> <span class="st">"all"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb113-971"><a href="#cb113-971" aria-hidden="true" tabindex="-1"></a>  <span class="fu">autofit</span>()</span>
<span id="cb113-972"><a href="#cb113-972" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-973"><a href="#cb113-973" aria-hidden="true" tabindex="-1"></a><span class="co"># print(ft_region)</span></span>
<span id="cb113-974"><a href="#cb113-974" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-975"><a href="#cb113-975" aria-hidden="true" tabindex="-1"></a><span class="co"># Save 'ft_region'</span></span>
<span id="cb113-976"><a href="#cb113-976" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert flextable to a grob</span></span>
<span id="cb113-977"><a href="#cb113-977" aria-hidden="true" tabindex="-1"></a><span class="co"># ft_grob2 &lt;- flextable::gen_grob(ft_region)</span></span>
<span id="cb113-978"><a href="#cb113-978" aria-hidden="true" tabindex="-1"></a><span class="co"># ggsave("~path/to/your/data/ft_region_brainregions_descriptive_table.png", ft_grob2, width = 10, height = 8, dpi = 600)</span></span>
<span id="cb113-979"><a href="#cb113-979" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-980"><a href="#cb113-980" aria-hidden="true" tabindex="-1"></a><span class="co"># view(brain@meta.data)</span></span>
<span id="cb113-981"><a href="#cb113-981" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-982"><a href="#cb113-982" aria-hidden="true" tabindex="-1"></a><span class="co"># Render flextable for regions</span></span>
<span id="cb113-983"><a href="#cb113-983" aria-hidden="true" tabindex="-1"></a>flextable<span class="sc">::</span><span class="fu">htmltools_value</span>(ft_region)</span>
<span id="cb113-984"><a href="#cb113-984" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-985"><a href="#cb113-985" aria-hidden="true" tabindex="-1"></a><span class="do">#################</span></span>
<span id="cb113-986"><a href="#cb113-986" aria-hidden="true" tabindex="-1"></a><span class="do">#################</span></span>
<span id="cb113-987"><a href="#cb113-987" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify the top markers for your specific dataset</span></span>
<span id="cb113-988"><a href="#cb113-988" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. ROC Analysis (AUC calculation)</span></span>
<span id="cb113-989"><a href="#cb113-989" aria-hidden="true" tabindex="-1"></a>top_markers_roc <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(</span>
<span id="cb113-990"><a href="#cb113-990" aria-hidden="true" tabindex="-1"></a>  brain,</span>
<span id="cb113-991"><a href="#cb113-991" aria-hidden="true" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">"brain_region"</span>,</span>
<span id="cb113-992"><a href="#cb113-992" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"roc"</span>,            <span class="co"># ROC test for AUC values</span></span>
<span id="cb113-993"><a href="#cb113-993" aria-hidden="true" tabindex="-1"></a>  <span class="at">logfc.threshold =</span> <span class="fl">0.25</span>,</span>
<span id="cb113-994"><a href="#cb113-994" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> <span class="cn">TRUE</span>,</span>
<span id="cb113-995"><a href="#cb113-995" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.1</span>                <span class="co"># Gene detected in ≥10% of spots in either region</span></span>
<span id="cb113-996"><a href="#cb113-996" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-997"><a href="#cb113-997" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-998"><a href="#cb113-998" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Get top 5 markers per region (sorted by AUC power)</span></span>
<span id="cb113-999"><a href="#cb113-999" aria-hidden="true" tabindex="-1"></a>top5_markers <span class="ot">&lt;-</span> top_markers_roc <span class="sc">%&gt;%</span></span>
<span id="cb113-1000"><a href="#cb113-1000" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span>        <span class="co"># "cluster" = brain_region</span></span>
<span id="cb113-1001"><a href="#cb113-1001" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(power)) <span class="sc">%&gt;%</span>     <span class="co"># "power" = AUC (0.5 = random, 1 = perfect)</span></span>
<span id="cb113-1002"><a href="#cb113-1002" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">5</span>)            <span class="co"># Top 5 per region</span></span>
<span id="cb113-1003"><a href="#cb113-1003" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1004"><a href="#cb113-1004" aria-hidden="true" tabindex="-1"></a><span class="co"># View results</span></span>
<span id="cb113-1005"><a href="#cb113-1005" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(top5_markers, <span class="at">n =</span> <span class="dv">45</span>)</span>
<span id="cb113-1006"><a href="#cb113-1006" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1007"><a href="#cb113-1007" aria-hidden="true" tabindex="-1"></a><span class="co"># Dot plot of top markers (expression + percentage)</span></span>
<span id="cb113-1008"><a href="#cb113-1008" aria-hidden="true" tabindex="-1"></a><span class="fu">DotPlot</span>(</span>
<span id="cb113-1009"><a href="#cb113-1009" aria-hidden="true" tabindex="-1"></a>  brain,</span>
<span id="cb113-1010"><a href="#cb113-1010" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="fu">unique</span>(top5_markers<span class="sc">$</span>gene),</span>
<span id="cb113-1011"><a href="#cb113-1011" aria-hidden="true" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">"brain_region"</span>,</span>
<span id="cb113-1012"><a href="#cb113-1012" aria-hidden="true" tabindex="-1"></a>  <span class="at">dot.scale =</span> <span class="dv">8</span>,</span>
<span id="cb113-1013"><a href="#cb113-1013" aria-hidden="true" tabindex="-1"></a>  <span class="at">col.min =</span> <span class="dv">0</span>,</span>
<span id="cb113-1014"><a href="#cb113-1014" aria-hidden="true" tabindex="-1"></a>  <span class="at">col.max =</span> <span class="dv">3</span></span>
<span id="cb113-1015"><a href="#cb113-1015" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb113-1016"><a href="#cb113-1016" aria-hidden="true" tabindex="-1"></a>  <span class="fu">RotatedAxis</span>() <span class="sc">+</span></span>
<span id="cb113-1017"><a href="#cb113-1017" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis_c</span>(<span class="at">option =</span> <span class="st">"magma"</span>) <span class="sc">+</span></span>
<span id="cb113-1018"><a href="#cb113-1018" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb113-1019"><a href="#cb113-1019" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb113-1020"><a href="#cb113-1020" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"right"</span></span>
<span id="cb113-1021"><a href="#cb113-1021" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb113-1022"><a href="#cb113-1022" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb113-1023"><a href="#cb113-1023" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Genes"</span>,</span>
<span id="cb113-1024"><a href="#cb113-1024" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Brain Region"</span>,</span>
<span id="cb113-1025"><a href="#cb113-1025" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Avg. Expression"</span>,</span>
<span id="cb113-1026"><a href="#cb113-1026" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="st">"% Spots Expressed"</span></span>
<span id="cb113-1027"><a href="#cb113-1027" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb113-1028"><a href="#cb113-1028" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1029"><a href="#cb113-1029" aria-hidden="true" tabindex="-1"></a><span class="co"># DEGs between Neocortex &amp; Thalamus</span></span>
<span id="cb113-1030"><a href="#cb113-1030" aria-hidden="true" tabindex="-1"></a>neocortex_vs_thalamus <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(</span>
<span id="cb113-1031"><a href="#cb113-1031" aria-hidden="true" tabindex="-1"></a>  brain, </span>
<span id="cb113-1032"><a href="#cb113-1032" aria-hidden="true" tabindex="-1"></a>  <span class="at">ident.1 =</span> <span class="st">"Neocortex"</span>, </span>
<span id="cb113-1033"><a href="#cb113-1033" aria-hidden="true" tabindex="-1"></a>  <span class="at">ident.2 =</span> <span class="st">"Thalamus"</span>, </span>
<span id="cb113-1034"><a href="#cb113-1034" aria-hidden="true" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">"brain_region"</span>, </span>
<span id="cb113-1035"><a href="#cb113-1035" aria-hidden="true" tabindex="-1"></a>  <span class="at">logfc.threshold =</span> <span class="fl">0.25</span>, </span>
<span id="cb113-1036"><a href="#cb113-1036" aria-hidden="true" tabindex="-1"></a>  <span class="at">test.use =</span> <span class="st">"wilcox"</span></span>
<span id="cb113-1037"><a href="#cb113-1037" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-1038"><a href="#cb113-1038" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1039"><a href="#cb113-1039" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(neocortex_vs_thalamus)</span>
<span id="cb113-1040"><a href="#cb113-1040" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1041"><a href="#cb113-1041" aria-hidden="true" tabindex="-1"></a><span class="co"># library(clusterProfiler)</span></span>
<span id="cb113-1042"><a href="#cb113-1042" aria-hidden="true" tabindex="-1"></a><span class="co"># library(org.Mm.eg.db)  # Mouse gene annotations</span></span>
<span id="cb113-1043"><a href="#cb113-1043" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1044"><a href="#cb113-1044" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert top genes to ENTREZ IDs</span></span>
<span id="cb113-1045"><a href="#cb113-1045" aria-hidden="true" tabindex="-1"></a>genes_entrez <span class="ot">&lt;-</span> <span class="fu">mapIds</span>(</span>
<span id="cb113-1046"><a href="#cb113-1046" aria-hidden="true" tabindex="-1"></a>  org.Mm.eg.db, </span>
<span id="cb113-1047"><a href="#cb113-1047" aria-hidden="true" tabindex="-1"></a>  <span class="at">keys =</span> top5_markers<span class="sc">$</span>gene, </span>
<span id="cb113-1048"><a href="#cb113-1048" aria-hidden="true" tabindex="-1"></a>  <span class="at">column =</span> <span class="st">"ENTREZID"</span>, </span>
<span id="cb113-1049"><a href="#cb113-1049" aria-hidden="true" tabindex="-1"></a>  <span class="at">keytype =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb113-1050"><a href="#cb113-1050" aria-hidden="true" tabindex="-1"></a>  <span class="at">multiVals =</span> <span class="st">"first"</span></span>
<span id="cb113-1051"><a href="#cb113-1051" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-1052"><a href="#cb113-1052" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1053"><a href="#cb113-1053" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform GO enrichment (Biological Process)</span></span>
<span id="cb113-1054"><a href="#cb113-1054" aria-hidden="true" tabindex="-1"></a>go_results <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(</span>
<span id="cb113-1055"><a href="#cb113-1055" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene =</span> <span class="fu">na.omit</span>(genes_entrez), </span>
<span id="cb113-1056"><a href="#cb113-1056" aria-hidden="true" tabindex="-1"></a>  <span class="at">OrgDb =</span> org.Mm.eg.db, </span>
<span id="cb113-1057"><a href="#cb113-1057" aria-hidden="true" tabindex="-1"></a>  <span class="at">keyType =</span> <span class="st">"ENTREZID"</span>, </span>
<span id="cb113-1058"><a href="#cb113-1058" aria-hidden="true" tabindex="-1"></a>  <span class="at">ont =</span> <span class="st">"BP"</span>, </span>
<span id="cb113-1059"><a href="#cb113-1059" aria-hidden="true" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span></span>
<span id="cb113-1060"><a href="#cb113-1060" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb113-1061"><a href="#cb113-1061" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1062"><a href="#cb113-1062" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot top GO terms</span></span>
<span id="cb113-1063"><a href="#cb113-1063" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(go_results, <span class="at">showCategory =</span> <span class="dv">10</span>, <span class="at">title =</span> <span class="st">"GO Enrichment: Brain Regions"</span>)</span>
<span id="cb113-1064"><a href="#cb113-1064" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1065"><a href="#cb113-1065" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1066"><a href="#cb113-1066" aria-hidden="true" tabindex="-1"></a><span class="do">######### Flextable of highly expressed genes per brain region ###########</span></span>
<span id="cb113-1067"><a href="#cb113-1067" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1068"><a href="#cb113-1068" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare table</span></span>
<span id="cb113-1069"><a href="#cb113-1069" aria-hidden="true" tabindex="-1"></a>marker_table <span class="ot">&lt;-</span> top5_markers <span class="sc">%&gt;%</span></span>
<span id="cb113-1070"><a href="#cb113-1070" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(gene, cluster, power, avg_log2FC) <span class="sc">%&gt;%</span></span>
<span id="cb113-1071"><a href="#cb113-1071" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(</span>
<span id="cb113-1072"><a href="#cb113-1072" aria-hidden="true" tabindex="-1"></a>    <span class="at">Gene =</span> gene,</span>
<span id="cb113-1073"><a href="#cb113-1073" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">Brain Region</span><span class="st">`</span> <span class="ot">=</span> cluster,</span>
<span id="cb113-1074"><a href="#cb113-1074" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">AUC Power</span><span class="st">`</span> <span class="ot">=</span> power,</span>
<span id="cb113-1075"><a href="#cb113-1075" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">Log2 FC</span><span class="st">`</span> <span class="ot">=</span> avg_log2FC</span>
<span id="cb113-1076"><a href="#cb113-1076" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb113-1077"><a href="#cb113-1077" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">AUC Power</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">round</span>(<span class="st">`</span><span class="at">AUC Power</span><span class="st">`</span>, <span class="dv">2</span>))  <span class="co"># Round AUC to 2 decimals</span></span>
<span id="cb113-1078"><a href="#cb113-1078" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1079"><a href="#cb113-1079" aria-hidden="true" tabindex="-1"></a><span class="co"># Create flextable</span></span>
<span id="cb113-1080"><a href="#cb113-1080" aria-hidden="true" tabindex="-1"></a>ft_markers <span class="ot">&lt;-</span> <span class="fu">flextable</span>(marker_table) <span class="sc">%&gt;%</span></span>
<span id="cb113-1081"><a href="#cb113-1081" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_vanilla</span>() <span class="sc">%&gt;%</span></span>
<span id="cb113-1082"><a href="#cb113-1082" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_header_labels</span>(</span>
<span id="cb113-1083"><a href="#cb113-1083" aria-hidden="true" tabindex="-1"></a>    <span class="at">Gene =</span> <span class="st">"Gene"</span>,</span>
<span id="cb113-1084"><a href="#cb113-1084" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">Brain Region</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Brain Region"</span>,</span>
<span id="cb113-1085"><a href="#cb113-1085" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">AUC Power</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"AUC Power"</span>,</span>
<span id="cb113-1086"><a href="#cb113-1086" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">Log2 FC</span><span class="st">`</span> <span class="ot">=</span> <span class="st">"Log2 Fold Change"</span></span>
<span id="cb113-1087"><a href="#cb113-1087" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb113-1088"><a href="#cb113-1088" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_header_lines</span>(<span class="at">values =</span> <span class="st">"Top 5 Predictive Genes per Brain Region (ROC Analysis)*"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-1089"><a href="#cb113-1089" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_footer_lines</span>(<span class="at">values =</span> <span class="st">"*AUC Power: 1 = perfect classifier, 0.5 = random."</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-1090"><a href="#cb113-1090" aria-hidden="true" tabindex="-1"></a>  <span class="fu">align</span>(<span class="at">align =</span> <span class="st">"center"</span>, <span class="at">part =</span> <span class="st">"header"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-1091"><a href="#cb113-1091" aria-hidden="true" tabindex="-1"></a>  <span class="fu">align</span>(<span class="at">align =</span> <span class="st">"center"</span>, <span class="at">part =</span> <span class="st">"body"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-1092"><a href="#cb113-1092" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bg</span>(<span class="sc">~</span> <span class="st">`</span><span class="at">AUC Power</span><span class="st">`</span> <span class="sc">&gt;</span> <span class="fl">0.9</span>, <span class="at">bg =</span> <span class="st">"#E6F3FF"</span>, <span class="at">part =</span> <span class="st">"body"</span>) <span class="sc">%&gt;%</span>  <span class="co"># Highlight high AUC</span></span>
<span id="cb113-1093"><a href="#cb113-1093" aria-hidden="true" tabindex="-1"></a>  <span class="fu">border_outer</span>(<span class="at">border =</span> <span class="fu">fp_border</span>(<span class="at">color =</span> <span class="st">"#D3D3D3"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb113-1094"><a href="#cb113-1094" aria-hidden="true" tabindex="-1"></a>  <span class="fu">font</span>(<span class="at">fontname =</span> <span class="st">"Arial"</span>, <span class="at">part =</span> <span class="st">"all"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb113-1095"><a href="#cb113-1095" aria-hidden="true" tabindex="-1"></a>  <span class="fu">autofit</span>()</span>
<span id="cb113-1096"><a href="#cb113-1096" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1097"><a href="#cb113-1097" aria-hidden="true" tabindex="-1"></a><span class="co"># print(ft_markers)</span></span>
<span id="cb113-1098"><a href="#cb113-1098" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1099"><a href="#cb113-1099" aria-hidden="true" tabindex="-1"></a><span class="co"># Save 'ft_markers'</span></span>
<span id="cb113-1100"><a href="#cb113-1100" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert flextable to a grob</span></span>
<span id="cb113-1101"><a href="#cb113-1101" aria-hidden="true" tabindex="-1"></a><span class="co"># ft_grob2 &lt;- flextable::gen_grob(ft_markers)</span></span>
<span id="cb113-1102"><a href="#cb113-1102" aria-hidden="true" tabindex="-1"></a><span class="co"># ggsave("~path/to/your/data/ft_markers_brainregions_Top5_predictive_genes_table.png", ft_grob2, width = 10, height = 8, dpi = 600)</span></span>
<span id="cb113-1103"><a href="#cb113-1103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1104"><a href="#cb113-1104" aria-hidden="true" tabindex="-1"></a><span class="co"># Render flextable for markers</span></span>
<span id="cb113-1105"><a href="#cb113-1105" aria-hidden="true" tabindex="-1"></a>flextable<span class="sc">::</span><span class="fu">htmltools_value</span>(ft_markers)</span>
<span id="cb113-1106"><a href="#cb113-1106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1107"><a href="#cb113-1107" aria-hidden="true" tabindex="-1"></a><span class="co"># view(brain@meta.data)</span></span>
<span id="cb113-1108"><a href="#cb113-1108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1109"><a href="#cb113-1109" aria-hidden="true" tabindex="-1"></a><span class="co"># Save Seurat object 'with markers'brain'</span></span>
<span id="cb113-1110"><a href="#cb113-1110" aria-hidden="true" tabindex="-1"></a><span class="co"># saveRDS(brain, file = "~/path/to/your/data/Spatial_transcript_Visium_mouse_brain_regions.rds")</span></span>
<span id="cb113-1111"><a href="#cb113-1111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1112"><a href="#cb113-1112" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb113-1113"><a href="#cb113-1113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1114"><a href="#cb113-1114" aria-hidden="true" tabindex="-1"></a><span class="fu">## *Converting Seurat object "brain" into H5AD format*</span></span>
<span id="cb113-1115"><a href="#cb113-1115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1118"><a href="#cb113-1118" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb113-1119"><a href="#cb113-1119" aria-hidden="true" tabindex="-1"></a><span class="co"># # Activate the package</span></span>
<span id="cb113-1120"><a href="#cb113-1120" aria-hidden="true" tabindex="-1"></a><span class="co"># library(SeuratDisk)</span></span>
<span id="cb113-1121"><a href="#cb113-1121" aria-hidden="true" tabindex="-1"></a><span class="co"># library(reticulate)</span></span>
<span id="cb113-1122"><a href="#cb113-1122" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-1123"><a href="#cb113-1123" aria-hidden="true" tabindex="-1"></a><span class="co"># library(Seurat) # &lt;-- Critical! This provides the method for Seurat → SingleCellExperiment</span></span>
<span id="cb113-1124"><a href="#cb113-1124" aria-hidden="true" tabindex="-1"></a><span class="co"># library(SingleCellExperiment)</span></span>
<span id="cb113-1125"><a href="#cb113-1125" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-1126"><a href="#cb113-1126" aria-hidden="true" tabindex="-1"></a><span class="co"># # Start conversion to H5AD:</span></span>
<span id="cb113-1127"><a href="#cb113-1127" aria-hidden="true" tabindex="-1"></a><span class="co"># # 1. Remove problematic layers (scale.data causes HDF5 issues)</span></span>
<span id="cb113-1128"><a href="#cb113-1128" aria-hidden="true" tabindex="-1"></a><span class="co"># pbmc_postinfusion_idecel[["RNA"]] &lt;- subset(</span></span>
<span id="cb113-1129"><a href="#cb113-1129" aria-hidden="true" tabindex="-1"></a><span class="co">#   pbmc_postinfusion_idecel[["RNA"]],</span></span>
<span id="cb113-1130"><a href="#cb113-1130" aria-hidden="true" tabindex="-1"></a><span class="co">#   layers = c("counts", "data")  # Keep only counts/data</span></span>
<span id="cb113-1131"><a href="#cb113-1131" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb113-1132"><a href="#cb113-1132" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-1133"><a href="#cb113-1133" aria-hidden="true" tabindex="-1"></a><span class="co"># # 2. Convert to SingleCellExperiment and to H5AD</span></span>
<span id="cb113-1134"><a href="#cb113-1134" aria-hidden="true" tabindex="-1"></a><span class="co"># sce &lt;- as.SingleCellExperiment(pbmc_postinfusion_idecel)</span></span>
<span id="cb113-1135"><a href="#cb113-1135" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-1136"><a href="#cb113-1136" aria-hidden="true" tabindex="-1"></a><span class="co"># library(zellkonverter)</span></span>
<span id="cb113-1137"><a href="#cb113-1137" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-1138"><a href="#cb113-1138" aria-hidden="true" tabindex="-1"></a><span class="co"># writeH5AD(sce, file = "./Data_Grieb_CAR_Zenodo/pbmc_postinfusion_idecel.h5ad")</span></span>
<span id="cb113-1139"><a href="#cb113-1139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1140"><a href="#cb113-1140" aria-hidden="true" tabindex="-1"></a>brain</span>
<span id="cb113-1141"><a href="#cb113-1141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1142"><a href="#cb113-1142" aria-hidden="true" tabindex="-1"></a><span class="do">########### METHOD 1 #############</span></span>
<span id="cb113-1143"><a href="#cb113-1143" aria-hidden="true" tabindex="-1"></a><span class="co"># Load required packages</span></span>
<span id="cb113-1144"><a href="#cb113-1144" aria-hidden="true" tabindex="-1"></a><span class="co"># library(Seurat)</span></span>
<span id="cb113-1145"><a href="#cb113-1145" aria-hidden="true" tabindex="-1"></a><span class="co"># library(SingleCellExperiment)</span></span>
<span id="cb113-1146"><a href="#cb113-1146" aria-hidden="true" tabindex="-1"></a><span class="co"># library(zellkonverter)</span></span>
<span id="cb113-1147"><a href="#cb113-1147" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-1148"><a href="#cb113-1148" aria-hidden="true" tabindex="-1"></a><span class="co"># # Remove problematic layers (scale.data can cause HDF5 issues)</span></span>
<span id="cb113-1149"><a href="#cb113-1149" aria-hidden="true" tabindex="-1"></a><span class="co"># brain[["SCT"]] &lt;- subset(</span></span>
<span id="cb113-1150"><a href="#cb113-1150" aria-hidden="true" tabindex="-1"></a><span class="co">#   brain[["SCT"]], </span></span>
<span id="cb113-1151"><a href="#cb113-1151" aria-hidden="true" tabindex="-1"></a><span class="co">#   layers = c("counts", "data")  # Keep only essential layers</span></span>
<span id="cb113-1152"><a href="#cb113-1152" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb113-1153"><a href="#cb113-1153" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-1154"><a href="#cb113-1154" aria-hidden="true" tabindex="-1"></a><span class="co"># # Convert to SingleCellExperiment</span></span>
<span id="cb113-1155"><a href="#cb113-1155" aria-hidden="true" tabindex="-1"></a><span class="co"># sce &lt;- as.SingleCellExperiment(brain)</span></span>
<span id="cb113-1156"><a href="#cb113-1156" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-1157"><a href="#cb113-1157" aria-hidden="true" tabindex="-1"></a><span class="co"># # Save to H5AD (AnnData format)</span></span>
<span id="cb113-1158"><a href="#cb113-1158" aria-hidden="true" tabindex="-1"></a><span class="co"># writeH5AD(</span></span>
<span id="cb113-1159"><a href="#cb113-1159" aria-hidden="true" tabindex="-1"></a><span class="co">#   sce, </span></span>
<span id="cb113-1160"><a href="#cb113-1160" aria-hidden="true" tabindex="-1"></a><span class="co">#   file = "~path/to/your/data/Spatial_Data_MouseBrainCoronal/brain_metadata.h5ad",</span></span>
<span id="cb113-1161"><a href="#cb113-1161" aria-hidden="true" tabindex="-1"></a><span class="co">#   verbose = TRUE  # Monitor conversion progress</span></span>
<span id="cb113-1162"><a href="#cb113-1162" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb113-1163"><a href="#cb113-1163" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-1164"><a href="#cb113-1164" aria-hidden="true" tabindex="-1"></a><span class="co"># ########### METHOD 2 #############</span></span>
<span id="cb113-1165"><a href="#cb113-1165" aria-hidden="true" tabindex="-1"></a><span class="co"># # Load required packages</span></span>
<span id="cb113-1166"><a href="#cb113-1166" aria-hidden="true" tabindex="-1"></a><span class="co"># library(Seurat)</span></span>
<span id="cb113-1167"><a href="#cb113-1167" aria-hidden="true" tabindex="-1"></a><span class="co"># library(SingleCellExperiment)</span></span>
<span id="cb113-1168"><a href="#cb113-1168" aria-hidden="true" tabindex="-1"></a><span class="co"># library(zellkonverter)</span></span>
<span id="cb113-1169"><a href="#cb113-1169" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-1170"><a href="#cb113-1170" aria-hidden="true" tabindex="-1"></a><span class="co"># # Clean up layers properly</span></span>
<span id="cb113-1171"><a href="#cb113-1171" aria-hidden="true" tabindex="-1"></a><span class="co"># brain &lt;- DietSeurat(</span></span>
<span id="cb113-1172"><a href="#cb113-1172" aria-hidden="true" tabindex="-1"></a><span class="co">#   brain,</span></span>
<span id="cb113-1173"><a href="#cb113-1173" aria-hidden="true" tabindex="-1"></a><span class="co">#   layers = c("counts", "data"),</span></span>
<span id="cb113-1174"><a href="#cb113-1174" aria-hidden="true" tabindex="-1"></a><span class="co">#   assays = c("SCT", "Spatial")</span></span>
<span id="cb113-1175"><a href="#cb113-1175" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb113-1176"><a href="#cb113-1176" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-1177"><a href="#cb113-1177" aria-hidden="true" tabindex="-1"></a><span class="co"># # Convert to SingleCellExperiment</span></span>
<span id="cb113-1178"><a href="#cb113-1178" aria-hidden="true" tabindex="-1"></a><span class="co"># sce &lt;- as.SingleCellExperiment(brain)</span></span>
<span id="cb113-1179"><a href="#cb113-1179" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-1180"><a href="#cb113-1180" aria-hidden="true" tabindex="-1"></a><span class="co"># # Add spatial coordinates</span></span>
<span id="cb113-1181"><a href="#cb113-1181" aria-hidden="true" tabindex="-1"></a><span class="co"># spatial_coords &lt;- GetTissueCoordinates(brain)</span></span>
<span id="cb113-1182"><a href="#cb113-1182" aria-hidden="true" tabindex="-1"></a><span class="co"># colData(sce)$imagecol &lt;- spatial_coords$imagecol</span></span>
<span id="cb113-1183"><a href="#cb113-1183" aria-hidden="true" tabindex="-1"></a><span class="co"># colData(sce)$imagerow &lt;- spatial_coords$imagerow</span></span>
<span id="cb113-1184"><a href="#cb113-1184" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-1185"><a href="#cb113-1185" aria-hidden="true" tabindex="-1"></a><span class="co"># # Save to H5AD</span></span>
<span id="cb113-1186"><a href="#cb113-1186" aria-hidden="true" tabindex="-1"></a><span class="co"># writeH5AD(sce, file = "~/path/to/your/data/brain_metadata_corrected.h5ad", verbose = TRUE)</span></span>
<span id="cb113-1187"><a href="#cb113-1187" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-1188"><a href="#cb113-1188" aria-hidden="true" tabindex="-1"></a><span class="co"># # Check metadata transfer</span></span>
<span id="cb113-1189"><a href="#cb113-1189" aria-hidden="true" tabindex="-1"></a><span class="co"># head(colData(sce))  # Should match brain@meta.data</span></span>
<span id="cb113-1190"><a href="#cb113-1190" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb113-1191"><a href="#cb113-1191" aria-hidden="true" tabindex="-1"></a><span class="co"># # Check spatial coordinates</span></span>
<span id="cb113-1192"><a href="#cb113-1192" aria-hidden="true" tabindex="-1"></a><span class="co"># head(colData(sce)$imagecol)</span></span>
<span id="cb113-1193"><a href="#cb113-1193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1194"><a href="#cb113-1194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1195"><a href="#cb113-1195" aria-hidden="true" tabindex="-1"></a><span class="do">########### METHOD 3: THE CORRECT WAY TO SAVE SPATIAL METADATA AS H5AD FORMAT ##############</span></span>
<span id="cb113-1196"><a href="#cb113-1196" aria-hidden="true" tabindex="-1"></a><span class="do">########### THIS METHOD INCLUDES THE INCORPORATION OF SPATIAL COORDINATES ########### </span></span>
<span id="cb113-1197"><a href="#cb113-1197" aria-hidden="true" tabindex="-1"></a><span class="co"># Load required packages</span></span>
<span id="cb113-1198"><a href="#cb113-1198" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb113-1199"><a href="#cb113-1199" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SpatialExperiment)</span>
<span id="cb113-1200"><a href="#cb113-1200" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(zellkonverter)</span>
<span id="cb113-1201"><a href="#cb113-1201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1202"><a href="#cb113-1202" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract spatial coordinates</span></span>
<span id="cb113-1203"><a href="#cb113-1203" aria-hidden="true" tabindex="-1"></a><span class="co"># spatial_coords &lt;- GetTissueCoordinates(brain)  # Columns are x, y, cell</span></span>
<span id="cb113-1204"><a href="#cb113-1204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1205"><a href="#cb113-1205" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to SpatialExperiment</span></span>
<span id="cb113-1206"><a href="#cb113-1206" aria-hidden="true" tabindex="-1"></a><span class="co"># spe &lt;- SpatialExperiment(</span></span>
<span id="cb113-1207"><a href="#cb113-1207" aria-hidden="true" tabindex="-1"></a><span class="co">#   assays = list(</span></span>
<span id="cb113-1208"><a href="#cb113-1208" aria-hidden="true" tabindex="-1"></a><span class="co">#     counts = LayerData(brain, assay = "SCT", layer = "counts"),  # SCT counts</span></span>
<span id="cb113-1209"><a href="#cb113-1209" aria-hidden="true" tabindex="-1"></a><span class="co">#     logcounts = LayerData(brain, assay = "SCT", layer = "data")   # SCT normalized data</span></span>
<span id="cb113-1210"><a href="#cb113-1210" aria-hidden="true" tabindex="-1"></a><span class="co">#   ),</span></span>
<span id="cb113-1211"><a href="#cb113-1211" aria-hidden="true" tabindex="-1"></a><span class="co">#   colData = brain@meta.data,  # Transfer metadata</span></span>
<span id="cb113-1212"><a href="#cb113-1212" aria-hidden="true" tabindex="-1"></a><span class="co">#   spatialCoords = as.matrix(spatial_coords[, c("x", "y")])  # Use x and y columns</span></span>
<span id="cb113-1213"><a href="#cb113-1213" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb113-1214"><a href="#cb113-1214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1215"><a href="#cb113-1215" aria-hidden="true" tabindex="-1"></a><span class="co"># ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span></span>
<span id="cb113-1216"><a href="#cb113-1216" aria-hidden="true" tabindex="-1"></a><span class="co"># KEY FIX: Add coordinates to reducedDims for proper H5AD mapping</span></span>
<span id="cb113-1217"><a href="#cb113-1217" aria-hidden="true" tabindex="-1"></a><span class="co"># ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</span></span>
<span id="cb113-1218"><a href="#cb113-1218" aria-hidden="true" tabindex="-1"></a><span class="co"># AnnData expects spatial coordinates in .obsm['spatial'], which maps to reducedDims in SingleCellExperiment</span></span>
<span id="cb113-1219"><a href="#cb113-1219" aria-hidden="true" tabindex="-1"></a><span class="co"># reducedDim(spe, "spatial") &lt;- as.matrix(spatial_coords[, c("x", "y")])</span></span>
<span id="cb113-1220"><a href="#cb113-1220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1221"><a href="#cb113-1221" aria-hidden="true" tabindex="-1"></a><span class="co"># Verify spatial coordinates</span></span>
<span id="cb113-1222"><a href="#cb113-1222" aria-hidden="true" tabindex="-1"></a><span class="co"># head(spatialCoords(spe))  # Check SpatialExperiment slot</span></span>
<span id="cb113-1223"><a href="#cb113-1223" aria-hidden="true" tabindex="-1"></a><span class="co"># Should show x and y</span></span>
<span id="cb113-1224"><a href="#cb113-1224" aria-hidden="true" tabindex="-1"></a><span class="co"># Output:</span></span>
<span id="cb113-1225"><a href="#cb113-1225" aria-hidden="true" tabindex="-1"></a><span class="co">#            x     y</span></span>
<span id="cb113-1226"><a href="#cb113-1226" aria-hidden="true" tabindex="-1"></a><span class="co"># AAACAAGTATCTCCCA-1 7237  8230</span></span>
<span id="cb113-1227"><a href="#cb113-1227" aria-hidden="true" tabindex="-1"></a><span class="co"># AAACAATCTACTAGCA-1 1611  4170</span></span>
<span id="cb113-1228"><a href="#cb113-1228" aria-hidden="true" tabindex="-1"></a><span class="co"># ...  </span></span>
<span id="cb113-1229"><a href="#cb113-1229" aria-hidden="true" tabindex="-1"></a><span class="co"># head(reducedDim(spe, "spatial"))  # Check reducedDims for H5AD</span></span>
<span id="cb113-1230"><a href="#cb113-1230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1231"><a href="#cb113-1231" aria-hidden="true" tabindex="-1"></a><span class="co"># Save to H5AD</span></span>
<span id="cb113-1232"><a href="#cb113-1232" aria-hidden="true" tabindex="-1"></a><span class="co"># writeH5AD(spe, file = "~/path/to/your/data/brain_metadata_corrected3_withCoordinatesRegions.h5ad", verbose = TRUE)</span></span>
<span id="cb113-1233"><a href="#cb113-1233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1234"><a href="#cb113-1234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1235"><a href="#cb113-1235" aria-hidden="true" tabindex="-1"></a><span class="co"># Explanation:</span></span>
<span id="cb113-1236"><a href="#cb113-1236" aria-hidden="true" tabindex="-1"></a><span class="co"># reducedDim(spe, "spatial"):</span></span>
<span id="cb113-1237"><a href="#cb113-1237" aria-hidden="true" tabindex="-1"></a><span class="co"># Explicitly stores coordinates in the reducedDims slot of the SpatialExperiment object.</span></span>
<span id="cb113-1238"><a href="#cb113-1238" aria-hidden="true" tabindex="-1"></a><span class="co"># When writing to H5AD, zellkonverter maps reducedDims to .obsm in AnnData.</span></span>
<span id="cb113-1239"><a href="#cb113-1239" aria-hidden="true" tabindex="-1"></a><span class="co"># This ensures coordinates appear in adata.obsm['spatial'] in Python.</span></span>
<span id="cb113-1240"><a href="#cb113-1240" aria-hidden="true" tabindex="-1"></a><span class="co"># Why This Works:</span></span>
<span id="cb113-1241"><a href="#cb113-1241" aria-hidden="true" tabindex="-1"></a><span class="co"># .obsm in AnnData is designed for multi-dimensional annotations (e.g., spatial coordinates).</span></span>
<span id="cb113-1242"><a href="#cb113-1242" aria-hidden="true" tabindex="-1"></a><span class="co"># The reducedDims slot in SingleCellExperiment/SpatialExperiment directly translates to .obsm in H5AD.</span></span>
<span id="cb113-1243"><a href="#cb113-1243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1244"><a href="#cb113-1244" aria-hidden="true" tabindex="-1"></a><span class="co"># VERIFICATION OF H5AD: Read H5AD back into R (optional)</span></span>
<span id="cb113-1245"><a href="#cb113-1245" aria-hidden="true" tabindex="-1"></a><span class="co"># library(zellkonverter)</span></span>
<span id="cb113-1246"><a href="#cb113-1246" aria-hidden="true" tabindex="-1"></a><span class="co"># spe_check &lt;- readH5AD("~/path/to/your/data/brain_metadata_corrected3_withCoordinatesRegions.h5ad")</span></span>
<span id="cb113-1247"><a href="#cb113-1247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1248"><a href="#cb113-1248" aria-hidden="true" tabindex="-1"></a><span class="co"># Confirm coordinates are in reducedDims</span></span>
<span id="cb113-1249"><a href="#cb113-1249" aria-hidden="true" tabindex="-1"></a><span class="co"># head(reducedDim(spe_check, "spatial"))  # Should match original x/y</span></span>
<span id="cb113-1250"><a href="#cb113-1250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1251"><a href="#cb113-1251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-1252"><a href="#cb113-1252" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>
